--- a/third_party/webrtc/modules/video_coding/codecs/av1/dav1d_decoder.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/av1/dav1d_decoder.cc
@@ -18,7 +18,7 @@
 #include "common_video/include/video_frame_buffer.h"
 #include "modules/video_coding/include/video_error_codes.h"
 #include "rtc_base/logging.h"
-#include "third_party/dav1d/libdav1d/include/dav1d/dav1d.h"
+#include <dav1d/dav1d.h>
 #include "third_party/libyuv/include/libyuv/convert.h"
 #include "third_party/libyuv/include/libyuv/planar_functions.h"
 
--- a/media/filters/dav1d_video_decoder.cc
+++ b/media/filters/dav1d_video_decoder.cc
@@ -22,7 +22,7 @@
 #include "media/base/video_util.h"
 
 extern "C" {
-#include "third_party/dav1d/libdav1d/include/dav1d/dav1d.h"
+#include <dav1d/dav1d.h>
 }
 
 namespace media {
--- a/third_party/libavif/src/src/codec_dav1d.c
+++ b/third_party/libavif/src/src/codec_dav1d.c
@@ -10,7 +10,7 @@
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wc11-extensions" // C11 extension used: nameless struct/union
 #endif
-#include "dav1d/dav1d.h"
+#include <dav1d/dav1d.h>
 #if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
