--- a/chrome/browser/browser_process_impl.cc
+++ b/chrome/browser/browser_process_impl.cc
@@ -264,8 +264,10 @@
 #if BUILDFLAG(IS_LINUX)
 #include "chrome/browser/browser_features.h"
 #include "components/os_crypt/async/browser/fallback_linux_key_provider.h"
+#if defined(USE_DBUS)
 #include "components/os_crypt/async/browser/freedesktop_secret_key_provider.h"
 #include "components/os_crypt/async/browser/secret_portal_key_provider.h"
+#endif  // defined(USE_DBUS)
 #endif
 
 #if BUILDFLAG(SAFE_BROWSING_AVAILABLE)
@@ -1149,7 +1151,7 @@
                                 GoogleUpdateSettings::GetCollectStatsConsent());
   registry->RegisterBooleanPref(prefs::kDevToolsRemoteDebuggingAllowed, true);
 
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && defined(USE_DBUS)
   os_crypt_async::SecretPortalKeyProvider::RegisterLocalPrefs(registry);
 #endif
 }
@@ -1416,7 +1418,7 @@
           local_state())));
 #endif  // BUILDFLAG(IS_WIN)
 
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) && defined(USE_DBUS)
   if (base::FeatureList::IsEnabled(features::kDbusSecretPortal)) {
     providers.emplace_back(
         /*precedence=*/10u,
--- a/chrome/browser/browser_features.cc
+++ b/chrome/browser/browser_features.cc
@@ -88,13 +88,19 @@
 // disabled without losing data.
 BASE_FEATURE(kDbusSecretPortal,
              "DbusSecretPortal",
+#if defined(USE_DBUS)
              base::FEATURE_ENABLED_BY_DEFAULT);
+#else
+             base::FEATURE_DISABLED_BY_DEFAULT);
+#endif  // defined(USE_DBUS)
 
 // Enables usage of os_crypt_async::FreedesktopSecretKeyProvider, which is
 // compatible with the synchronous backend.
+#if defined(USE_DBUS)
 BASE_FEATURE(kUseFreedesktopSecretKeyProvider,
              "UseFreedesktopSecretKeyProvider",
              base::FEATURE_DISABLED_BY_DEFAULT);
+#endif  // defined(USE_DBUS)
 #endif  // BUILDFLAG(IS_LINUX)
 
 // Destroy profiles when their last browser window is closed, instead of when
