--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -7008,6 +7008,7 @@
       "metrics/pressure/pressure_metrics_reporter.cc",
       "metrics/pressure/pressure_metrics_reporter.h",
       "obsolete_system/obsolete_system_linux.cc",
+      "platform_util_linux.cc",
       "shell_integration_linux.cc",
       "shell_integration_linux.h",
       "upgrade_detector/directory_monitor.cc",
@@ -7030,7 +7031,6 @@
         "dbus_memory_pressure_evaluator_linux.h",
         "notifications/notification_platform_bridge_linux.cc",
         "notifications/notification_platform_bridge_linux.h",
-        "platform_util_linux.cc",
       ]
       deps += [
         "//chrome/common:chrome_features",
