--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -5594,13 +5594,13 @@
      flag_descriptions::kVulkanFromAngleDescription, kOsLinux | kOsAndroid,
      FEATURE_VALUE_TYPE(features::kVulkanFromANGLE)},
 
-#if !BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(USE_DBUS) && !BUILDFLAG(IS_CHROMEOS)
     {"enable-system-notifications",
      flag_descriptions::kNotificationsSystemFlagName,
      flag_descriptions::kNotificationsSystemFlagDescription,
      kOsMac | kOsLinux | kOsWin,
      FEATURE_VALUE_TYPE(features::kSystemNotifications)},
-#endif  // !BUILDFLAG(IS_CHROMEOS)
+#endif  // BUILDFLAG(USE_DBUS) && !BUILDFLAG(IS_CHROMEOS)
 #if BUILDFLAG(IS_CHROMEOS)
     {"enable-ongoing-processes", flag_descriptions::kEnableOngoingProcessesName,
      flag_descriptions::kEnableOngoingProcessesDescription, kOsCrOS,
--- a/chrome/browser/browser_features.h
+++ b/chrome/browser/browser_features.h
@@ -18,6 +18,7 @@
 #include "build/branding_buildflags.h"
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
+#include "build/config/linux/dbus/buildflags.h"
 #include "extensions/buildflags/buildflags.h"
 
 namespace features {
--- a/chrome/browser/browser_process_impl.cc
+++ b/chrome/browser/browser_process_impl.cc
@@ -917,9 +917,13 @@
 }
 
 NotificationPlatformBridge* BrowserProcessImpl::notification_platform_bridge() {
+#if BUILDFLAG(USE_DBUS)
   if (!created_notification_bridge_)
     CreateNotificationPlatformBridge();
   return notification_bridge_.get();
+#else
+  return nullptr;
+#endif
 }
 
 policy::ChromeBrowserPolicyConnector*
@@ -1500,9 +1504,11 @@
 }
 
 void BrowserProcessImpl::CreateNotificationPlatformBridge() {
+#if BUILDFLAG(USE_DBUS)
   DCHECK(!notification_bridge_);
   notification_bridge_ = NotificationPlatformBridge::Create();
   created_notification_bridge_ = true;
+#endif
 }
 
 void BrowserProcessImpl::CreateNotificationUIManager() {
--- a/chrome/browser/browser_process_impl.h
+++ b/chrome/browser/browser_process_impl.h
@@ -23,6 +23,7 @@
 #include "base/sequence_checker.h"
 #include "base/timer/timer.h"
 #include "build/build_config.h"
+#include "build/config/linux/dbus/buildflags.h"
 #include "chrome/browser/browser_process.h"
 #include "chrome/common/buildflags.h"
 #include "components/keep_alive_registry/keep_alive_state_observer.h"
@@ -361,7 +362,9 @@
 
   std::unique_ptr<StatusTray> status_tray_;
 
+#if BUILDFLAG(USE_DBUS)
   bool created_notification_bridge_ = false;
+#endif
 
   std::unique_ptr<NotificationPlatformBridge> notification_bridge_;
 
--- a/chrome/browser/notifications/notification_interactive_uitest_support.cc
+++ b/chrome/browser/notifications/notification_interactive_uitest_support.cc
@@ -99,8 +99,10 @@
 NotificationsTest::NotificationsTest() {
 // Temporary change while the whole support class is changed to deal
 // with system notifications. crbug.com/714679
+#if BUILDFLAG(USE_DBUS)
   feature_list_.InitWithFeatures(
       {}, {features::kNativeNotifications, features::kSystemNotifications});
+#endif  // BUILDFLAG(USE_DBUS)
 }
 
 int NotificationsTest::GetNotificationCount() {
@@ -260,7 +262,12 @@
 
 NotificationsTestWithPermissionsEmbargo ::
     NotificationsTestWithPermissionsEmbargo() {
+#if BUILDFLAG(USE_DBUS)
   feature_list_.InitWithFeatures(
       {},
       {features::kSystemNotifications});
+#else
+  feature_list_.InitWithFeatures(
+      {});
+#endif  //  BUILDFLAG(USE_DBUS)
 }
--- a/chrome/browser/notifications/notification_platform_bridge_delegator.cc
+++ b/chrome/browser/notifications/notification_platform_bridge_delegator.cc
@@ -50,6 +50,7 @@
 // Please try to keep this comment up to date when changing behaviour on one of
 // the platforms supported by the browser.
 bool SystemNotificationsEnabled(Profile* profile) {
+#if BUILDFLAG(USE_DBUS)
 #if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID)
   return true;
 #elif BUILDFLAG(IS_WIN)
@@ -67,6 +68,9 @@
   return base::FeatureList::IsEnabled(features::kNativeNotifications) &&
          base::FeatureList::IsEnabled(features::kSystemNotifications);
 #endif  // BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID)
+#else
+  return false;
+#endif  // BUILDFLAG(USE_DBUS)
 }
 
 NotificationPlatformBridge* GetSystemNotificationPlatformBridge(
@@ -163,9 +167,11 @@
 NotificationPlatformBridge*
 NotificationPlatformBridgeDelegator::GetBridgeForType(
     NotificationHandler::Type type) {
+#if BUILDFLAG(USE_DBUS)
   // Prefer the system bridge if available and it can handle |type|.
   if (system_bridge_ && NotificationPlatformBridge::CanHandleType(type))
     return system_bridge_;
+#endif  // BUILDFLAG(USE_DBUS)
   return message_center_bridge_;
 }
 
--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -1021,9 +1021,11 @@
 // Enables the use of system notification centers instead of using the Message
 // Center for displaying the toasts. The feature is hardcoded to enabled for
 // Chrome OS.
+#if BUILDFLAG(USE_DBUS)
 BASE_FEATURE(NativeNotifications, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(SystemNotifications, base::FEATURE_ENABLED_BY_DEFAULT);
+#endif
 
 #if BUILDFLAG(IS_MAC)
 // Enables the usage of Apple's new Notification API.
--- a/chrome/common/chrome_features.h
+++ b/chrome/common/chrome_features.h
@@ -15,6 +15,7 @@
 #include "build/branding_buildflags.h"
 #include "build/build_config.h"
 #include "build/buildflag.h"
+#include "build/config/linux/dbus/buildflags.h"
 #include "chrome/common/buildflags.h"
 #include "device/vr/buildflags/buildflags.h"
 #include "extensions/buildflags/buildflags.h"
@@ -731,9 +732,11 @@
 BASE_DECLARE_FEATURE(kShowHiddenNetworkToggle);
 #endif
 
+#if BUILDFLAG(USE_DBUS)
 COMPONENT_EXPORT(CHROME_FEATURES) BASE_DECLARE_FEATURE(kNativeNotifications);
 
 COMPONENT_EXPORT(CHROME_FEATURES) BASE_DECLARE_FEATURE(kSystemNotifications);
+#endif
 
 #if BUILDFLAG(IS_MAC)
 COMPONENT_EXPORT(CHROME_FEATURES) BASE_DECLARE_FEATURE(kNewMacNotificationAPI);
