--- a/.rustfmt.toml
+++ b/.rustfmt.toml
@@ -36,3 +36,5 @@
   # (This entry "wins" over the `!third_party/rust/**` one because it is last.)
   "third_party/rust/chromium_crates_io/vendor/**",
 ]
+
+unstable_features = true
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1281,7 +1281,6 @@
 # Don't allow unstable features to be enabled by `#![feature()]` without
 # additional command line flags.
 config("disallow_unstable_features") {
-  rustflags = [ "-Zallow-features=" ]
 }
 
 # The BUILDCONFIG file sets this config on targets by default, which means when
