--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2487,6 +2487,7 @@
 } else {
   common_optimize_on_cflags = []
   common_optimize_on_ldflags = []
+  extra_cflags = []
 
   if (is_android && (use_lld || use_mold)) {
     # TODO(jdduke) Re-enable on mips after resolving linking
@@ -2627,7 +2628,7 @@
     # vectorization while otherwise optimizing for size.
     rustflags = [ "-Copt-level=s" ]
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags + extra_cflags
 
     if (use_lld) {
       ldflags += [ "-Wl,--lto-CGO2" ]
@@ -2685,7 +2686,7 @@
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags + extra_cflags
     if (use_lld) {
       ldflags += [ "-Wl,--lto-CGO2" ]
     }
@@ -2712,7 +2713,7 @@
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
-    cflags = [ "-O3" ] + common_optimize_on_cflags
+    cflags = [ "-O3" ] + common_optimize_on_cflags + extra_cflags
     if (use_lld) {
       ldflags += [ "-Wl,--lto-CGO3" ]
     }
