--- a/media/audio/BUILD.gn
+++ b/media/audio/BUILD.gn
@@ -151,8 +151,10 @@
   ]
   libs = []
   defines = []
+  configs -= [ "//build/config/compiler:default_optimization" ]
   configs += [
     ":platform_config",
+    "//build/config/compiler:optimize_speed",
     "//media:subcomponent_config",
   ]
 
--- a/media/base/BUILD.gn
+++ b/media/base/BUILD.gn
@@ -415,9 +415,10 @@
   ]
 
   libs = []
+  configs -= [ "//build/config/compiler:default_optimization" ]
   configs += [
     "//build/config:precompiled_headers",
-
+    "//build/config/compiler:optimize_speed",
     "//media:subcomponent_config",
   ]
 
--- a/media/cdm/BUILD.gn
+++ b/media/cdm/BUILD.gn
@@ -16,7 +16,11 @@
       ":*",
       "//media",
     ]
-    configs += [ "//media:subcomponent_config" ]
+    configs -= [ "//build/config/compiler:default_optimization" ]
+    configs += [
+      "//build/config/compiler:optimize_speed",
+      "//media:subcomponent_config",
+    ]
   } else {
     visibility = [ "//media/cdm/library_cdm/clear_key_cdm:clear_key_cdm" ]
   }
--- a/media/ffmpeg/BUILD.gn
+++ b/media/ffmpeg/BUILD.gn
@@ -14,7 +14,11 @@
     "//media",
     "//media/filters",
   ]
-  configs += [ "//media:subcomponent_config" ]
+  configs -= [ "//build/config/compiler:default_optimization" ]
+  configs += [
+    "//build/config/compiler:optimize_speed",
+    "//media:subcomponent_config",
+  ]
   sources = [
     "ffmpeg_common.cc",
     "ffmpeg_common.h",
--- a/media/filters/BUILD.gn
+++ b/media/filters/BUILD.gn
@@ -67,7 +67,11 @@
     "wsola_internals.h",
   ]
 
-  configs += [ "//media:subcomponent_config" ]
+  configs -= [ "//build/config/compiler:default_optimization" ]
+  configs += [
+    "//build/config/compiler:optimize_speed",
+    "//media:subcomponent_config",
+  ]
 
   deps = [
     "//base",
--- a/media/formats/BUILD.gn
+++ b/media/formats/BUILD.gn
@@ -90,7 +90,11 @@
     "//third_party/libwebm",
   ]
 
-  configs += [ "//media:subcomponent_config" ]
+  configs -= [ "//build/config/compiler:default_optimization" ]
+  configs += [
+    "//build/config/compiler:optimize_speed",
+    "//media:subcomponent_config",
+  ]
 
   if (proprietary_codecs) {
     deps += [ "//media/video" ]
--- a/media/gpu/BUILD.gn
+++ b/media/gpu/BUILD.gn
@@ -84,7 +84,9 @@
     "gpu_video_encode_accelerator_factory.h",
   ]
 
+  configs -= [ "//build/config/compiler:default_optimization" ]
   configs += [
+    "//build/config/compiler:optimize_speed",
     "//build/config/compiler:wexit_time_destructors",
     "//build/config/compiler:wglobal_constructors",
   ]
@@ -385,7 +387,9 @@
     ]
   }
 
+  configs -= [ "//build/config/compiler:default_optimization" ]
   configs += [
+    "//build/config/compiler:optimize_speed",
     "//build/config/compiler:wexit_time_destructors",
     "//build/config/compiler:wglobal_constructors",
   ]
@@ -470,7 +474,9 @@
     "command_buffer_helper.h",
   ]
 
+  configs -= [ "//build/config/compiler:default_optimization" ]
   configs += [
+    "//build/config/compiler:optimize_speed",
     "//build/config/compiler:wexit_time_destructors",
     "//build/config/compiler:wglobal_constructors",
   ]
@@ -516,7 +522,9 @@
       "video_frame_mapper_factory.h",
     ]
 
+    configs -= [ "//build/config/compiler:default_optimization" ]
     configs += [
+      "//build/config/compiler:optimize_speed",
       "//build/config/compiler:wexit_time_destructors",
       "//build/config/compiler:wglobal_constructors",
     ]
--- a/media/muxers/BUILD.gn
+++ b/media/muxers/BUILD.gn
@@ -56,7 +56,11 @@
   assert(use_libgav1_parser)
   deps += [ "//third_party/libgav1:libgav1_parser" ]
 
-  configs += [ "//media:subcomponent_config" ]
+  configs -= [ "//build/config/compiler:default_optimization" ]
+  configs += [
+    "//build/config/compiler:optimize_speed",
+    "//media:subcomponent_config",
+  ]
 }
 
 source_set("unit_tests") {
--- a/media/renderers/BUILD.gn
+++ b/media/renderers/BUILD.gn
@@ -98,7 +98,11 @@
     ]
   }
 
-  configs += [ "//media:subcomponent_config" ]
+  configs -= [ "//build/config/compiler:default_optimization" ]
+  configs += [
+    "//build/config/compiler:optimize_speed",
+    "//media:subcomponent_config",
+  ]
 }
 
 # Note: This is a roll-up only target; do not expand the visibility. DEPS should
--- a/media/video/BUILD.gn
+++ b/media/video/BUILD.gn
@@ -54,7 +54,11 @@
     "//ui/gl",
   ]
 
-  configs += [ "//media:subcomponent_config" ]
+  configs -= [ "//build/config/compiler:default_optimization" ]
+  configs += [
+    "//build/config/compiler:optimize_speed",
+    "//media:subcomponent_config",
+  ]
 
   if (enable_libaom) {
     sources += [
