--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2468,6 +2468,17 @@
   common_optimize_on_ldflags = []
   extra_cflags = []
 
+  if (is_clang) {
+    extra_cflags = [
+      "-mllvm",
+      "-polly",
+      "-mllvm",
+      "-polly-run-dce",
+      "-mllvm",
+      "-polly-invariant-load-hoisting",
+    ]
+  }
+
   if (is_android && (use_lld || use_mold)) {
     # TODO(jdduke) Re-enable on mips after resolving linking
     # issues with libc++ (crbug.com/456380).
