--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2387,6 +2387,17 @@
   common_optimize_on_cflags = []
   common_optimize_on_ldflags = []
 
+  if (is_clang) {
+    polly_cflags = [
+      "-mllvm",
+      "-polly",
+      "-mllvm",
+      "-polly-run-dce",
+      "-mllvm",
+      "-polly-invariant-load-hoisting",
+    ]
+  }
+
   if (is_android) {
     # TODO(jdduke) Re-enable on mips after resolving linking
     # issues with libc++ (crbug.com/456380).
@@ -2517,7 +2528,7 @@
     # vectorization while otherwise optimizing for size.
     rustflags = [ "-Copt-level=s" ]
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
@@ -2593,7 +2604,7 @@
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O2" ] + common_optimize_on_cflags
+      cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
     }
     rustflags = [ "-Copt-level=3" ]
   }
@@ -2629,7 +2640,7 @@
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O3" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags + polly_cflags
     }
     rustflags = [ "-Copt-level=3" ]
   }
