--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2491,7 +2491,9 @@
 config("default_init_stack_vars") {
   cflags = []
   if (init_stack_vars && is_clang && !is_nacl && !using_sanitizer) {
-    cflags += [ "-ftrivial-auto-var-init=pattern" ]
+    cflags += [ "-ftrivial-auto-var-init=zero",
+                "-enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang",
+    ]
   }
 }
 
