--- a/build/config/compiler/BUILD.gn	2021-05-19 06:38:44.504582887 +0100
+++ b/build/config/compiler/BUILD.gn	2021-05-22 22:36:56.907722674 +0100
@@ -2563,7 +2564,9 @@
 config("default_init_stack_vars") {
   cflags = []
   if (init_stack_vars && is_clang && !is_nacl && !using_sanitizer) {
-    cflags += [ "-ftrivial-auto-var-init=pattern" ]
+    cflags += [ "-ftrivial-auto-var-init=zero",
+                "-enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang",
+    ]
   }
 }
 
