--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -718,6 +718,17 @@
         }
       }
 
+      # Increase the hot function import multiplier when building with PGO.
+      # The size savings gotten from heavier optimization of cold functions can be
+      # spent on speeding up hot ones. The default is 10, and firefox uses 30 along
+      # with -import-instr-limit=10. To get the same effect as firefox with our
+      # import_instr_limit of 5, we need to use a hot multiplier of 60.
+      if (is_linux && is_clang && chrome_pgo_phase != 0) {
+        ldflags += [
+          "-Wl,-mllvm,-import-hot-multiplier=60",
+        ]
+      }
+
       ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
     }
 
