--- a/DEPS
+++ b/DEPS
@@ -916,7 +916,7 @@
         'sha256sum': '0c4092f5769c5ececd096732b00f2c5c4e1741d4112881a2c83fe7e78271d792',
         'size_bytes': 122758604,
         'generation': 1729629177370195,
-        'condition': 'host_os == "linux" and non_git_source',
+        'condition': 'host_os != "linux" and non_git_source',
       },
       {
         'object_name': 'Mac/rust-toolchain-bca5fdebe0e539d123f33df5f2149d5976392e76-1-llvmorg-20-init-9764-gb81d8e90.tar.xz',
