--- a/DEPS
+++ b/DEPS
@@ -934,7 +934,7 @@
         'sha256sum': '6487fc26114c84c5682c95dc7640ddaed9986f8ec7bd757481c1f682c1be0bf5',
         'size_bytes': 122848984,
         'generation': 1728580687453535,
-        'condition': 'host_os == "linux" and non_git_source',
+        'condition': 'host_os != "linux" and non_git_source',
       },
       {
         'object_name': 'Mac/rust-toolchain-f5cd2c5888011d4d80311e5b771c6da507d860dd-2-llvmorg-20-init-6794-g3dbd929e.tar.xz',
