--- a/DEPS
+++ b/DEPS
@@ -555,23 +555,9 @@
 ]
 
 deps = {
-  # NPM dependencies for JavaScript code coverage.
-  'src/third_party/js_code_coverage/node_modules': {
-    'dep_type': 'gcs',
-    'bucket': 'chromium-nodejs',
-    'objects': [
-      {
-        'object_name': 'js_code_coverage/90d7a8ecae092222e585481b64e9928bcf4a689e723a0af4b94922280cd55a48',
-        'sha256sum': '90d7a8ecae092222e585481b64e9928bcf4a689e723a0af4b94922280cd55a48',
-        'size_bytes': 1472459,
-        'generation': 1716929997740855
-      }
-    ]
-
-  },
   'src/build/linux/debian_bullseye_amd64-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_x64 and non_git_source',
+    'condition': 'checkout_x64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -584,7 +570,7 @@
   },
   'src/build/linux/debian_bullseye_arm64-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_arm64 and non_git_source',
+    'condition': 'checkout_arm64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -597,7 +583,7 @@
   },
   'src/build/linux/debian_bullseye_armhf-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_arm and non_git_source',
+    'condition': 'checkout_arm and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -610,7 +596,7 @@
   },
   'src/build/linux/debian_bullseye_i386-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and (checkout_x86 or checkout_x64) and non_git_source',
+    'condition': '(checkout_x86 or checkout_x64) and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -623,7 +609,7 @@
   },
   'src/build/linux/debian_bullseye_mips64el-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_mips64 and non_git_source',
+    'condition': 'checkout_mips64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -636,7 +622,7 @@
   },
   'src/build/linux/debian_bullseye_mipsel-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_mips and non_git_source',
+    'condition': 'checkout_mips and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -691,7 +677,7 @@
   },
   'src/buildtools/linux64-format': {
     'bucket': 'chromium-clang-format',
-    'condition': 'host_os == "linux" and non_git_source',
+    'condition': 'host_os != "linux" and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -975,10 +961,6 @@
       },
     ],
   },
-  'src/third_party/clang-format/script':
-    Var('chromium_git') +
-    '/external/github.com/llvm/llvm-project/clang/tools/clang-format.git@' +
-    Var('clang_format_revision'),
   'src/buildtools/linux64': {
     'packages': [
       {
@@ -987,7 +969,7 @@
       }
     ],
     'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and non_git_source',
+    'condition': 'host_os != "linux" and non_git_source',
   },
   'src/buildtools/mac': {
     'packages': [
@@ -1016,7 +998,7 @@
         'version': Var('reclient_version'),
       }
     ],
-    'condition': 'non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
 
@@ -1052,7 +1034,7 @@
       },
     ],
     'dep_type': 'cipd',
-    'condition': '(host_os == "linux") and non_git_source',
+    'condition': '(host_os != "linux") and non_git_source',
   },
 
   'src/third_party/apache-windows-arm64': {
@@ -1417,12 +1399,6 @@
     'condition': 'checkout_chromium_password_manager_test_dependencies',
   },
 
-  'src/chrome/test/data/perf/canvas_bench':
-    Var('chromium_git') + '/chromium/canvas_bench.git' + '@' + 'a7b40ea5ae0239517d78845a5fc9b12976bfc732',
-
-  'src/chrome/test/data/perf/frame_rate/content':
-    Var('chromium_git') + '/chromium/frame_rate/content.git' + '@' + 'c10272c88463efeef6bb19c9ec07c42bc8fe22b9',
-
   'src/chrome/test/data/safe_browsing/dmg': {
     'packages': [
       {
@@ -1434,19 +1410,12 @@
     'dep_type': 'cipd',
   },
 
-  'src/chrome/test/data/xr/webvr_info':
-    Var('chromium_git') + '/external/github.com/toji/webvr.info.git' + '@' + 'c58ae99b9ff9e2aa4c524633519570bf33536248',
-
   'src/clank': {
     'url': Var('chrome_git') + '/clank/internal/apps.git' + '@' +
     '025c4822c5fbbb60e1319e771ad023582ddfd279',
     'condition': 'checkout_android and checkout_src_internal',
   },
 
-  'src/docs/website': {
-    'url': Var('chromium_git') + '/website.git' + '@' + '4d214097ba7305ec2b3f2dfcf0aa4fd2104c960a',
-  },
-
   'src/ios/third_party/earl_grey2/src': {
       'url': Var('chromium_git') + '/external/github.com/google/EarlGrey.git' + '@' + '7bf6fffe602a0149a6a028a923484819352ed796',
       'condition': 'checkout_ios',
@@ -1534,9 +1503,6 @@
   'src/net/third_party/quiche/src':
     Var('quiche_git') + '/quiche.git' + '@' +  Var('quiche_revision'),
 
-  'src/testing/libfuzzer/fuzzers/wasm_corpus':
-    Var('chromium_git') + '/v8/fuzzer_wasm_corpus.git' + '@' +  '1df5e50a45db9518a56ebb42cb020a94a090258b',
-
   'src/tools/copybara': {
       'packages' : [
           {
@@ -1568,10 +1534,6 @@
       'dep_type': 'cipd',
   },
 
-  'src/third_party/accessibility_test_framework/src': {
-      'url': Var('chromium_git') + '/external/github.com/google/Accessibility-Test-Framework-for-Android.git' + '@' + '4a764c690353ea136c82f1a696a70bf38d1ef5fe',
-  },
-
   'src/third_party/android_build_tools/protoc/cipd': {
       'packages': [
           {
@@ -1755,7 +1717,7 @@
 
     'src/third_party/blink/renderer/core/css/perftest_data': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-style-perftest',
       'objects': [
           {
@@ -1852,7 +1814,7 @@
   # Download test data for Maps telemetry_gpu_integration_test.
   'src/tools/perf/page_sets/maps_perf_test/dataset': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-telemetry',
       'objects': [
           {
@@ -1888,13 +1850,10 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit.
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
-  'src/third_party/chromium-variations':
-    Var('chromium_git') + '/chromium-variations.git' + '@' + Var('chromium_variations_revision'),
-
   # Dependency for ChromeVox.
   'src/third_party/chromevox/third_party/sre/src': {
       'url': Var('chromium_git') + '/external/github.com/zorkow/speech-rule-engine.git' + '@' + '5a56d4d33d67dc7c692da032d2ebbdefd7de780e',
@@ -1911,9 +1870,6 @@
   'src/third_party/cld_3/src':
     Var('chromium_git') + '/external/github.com/google/cld_3.git' + '@' + 'b48dc46512566f5a2d41118c8c1116c4f96dc661',
 
-  'src/third_party/colorama/src':
-    Var('chromium_git') + '/external/colorama.git' + '@' + '3de9f013df4b470069d03d250224062e8cf15c49',
-
   'src/third_party/cpu_features/src':
     Var('chromium_git') + '/external/github.com/google/cpu_features.git' + '@' + '936b9ab5515dead115606559502e3864958f7f6e',
 
@@ -1929,8 +1885,6 @@
       'condition': 'checkout_linux or checkout_chromeos',
   },
 
-  'src/third_party/crossbench':
-    Var('chromium_git') + '/crossbench.git' + '@' + Var('crossbench_revision'),
 
 
   'src/third_party/depot_tools':
@@ -1995,9 +1949,6 @@
   'src/third_party/freetype/src':
     Var('chromium_git') + '/chromium/src/third_party/freetype2.git' + '@' + Var('freetype_revision'),
 
-  'src/third_party/freetype-testing/src':
-    Var('chromium_git') + '/external/github.com/freetype/freetype2-testing.git' + '@' + Var('freetype_testing_revision'),
-
   'src/third_party/fxdiv/src':
     Var('chromium_git') + '/external/github.com/Maratyszcza/FXdiv.git' + '@' + '63058eff77e11aa15bf531df5dd34395ec3017c8',
 
@@ -2090,7 +2041,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit.
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -2146,7 +2097,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit (for checkstyle).
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -2197,15 +2148,6 @@
   'src/third_party/leveldatabase/src':
     Var('chromium_git') + '/external/leveldb.git' + '@' + '578eeb702ec0fbb6b9780f3d4147b1076630d633',
 
-  'src/third_party/libFuzzer/src':
-    Var('chromium_git') + '/external/github.com/llvm/llvm-project/compiler-rt/lib/fuzzer.git' + '@' +  Var('libfuzzer_revision'),
-
-  'src/third_party/fuzztest/src':
-    Var('chromium_git') + '/external/github.com/google/fuzztest.git' + '@' + Var('fuzztest_revision'),
-
-  'src/third_party/domato/src':
-    Var('chromium_git') + '/external/github.com/googleprojectzero/domato.git' + '@' + Var('domato_revision'),
-
   'src/third_party/libaddressinput/src':
     Var('chromium_git') + '/external/libaddressinput.git' + '@' + 'e8712e415627f22d0b00ebee8db99547077f39bd',
 
@@ -2218,33 +2160,6 @@
   'src/third_party/crabbyavif/src':
     Var('chromium_git') + '/external/github.com/webmproject/CrabbyAvif.git' + '@' + Var('crabbyavif_revision'),
 
-  'src/third_party/nearby/src':
-    Var('chromium_git') + '/external/github.com/google/nearby-connections.git' + '@' + Var('nearby_revision'),
-
-  'src/third_party/beto-core/src':
-    Var('betocore_git') + '/beto-core.git' + '@' + Var('betocore_revision'),
-
-  'src/third_party/securemessage/src':
-    Var('chromium_git') + '/external/github.com/google/securemessage.git' + '@' + Var('securemessage_revision'),
-
-  'src/third_party/speedometer/main':
-    Var('chromium_git') + '/external/github.com/WebKit/Speedometer.git' + '@' + Var('speedometer_main_revision'),
-
-  'src/third_party/speedometer/v3.0':
-    Var('chromium_git') + '/external/github.com/WebKit/Speedometer.git' + '@' + Var('speedometer_3.0_revision'),
-
-  'src/third_party/speedometer/v2.0':
-    Var('chromium_git') + '/external/github.com/WebKit/Speedometer.git' + '@' + Var('speedometer_2.0_revision'),
-
-  'src/third_party/speedometer/v2.1':
-    Var('chromium_git') + '/external/github.com/WebKit/Speedometer.git' + '@' + Var('speedometer_2.1_revision'),
-
-  'src/third_party/ukey2/src':
-    Var('chromium_git') + '/external/github.com/google/ukey2.git' + '@' + Var('ukey2_revision'),
-
-  'src/third_party/cros-components/src':
-    Var('chromium_git') + '/external/google3/cros_components.git' + '@' + Var('cros_components_revision'),
-
   # Userspace interface to kernel DRM services.
   'src/third_party/libdrm/src': {
       'url': Var('chromium_git') + '/chromiumos/third_party/libdrm.git' + '@' + 'ad78bb591d02162d3b90890aa4d0a238b2a37cde',
@@ -2257,7 +2172,7 @@
   # The library for IPP protocol (Chrome OS).
   'src/third_party/libipp/libipp': {
       'url': Var('chromium_git') + '/chromiumos/platform2/libipp.git' + '@' + '2209bb84a8e122dab7c02fe66cc61a7b42873d7f',
-      'condition': 'checkout_linux',
+      'condition': 'checkout_src_internal and checkout_linux',
   },
 
   'src/third_party/libjpeg_turbo':
@@ -2265,7 +2180,7 @@
 
   'src/third_party/liblouis/src': {
       'url': Var('chromium_git') + '/external/liblouis-github.git' + '@' + '9700847afb92cb35969bdfcbbfbbb74b9c7b3376',
-      'condition': 'checkout_linux',
+      'condition': 'checkout_src_internal and checkout_linux',
   },
 
   'src/third_party/libphonenumber/dist':
@@ -2288,9 +2203,6 @@
       'condition': 'checkout_android',
   },
 
-  'src/third_party/libva-fake-driver/src':
-    Var('chromium_git') + '/chromiumos/platform/libva-fake-driver.git' + '@' + 'a9bcab9cd6b15d4e3634ca44d5e5f7652c612194',
-
   'src/third_party/libvpx/source/libvpx':
     Var('chromium_git') + '/webm/libvpx.git' + '@' +  '8058a0b54991257a0e1a2fcf08d993a8b70c1d3a',
 
@@ -2392,7 +2304,7 @@
         'version': Var('ninja_version'),
       }
     ],
-    'condition': 'non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
   'src/third_party/siso/cipd': {
@@ -2402,7 +2314,7 @@
         'version': Var('siso_version'),
       }
     ],
-    'condition': 'non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
 
@@ -2419,7 +2331,7 @@
 
   'src/third_party/opus/tests/resources': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-webrtc-resources',
       'objects': [
           {
@@ -2440,6 +2352,7 @@
 
   'src/base/tracing/test/data': {
     'bucket': 'perfetto',
+    'condition': 'checkout_src_internal',
     'objects': [
       {
         'object_name': 'test_data/chrome_fcp_lcp_navigations.pftrace-ae01d849fbd75a98be1b7ddd5a8873217c377b393a1d5bbd788ed3364f7fefc3',
@@ -2581,12 +2494,6 @@
       'condition': 'checkout_linux',
   },
 
-  'src/third_party/quic_trace/src':
-    Var('chromium_git') + '/external/github.com/google/quic-trace.git' + '@' + '413da873d93a03d3662f24b881ea459a79f9c589',
-
-  'src/third_party/pywebsocket3/src':
-    Var('chromium_git') + '/external/github.com/GoogleChromeLabs/pywebsocket3.git' + '@' + '50602a14f1b6da17e0b619833a13addc6ea78bc2',
-
   'src/third_party/qemu-linux-arm64': {
       'packages': [
           {
@@ -2683,7 +2590,7 @@
 
   'src/third_party/subresource-filter-ruleset/data': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-ads-detection',
       'objects': [
           {
@@ -2723,7 +2630,7 @@
 
   'src/third_party/test_fonts/test_fonts': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-fonts',
       'objects': [
           {
@@ -2735,9 +2642,6 @@
       ],
   },
 
-  'src/third_party/text-fragments-polyfill/src':
-    Var('chromium_git') + '/external/github.com/GoogleChromeLabs/text-fragments-polyfill.git' + '@' + 'c036420683f672d685e27415de0a5f5e85bdc23f',
-
   'src/third_party/tflite/src':
     Var('chromium_git') + '/external/github.com/tensorflow/tensorflow.git' + '@' + 'b25df276c8e912c22f57263ffcae6ca8f4c64342',
 
@@ -2802,9 +2706,6 @@
   'src/third_party/webgpu-cts/src':
     Var('chromium_git') + '/external/github.com/gpuweb/cts.git' + '@' + 'e99550b44ea9eedb364beb553d1a48549d0da115',
 
-  'src/third_party/webpagereplay':
-    Var('chromium_git') + '/webpagereplay.git' + '@' + Var('webpagereplay_revision'),
-
   'src/third_party/webrtc':
     Var('webrtc_git') + '/src.git' + '@' + 'cd3e2951ff0f36fa12bea747862c52533a2b39f3',
 
@@ -2833,7 +2734,7 @@
 
   'src/third_party/xdg-utils': {
       'url': Var('chromium_git') + '/chromium/deps/xdg-utils.git' + '@' + 'cb54d9db2e535ee4ef13cc91b65a1e2741a94a44',
-      'condition': 'checkout_linux',
+      'condition': 'checkout_src_internal',
   },
 
   'src/third_party/xnnpack/src':
@@ -2848,7 +2749,7 @@
       },
     ],
 
-    'condition': 'checkout_linux and non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
 
@@ -4319,7 +4220,7 @@
           'version': Var('resultdb_version'),
         },
       ],
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -4742,7 +4643,7 @@
               'version': Var('screen_ai_linux'),
           },
       ],
-      'condition': 'checkout_linux and non_git_source',
+      'condition': 'non_git_source',
       'dep_type': 'cipd',
   },
 
