--- a/DEPS
+++ b/DEPS
@@ -581,22 +581,9 @@
 ]
 
 deps = {
-  # NPM dependencies for JavaScript code coverage.
-  'src/third_party/js_code_coverage/node_modules': {
-    'dep_type': 'gcs',
-    'bucket': 'chromium-nodejs',
-    'objects': [
-      {
-        'object_name': 'js_code_coverage/e932c86d2d4f250416970dc270002a9cb6acecbec034998cdadf9a394d0f1abc',
-        'sha256sum': 'e932c86d2d4f250416970dc270002a9cb6acecbec034998cdadf9a394d0f1abc',
-        'size_bytes': 1557200,
-        'generation': 1742338539536352,
-      }
-    ]
-  },
   'src/build/linux/debian_bullseye_amd64-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_x64 and non_git_source',
+    'condition': 'checkout_x64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -609,7 +596,7 @@
   },
   'src/build/linux/debian_bullseye_arm64-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_arm64 and non_git_source',
+    'condition': 'checkout_arm64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -622,7 +609,7 @@
   },
   'src/build/linux/debian_bullseye_armhf-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_arm and non_git_source',
+    'condition': 'checkout_arm and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -635,7 +622,7 @@
   },
   'src/build/linux/debian_bullseye_i386-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and (checkout_x86 or checkout_x64) and non_git_source',
+    'condition': '(checkout_x86 or checkout_x64) and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -648,7 +635,7 @@
   },
   'src/build/linux/debian_bullseye_mips64el-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_mips64 and non_git_source',
+    'condition': 'checkout_mips64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -661,7 +648,7 @@
   },
   'src/build/linux/debian_bullseye_mipsel-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_mips and non_git_source',
+    'condition': 'checkout_mips and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -729,7 +716,7 @@
   },
   'src/buildtools/linux64-format': {
     'bucket': 'chromium-clang-format',
-    'condition': 'host_os == "linux" and non_git_source',
+    'condition': 'host_os != "linux" and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -1013,10 +1000,6 @@
       },
     ],
   },
-  'src/third_party/clang-format/script':
-    Var('chromium_git') +
-    '/external/github.com/llvm/llvm-project/clang/tools/clang-format.git@' +
-    Var('clang_format_revision'),
   'src/buildtools/linux64': {
     'packages': [
       {
@@ -1025,7 +1008,7 @@
       }
     ],
     'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and non_git_source',
+    'condition': 'host_os != "linux" and non_git_source',
   },
   'src/buildtools/mac': {
     'packages': [
@@ -1054,7 +1037,7 @@
         'version': Var('reclient_version'),
       }
     ],
-    'condition': '(host_cpu != "arm64" or host_os == "mac") and non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
 
@@ -1112,7 +1095,7 @@
       },
     ],
     'dep_type': 'cipd',
-    'condition': '(host_os == "linux") and non_git_source',
+    'condition': '(host_os != "linux") and non_git_source',
   },
 
   'src/third_party/apache-windows-arm64': {
@@ -1476,12 +1459,6 @@
     'condition': 'checkout_chromium_password_manager_test_dependencies',
   },
 
-  'src/chrome/test/data/perf/canvas_bench':
-    Var('chromium_git') + '/chromium/canvas_bench.git' + '@' + 'a7b40ea5ae0239517d78845a5fc9b12976bfc732',
-
-  'src/chrome/test/data/perf/frame_rate/content':
-    Var('chromium_git') + '/chromium/frame_rate/content.git' + '@' + 'c10272c88463efeef6bb19c9ec07c42bc8fe22b9',
-
   'src/chrome/test/data/safe_browsing/dmg': {
     'packages': [
       {
@@ -1493,29 +1470,12 @@
     'dep_type': 'cipd',
   },
 
-  'src/components/variations/test_data/cipd': {
-    'packages': [
-      {
-        'package': 'chromium/chrome/test/data/variations/cipd',
-        'version': 'ReWAdAhFsmTN1Z6g09iwNFqpNxVIeGzINfzDf4EumCUC',
-      },
-    ],
-    'dep_type': 'cipd',
-  },
-
-  'src/chrome/test/data/xr/webvr_info':
-    Var('chromium_git') + '/external/github.com/toji/webvr.info.git' + '@' + 'c58ae99b9ff9e2aa4c524633519570bf33536248',
-
   'src/clank': {
     'url': Var('chrome_git') + '/clank/internal/apps.git' + '@' +
     '4b036b79faa808b94a0217e847c71a0e64663fd5',
     'condition': 'checkout_android and checkout_src_internal',
   },
 
-  'src/docs/website': {
-    'url': Var('chromium_git') + '/website.git' + '@' + 'd21d90790d8ea421b317c4cb52a0d94133422796',
-  },
-
   'src/ios/third_party/earl_grey2/src': {
       'url': Var('chromium_git') + '/external/github.com/google/EarlGrey.git' + '@' + '184a28ad9394b196417b6cedfe1121e93533b94d',
       'condition': 'checkout_ios',
@@ -1603,9 +1563,6 @@
   'src/net/third_party/quiche/src':
     Var('quiche_git') + '/quiche.git' + '@' +  Var('quiche_revision'),
 
-  'src/testing/libfuzzer/fuzzers/wasm_corpus':
-    Var('chromium_git') + '/v8/fuzzer_wasm_corpus.git' + '@' +  '1df5e50a45db9518a56ebb42cb020a94a090258b',
-
   'src/tools/copybara': {
       'packages' : [
           {
@@ -1637,10 +1594,6 @@
       'dep_type': 'cipd',
   },
 
-  'src/third_party/accessibility_test_framework/src': {
-      'url': Var('chromium_git') + '/external/github.com/google/Accessibility-Test-Framework-for-Android.git' + '@' + '4a764c690353ea136c82f1a696a70bf38d1ef5fe',
-  },
-
   'src/third_party/android_build_tools/protoc/cipd': {
       'packages': [
           {
@@ -1824,7 +1777,7 @@
 
     'src/third_party/blink/renderer/core/css/perftest_data': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-style-perftest',
       'objects': [
           {
@@ -1921,7 +1874,7 @@
   # Download test data for Maps telemetry_gpu_integration_test.
   'src/tools/perf/page_sets/maps_perf_test/dataset': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-telemetry',
       'objects': [
           {
@@ -1957,7 +1910,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit.
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -1977,9 +1930,6 @@
   'src/third_party/cld_3/src':
     Var('chromium_git') + '/external/github.com/google/cld_3.git' + '@' + 'b48dc46512566f5a2d41118c8c1116c4f96dc661',
 
-  'src/third_party/colorama/src':
-    Var('chromium_git') + '/external/colorama.git' + '@' + '3de9f013df4b470069d03d250224062e8cf15c49',
-
   'src/third_party/cpu_features/src':
     Var('chromium_git') + '/external/github.com/google/cpu_features.git' + '@' + '936b9ab5515dead115606559502e3864958f7f6e',
 
@@ -1995,8 +1945,6 @@
       'condition': 'checkout_linux or checkout_chromeos',
   },
 
-  'src/third_party/crossbench':
-    Var('chromium_git') + '/crossbench.git' + '@' + Var('crossbench_revision'),
 
 
   'src/third_party/depot_tools':
@@ -2070,9 +2018,6 @@
   'src/third_party/freetype/src':
     Var('chromium_git') + '/chromium/src/third_party/freetype2.git' + '@' + Var('freetype_revision'),
 
-  'src/third_party/freetype-testing/src':
-    Var('chromium_git') + '/external/github.com/freetype/freetype2-testing.git' + '@' + Var('freetype_testing_revision'),
-
   'src/third_party/fxdiv/src':
     Var('chromium_git') + '/external/github.com/Maratyszcza/FXdiv.git' + '@' + '63058eff77e11aa15bf531df5dd34395ec3017c8',
 
@@ -2165,7 +2110,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit.
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -2221,7 +2166,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit (for checkstyle).
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -2272,15 +2217,6 @@
   'src/third_party/leveldatabase/src':
     Var('chromium_git') + '/external/leveldb.git' + '@' + '4ee78d7ea98330f7d7599c42576ca99e3c6ff9c5',
 
-  'src/third_party/libFuzzer/src':
-    Var('chromium_git') + '/external/github.com/llvm/llvm-project/compiler-rt/lib/fuzzer.git' + '@' +  Var('libfuzzer_revision'),
-
-  'src/third_party/fuzztest/src':
-    Var('chromium_git') + '/external/github.com/google/fuzztest.git' + '@' + Var('fuzztest_revision'),
-
-  'src/third_party/domato/src':
-    Var('chromium_git') + '/external/github.com/googleprojectzero/domato.git' + '@' + Var('domato_revision'),
-
   'src/third_party/libaddressinput/src':
     Var('chromium_git') + '/external/libaddressinput.git' + '@' + '2610f7b1043d6784ada41392fc9392d1ea09ea07',
 
@@ -2290,12 +2226,6 @@
   'src/third_party/crabbyavif/src':
     Var('chromium_git') + '/external/github.com/webmproject/CrabbyAvif.git' + '@' + Var('crabbyavif_revision'),
 
-  'src/third_party/nearby/src':
-    Var('chromium_git') + '/external/github.com/google/nearby-connections.git' + '@' + Var('nearby_revision'),
-
-  'src/third_party/securemessage/src':
-    Var('chromium_git') + '/external/github.com/google/securemessage.git' + '@' + Var('securemessage_revision'),
-
   'src/third_party/jetstream/main': {
     'url': Var('chromium_git') + '/external/github.com/WebKit/JetStream.git' + '@' + Var('jetstream_main_revision'),
     'condition': 'checkout_press_benchmarks',
@@ -2331,12 +2261,6 @@
     'condition': 'checkout_press_benchmarks',
   },
 
-  'src/third_party/ukey2/src':
-    Var('chromium_git') + '/external/github.com/google/ukey2.git' + '@' + Var('ukey2_revision'),
-
-  'src/third_party/cros-components/src':
-    Var('chromium_git') + '/external/google3/cros_components.git' + '@' + Var('cros_components_revision'),
-
   'src/third_party/libdisplay-info/src': {
       'url': Var('chromium_git') + '/external/gitlab.freedesktop.org/emersion/libdisplay-info.git' + '@' + '66b802d05b374cd8f388dc6ad1e7ae4f08cb3300',
       'condition': 'checkout_linux and checkout_mutter',
@@ -2364,7 +2288,7 @@
   # The library for IPP protocol (Chrome OS).
   'src/third_party/libipp/libipp': {
       'url': Var('chromium_git') + '/chromiumos/platform2/libipp.git' + '@' + '2209bb84a8e122dab7c02fe66cc61a7b42873d7f',
-      'condition': 'checkout_linux',
+      'condition': 'checkout_src_internal and checkout_linux',
   },
 
   'src/third_party/libjpeg_turbo':
@@ -2372,7 +2296,7 @@
 
   'src/third_party/liblouis/src': {
       'url': Var('chromium_git') + '/external/liblouis-github.git' + '@' + '9700847afb92cb35969bdfcbbfbbb74b9c7b3376',
-      'condition': 'checkout_linux',
+      'condition': 'checkout_src_internal and checkout_linux',
   },
 
   'src/third_party/libphonenumber/dist':
@@ -2395,9 +2319,6 @@
       'condition': 'checkout_android',
   },
 
-  'src/third_party/libva-fake-driver/src':
-    Var('chromium_git') + '/chromiumos/platform/libva-fake-driver.git' + '@' + 'a9bcab9cd6b15d4e3634ca44d5e5f7652c612194',
-
   'src/third_party/libvpx/source/libvpx':
     Var('chromium_git') + '/webm/libvpx.git' + '@' +  'b84ca9b63730e7d4563573a56a66317eb0087ebf',
 
@@ -2504,7 +2425,7 @@
         'version': Var('ninja_version'),
       }
     ],
-    'condition': 'non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
   'src/third_party/siso/cipd': {
@@ -2514,7 +2435,7 @@
         'version': Var('siso_version'),
       }
     ],
-    'condition': 'non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
 
@@ -2531,7 +2452,7 @@
 
   'src/third_party/opus/tests/resources': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-webrtc-resources',
       'objects': [
           {
@@ -2552,6 +2473,7 @@
 
   'src/base/tracing/test/data': {
     'bucket': 'perfetto',
+    'condition': 'checkout_src_internal',
     'objects': [
       {
         'object_name': 'test_data/chrome_fcp_lcp_navigations.pftrace-ae01d849fbd75a98be1b7ddd5a8873217c377b393a1d5bbd788ed3364f7fefc3',
@@ -2700,12 +2622,6 @@
       'condition': 'checkout_linux',
   },
 
-  'src/third_party/quic_trace/src':
-    Var('chromium_git') + '/external/github.com/google/quic-trace.git' + '@' + 'ed3deb8a056b260c59f2fd42af6dfa3db48a8cad',
-
-  'src/third_party/pywebsocket3/src':
-    Var('chromium_git') + '/external/github.com/GoogleChromeLabs/pywebsocket3.git' + '@' + '50602a14f1b6da17e0b619833a13addc6ea78bc2',
-
   'src/third_party/qemu-linux-arm64': {
       'packages': [
           {
@@ -2798,7 +2714,7 @@
 
   'src/third_party/subresource-filter-ruleset/data': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-ads-detection',
       'objects': [
           {
@@ -2838,7 +2754,7 @@
 
   'src/third_party/test_fonts/test_fonts': {
       'dep_type': 'gcs',
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'bucket': 'chromium-fonts',
       'objects': [
           {
@@ -2850,9 +2766,6 @@
       ],
   },
 
-  'src/third_party/text-fragments-polyfill/src':
-    Var('chromium_git') + '/external/github.com/GoogleChromeLabs/text-fragments-polyfill.git' + '@' + 'c036420683f672d685e27415de0a5f5e85bdc23f',
-
   'src/third_party/tflite/src':
     Var('chromium_git') + '/external/github.com/tensorflow/tensorflow.git' + '@' + '151774faba661a5985a8264653f4457c70a56dea',
 
@@ -2941,7 +2854,7 @@
 
   'src/third_party/xdg-utils': {
       'url': Var('chromium_git') + '/chromium/deps/xdg-utils.git' + '@' + 'cb54d9db2e535ee4ef13cc91b65a1e2741a94a44',
-      'condition': 'checkout_linux',
+      'condition': 'checkout_src_internal',
   },
 
   'src/third_party/xnnpack/src':
@@ -2956,7 +2869,7 @@
       },
     ],
 
-    'condition': 'checkout_linux and non_git_source',
+    'condition': 'checkout_src_internal and non_git_source',
     'dep_type': 'cipd',
   },
 
@@ -4386,7 +4299,7 @@
           'version': Var('resultdb_version'),
         },
       ],
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -4760,7 +4673,7 @@
               'version': Var('screen_ai_linux'),
           },
       ],
-      'condition': 'checkout_linux and non_git_source',
+      'condition': 'non_git_source',
       'dep_type': 'cipd',
   },
 
