--- a/DEPS
+++ b/DEPS
@@ -1031,7 +1031,7 @@
   'src/third_party/rust-toolchain': {
     'dep_type': 'gcs',
     'bucket': 'chromium-browser-clang',
-    'condition': 'not rust_force_head_revision',
+    'condition': 'rust_force_head_revision',
     'objects': [
       {
         'object_name': 'Linux_x64/rust-toolchain-b998449636a48e2c4a362809085b600a0174e1f2-5-llvmorg-23-init-19482-g53d18800.tar.xz',
