--- a/DEPS
+++ b/DEPS
@@ -259,6 +259,9 @@
   # instead of downloading the prebuilt pinned revision.
   'llvm_force_head_revision': False,
 
+  # Fetch Rust toolchain.
+  'checkout_rust': False,
+
   # Make Dawn skip its standalone dependencies
   'dawn_standalone': False,
 
@@ -4540,6 +4543,7 @@
     # Update prebuilt Rust toolchain.
     'name': 'rust-toolchain',
     'pattern': '.',
+    'condition': 'checkout_rust',
     'action': ['python3', 'src/tools/rust/update_rust.py'],
   },
   {
