--- a/DEPS
+++ b/DEPS
@@ -2281,7 +2281,7 @@
           'version': Var('cpython3_version'),
         },
       ],
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -2294,7 +2294,7 @@
           'version': Var('cpython3_version'),
         },
       ],
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -4303,6 +4303,7 @@
   {
     'name': 'vpython3_common',
     'pattern': '.',
+    'condition': 'checkout_src_internal',
     'action': [ 'vpython3',
                 '-vpython-spec', 'src/.vpython3',
                 '-vpython-tool', 'install',
