--- a/DEPS
+++ b/DEPS
@@ -1013,7 +1013,7 @@
         'sha256sum': '3e5cf980edb893cbdc915d62bce1b29b896eda6df6455e145200bf25a52576b1',
         'size_bytes': 159517088,
         'generation': 1756377175296503,
-        'condition': 'host_os == "linux" and non_git_source',
+        'condition': 'host_os != "linux" and non_git_source',
       },
       {
         'object_name': 'Mac/rust-toolchain-22be76b7e259f27bf3e55eb931f354cd8b69d55f-4-llvmorg-21-init-16348-gbd809ffb.tar.xz',
