--- a/DEPS
+++ b/DEPS
@@ -556,23 +556,9 @@
 ]
 
 deps = {
-  # NPM dependencies for JavaScript code coverage.
-  'src/third_party/js_code_coverage/node_modules': {
-    'dep_type': 'gcs',
-    'bucket': 'chromium-nodejs',
-    'objects': [
-      {
-        'object_name': 'js_code_coverage/90d7a8ecae092222e585481b64e9928bcf4a689e723a0af4b94922280cd55a48',
-        'sha256sum': '90d7a8ecae092222e585481b64e9928bcf4a689e723a0af4b94922280cd55a48',
-        'size_bytes': 1472459,
-        'generation': 1716929997740855
-      }
-    ]
-
-  },
   'src/build/linux/debian_bullseye_amd64-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_x64 and non_git_source',
+    'condition': 'checkout_x64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -585,7 +571,7 @@
   },
   'src/build/linux/debian_bullseye_arm64-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_arm64 and non_git_source',
+    'condition': 'checkout_arm64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -598,7 +584,7 @@
   },
   'src/build/linux/debian_bullseye_armhf-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_arm and non_git_source',
+    'condition': 'checkout_arm and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -611,7 +597,7 @@
   },
   'src/build/linux/debian_bullseye_i386-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and (checkout_x86 or checkout_x64) and non_git_source',
+    'condition': '(checkout_x86 or checkout_x64) and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -624,7 +610,7 @@
   },
   'src/build/linux/debian_bullseye_mips64el-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_mips64 and non_git_source',
+    'condition': 'checkout_mips64 and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -637,7 +623,7 @@
   },
   'src/build/linux/debian_bullseye_mipsel-sysroot': {
     'bucket': 'chrome-linux-sysroot',
-    'condition': 'checkout_linux and checkout_mips and non_git_source',
+    'condition': 'checkout_mips and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -690,20 +676,6 @@
       },
     ],
   },
-  'src/buildtools/linux64-format': {
-    'bucket': 'chromium-clang-format',
-    'condition': 'host_os == "linux" and non_git_source',
-    'dep_type': 'gcs',
-    'objects': [
-      {
-        'object_name': 'b42097ca924d1f1736a5a7806068fed9d7345eb4',
-        'sha256sum': '82df59a7d4390892c3eeaf0c8bf626e2869f1138a6ad3eb90dd51da0011ba630',
-        'size_bytes': 3539912,
-        'generation': 1699478806427152,
-        'output_file': 'clang-format',
-      },
-    ],
-  },
   # Pull down Node binaries for WebUI toolchain.
   # The Linux binary is always downloaded regardless of host os and architecture
   # since remote node actions run on Linux worker.
@@ -768,19 +740,6 @@
           },
       ],
   },
-  # Download selected models from TFHub as testdata.
-  'src/third_party/tfhub_models/testdata': {
-    'bucket': 'chromium-tfhub-models',
-    'dep_type': 'gcs',
-    'objects': [
-      {
-        'object_name': '0f037afd23a02321520951afd5c2c6078d26cbf1',
-        'sha256sum': '7130f43eb9889ff4dcd36ed2c5352053b88216e6b9186dfce08ea41b7dd142f3',
-        'size_bytes': 35504613,
-        'generation': 1691086948259727,
-      },
-    ],
-  },
   'src/third_party/llvm-build/Release+Asserts': {
     'dep_type': 'gcs',
     'bucket': 'chromium-browser-clang',
@@ -963,20 +922,6 @@
       },
     ],
   },
-  'src/third_party/clang-format/script':
-    Var('chromium_git') +
-    '/external/github.com/llvm/llvm-project/clang/tools/clang-format.git@' +
-    Var('clang_format_revision'),
-  'src/buildtools/linux64': {
-    'packages': [
-      {
-        'package': 'gn/gn/linux-${{arch}}',
-        'version': Var('gn_version'),
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and non_git_source',
-  },
   'src/buildtools/mac': {
     'packages': [
       {
@@ -997,16 +942,6 @@
     'dep_type': 'cipd',
     'condition': 'host_os == "win"',
   },
-  'src/buildtools/reclient': {
-    'packages': [
-      {
-        'package': Var('reclient_package') + '${{platform}}',
-        'version': Var('reclient_version'),
-      }
-    ],
-    'condition': 'non_git_source',
-    'dep_type': 'cipd',
-  },
 
   # We don't know target_cpu at deps time. At least until there's a universal
   # binary of httpd-php, pull both intel and arm versions in DEPS and then pick
@@ -1032,17 +967,6 @@
     'condition': 'checkout_mac or checkout_ios',
   },
 
-  'src/third_party/apache-linux': {
-    'packages': [
-      {
-        'package': 'infra/3pp/tools/httpd-php/linux-amd64',
-        'version': 'version:2@httpd2.4.38.php7.3.31.chromium.3',
-      },
-    ],
-    'dep_type': 'cipd',
-    'condition': '(host_os == "linux") and non_git_source',
-  },
-
   'src/third_party/apache-windows-arm64': {
     'packages': [
         {
@@ -1232,12 +1156,6 @@
     'condition': 'checkout_chromium_password_manager_test_dependencies',
   },
 
-  'src/chrome/test/data/perf/canvas_bench':
-    Var('chromium_git') + '/chromium/canvas_bench.git' + '@' + 'a7b40ea5ae0239517d78845a5fc9b12976bfc732',
-
-  'src/chrome/test/data/perf/frame_rate/content':
-    Var('chromium_git') + '/chromium/frame_rate/content.git' + '@' + 'c10272c88463efeef6bb19c9ec07c42bc8fe22b9',
-
   'src/chrome/test/data/safe_browsing/dmg': {
     'packages': [
       {
@@ -1249,9 +1167,6 @@
     'dep_type': 'cipd',
   },
 
-  'src/chrome/test/data/xr/webvr_info':
-    Var('chromium_git') + '/external/github.com/toji/webvr.info.git' + '@' + 'c58ae99b9ff9e2aa4c524633519570bf33536248',
-
   'src/clank': {
     'url': Var('chrome_git') + '/clank/internal/apps.git' + '@' +
     '65599068d951070dbecb482c21a1c3eb01f9dfa8',
@@ -1349,9 +1264,6 @@
   'src/net/third_party/quiche/src':
     Var('quiche_git') + '/quiche.git' + '@' +  Var('quiche_revision'),
 
-  'src/testing/libfuzzer/fuzzers/wasm_corpus':
-    Var('chromium_git') + '/v8/fuzzer_wasm_corpus.git' + '@' +  'f650ff816f2ef227f61ea2e9f222aa69708ab367',
-
   'src/tools/luci-go': {
       'packages': [
         {
@@ -1524,77 +1436,6 @@
   'src/third_party/anonymous_tokens/src':
     Var('chromium_git') + '/external/github.com/google/anonymous-tokens.git' + '@' + '76bfcccb6418239183df55111f2f24782d9f3680',
 
-    'src/third_party/blink/renderer/core/css/perftest_data': {
-      'dep_type': 'gcs',
-      'condition': 'non_git_source',
-      'bucket': 'chromium-style-perftest',
-      'objects': [
-          {
-              'object_name': 'e9ce994346c62f8c9fd6d0cecb2b2b0b93b4c2d8',
-              'sha256sum': '519019df16c628c6c0893df18928faeaa3150a9d8f26a787a16ce7c6b2cec2ad',
-              'size_bytes': 601672,
-              'generation': 1664794185950162,
-              'output_file': 'ecommerce.json',
-          },
-          {
-              'object_name': '756068da5e551516b23b0ba133e55c144f623d38',
-              'sha256sum': '84ef87a8163335a95111d9709306596f96742539da0b34fbe7397f799946a168',
-              'size_bytes': 2156935,
-              'generation': 1664794188995509,
-              'output_file': 'encyclopedia.json',
-          },
-          {
-              'object_name': '314e4e0d5e89ea9e9e9a234c617b4413adf48aa9',
-              'sha256sum': 'a721ada40011a286631baae6d76878f2023ff000151792228c83b1958ea8a197',
-              'size_bytes': 608840,
-              'generation': 1664794191929032,
-              'output_file': 'extension.json',
-          },
-          {
-              'object_name': '3a19b42a7c46257b716d55d6733f070c87180b1e',
-              'sha256sum': 'f203ff9e8c8a6a3b714f0a26db38cc940544a907435c62c79b21f4bd3f8bee8e',
-              'size_bytes': 1750837,
-              'generation': 1664794194891567,
-              'output_file': 'news.json',
-          },
-          {
-              'object_name': 'fdc43ee18cbd65487249441849f58aa13484aaef',
-              'sha256sum': '0e92de92f49abc9a521f7175106c80744196f8cefc0263bc0f4a6b4f724a7d10',
-              'size_bytes': 1310798,
-              'generation': 1664794197855470,
-              'output_file': 'search.json',
-          },
-          {
-              'object_name': '7fc9338af75b7d9d185c91ddf262a356def5623d',
-              'sha256sum': '34e92acae8aade2a186abe79ed1f379c266f04f72f1eb54bd3a912e889bc5cc0',
-              'size_bytes': 2280846,
-              'generation': 1664794200867034,
-              'output_file': 'social1.json',
-          },
-          {
-              'object_name': 'c2d7e9ce67522dad138c7feb0a6911b828bfb130',
-              'sha256sum': '95c6b148577b891310c024b2daa5d68faf644a37707ac0cb21501eefe8a399a3',
-              'size_bytes': 411708,
-              'generation': 1664794203829582,
-              'output_file': 'social2.json',
-          },
-          {
-              'object_name': '031d5599c8a21118754e30dbea141be66104f556',
-              'sha256sum': '8e7b765d72bb8e7742f5bf955f4bf64d5469f61197dad8b632304095a52322d7',
-              'size_bytes': 3203922,
-              'generation': 1664794206824773,
-              'output_file': 'sports.json',
-          },
-          {
-              'object_name': '8aac3db2a8c9e44babec81e539a3d60aeab4985c',
-              'sha256sum': '6aeb0036dfafaf5e905abdb0ffe515a3952ffe35a7c59afb0fc8b233b27c6ce4',
-              'size_bytes': 5902660,
-              'generation': 1664794209886788,
-              'output_file': 'video.json',
-          },
-      ],
-  },
-
   'src/third_party/content_analysis_sdk/src':
     Var('chromium_git') + '/external/github.com/chromium/content_analysis_sdk.git' + '@' + '9a408736204513e0e95dd2ab3c08de0d95963efc',
 
@@ -1620,22 +1461,6 @@
   'src/third_party/google_benchmark/src':
     Var('chromium_git') + '/external/github.com/google/benchmark.git' + '@' + '344117638c8ff7e239044fd0fa7085839fc03021',
 
-  # Download test data for Maps telemetry_gpu_integration_test.
-  'src/tools/perf/page_sets/maps_perf_test/dataset': {
-      'dep_type': 'gcs',
-      'condition': 'non_git_source',
-      'bucket': 'chromium-telemetry',
-      'objects': [
-          {
-              'object_name': 'e6bf26977c2fd80c18789d1f279d474096a7b0d1',
-              'sha256sum': 'f5f7fe360ad2b9c3d9dda2612f17336c0541bac15b4e4992f2c167e059a190fa',
-              'size_bytes': 3285237,
-              'generation': 1513305740113238,
-              'output_file': 'load_dataset',
-          },
-      ],
-  },
-
   'src/third_party/boringssl/src':
     Var('boringssl_git') + '/boringssl.git' + '@' +  Var('boringssl_revision'),
 
@@ -1659,7 +1484,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit.
-      'condition': 'checkout_android or checkout_linux',
+      'condition': 'checkout_android',
       'dep_type': 'cipd',
   },
 
@@ -1700,9 +1525,6 @@
       'condition': 'checkout_linux or checkout_chromeos',
   },
 
-  'src/third_party/crossbench':
-    Var('chromium_git') + '/crossbench.git' + '@' + Var('crossbench_revision'),
-
 
   'src/third_party/crubit/src': {
       'url': Var('chromium_git') + '/external/github.com/google/crubit.git' + '@' + Var('crubit_revision'),
@@ -1773,9 +1595,6 @@
   'src/third_party/freetype/src':
     Var('chromium_git') + '/chromium/src/third_party/freetype2.git' + '@' + Var('freetype_revision'),
 
-  'src/third_party/freetype-testing/src':
-    Var('chromium_git') + '/external/github.com/freetype/freetype2-testing.git' + '@' + Var('freetype_testing_revision'),
-
   'src/third_party/fxdiv/src':
     Var('chromium_git') + '/external/github.com/Maratyszcza/FXdiv.git' + '@' + '63058eff77e11aa15bf531df5dd34395ec3017c8',
 
@@ -1862,7 +1681,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit.
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -1918,7 +1737,7 @@
           },
       ],
       # Needed on Linux for use on chromium_presubmit (for checkstyle).
-      'condition': '(checkout_android or checkout_linux) and non_git_source',
+      'condition': 'checkout_android and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -1969,12 +1788,6 @@
   'src/third_party/leveldatabase/src':
     Var('chromium_git') + '/external/leveldb.git' + '@' + '068d5ee1a3ac40dabd00d211d5013af44be55bea',
 
-  'src/third_party/libFuzzer/src':
-    Var('chromium_git') + '/external/github.com/llvm/llvm-project/compiler-rt/lib/fuzzer.git' + '@' +  Var('libfuzzer_revision'),
-
-  'src/third_party/fuzztest/src':
-    Var('chromium_git') + '/external/github.com/google/fuzztest.git' + '@' + Var('fuzztest_revision'),
-
   'src/third_party/libaddressinput/src':
     Var('chromium_git') + '/external/libaddressinput.git' + '@' + 'e8712e415627f22d0b00ebee8db99547077f39bd',
 
@@ -1999,15 +1812,9 @@
   'src/third_party/securemessage/src':
     Var('chromium_git') + '/external/github.com/google/securemessage.git' + '@' + Var('securemessage_revision'),
 
-  'src/third_party/speedometer/v3.0':
-    Var('chromium_git') + '/external/github.com/WebKit/Speedometer.git' + '@' + Var('speedometer_3.0_revision'),
-
   'src/third_party/ukey2/src':
     Var('chromium_git') + '/external/github.com/google/ukey2.git' + '@' + Var('ukey2_revision'),
 
-  'src/third_party/cros-components/src':
-    Var('chromium_git') + '/external/google3/cros_components.git' + '@' + Var('cros_components_revision'),
-
   # Userspace interface to kernel DRM services.
   'src/third_party/libdrm/src': {
       'url': Var('chromium_git') + '/chromiumos/third_party/libdrm.git' + '@' + '98e1db501173303e58ef6a1def94ab7a2d84afc1',
@@ -2187,111 +1994,12 @@
     'condition': 'checkout_openxr',
   },
 
-  'src/third_party/opus/tests/resources': {
-      'dep_type': 'gcs',
-      'condition': 'non_git_source',
-      'bucket': 'chromium-webrtc-resources',
-      'objects': [
-          {
-              'object_name': '009a3ee778767c2402b1d2c920bc2449265f5a2c',
-              'sha256sum': '34de3161f242895a682d9cdcbbf4ad50246742b6db46873386104cfde8a24332',
-              'size_bytes': 26889600,
-              'generation': 1392811661954000,
-              'output_file': 'speech_mono_32_48kHz.pcm',
-          },
-      ],
-  },
-
   'src/third_party/pdfium':
     Var('pdfium_git') + '/pdfium.git' + '@' +  Var('pdfium_revision'),
 
   'src/third_party/perfetto':
     Var('android_git') + '/platform/external/perfetto.git' + '@' + '636a83a4915afdff2e5758e09a5231ff04278ede',
 
-  'src/base/tracing/test/data': {
-    'bucket': 'perfetto',
-    'objects': [
-      {
-        'object_name': 'test_data/chrome_fcp_lcp_navigations.pftrace-ae01d849fbd75a98be1b7ddd5a8873217c377b393a1d5bbd788ed3364f7fefc3',
-        'sha256sum': 'ae01d849fbd75a98be1b7ddd5a8873217c377b393a1d5bbd788ed3364f7fefc3',
-        'size_bytes': 2398645,
-        'generation': 1697714434866488,
-        'output_file': 'chrome_fcp_lcp_navigations.pftrace'
-      },
-      {
-        'object_name': 'test_data/chrome_input_with_frame_view.pftrace-a93548822e481508c728ccc5da3ad34afcd0aec02ca7a7a4dad84ff340ee5975',
-        'sha256sum': 'a93548822e481508c728ccc5da3ad34afcd0aec02ca7a7a4dad84ff340ee5975',
-        'size_bytes': 6392331,
-        'generation': 1711402389089075,
-        'output_file': 'chrome_input_with_frame_view.pftrace'
-      },
-      {
-        'object_name': 'test_data/scroll_offsets_trace_2.pftrace-2ddd9f78d91d51e39c72c520bb54fdc9dbf1333ae722e87633fc345159296289',
-        'sha256sum': '2ddd9f78d91d51e39c72c520bb54fdc9dbf1333ae722e87633fc345159296289',
-        'size_bytes': 1496388,
-        'generation': 1712592637141461,
-        'output_file': 'scroll_offsets_trace_2.pftrace'
-      },
-      {
-        'object_name': 'test_data/top_level_java_choreographer_slices-8001e73b2458e94f65a606bb558a645ba5bca553b57fe416001f6c2175675a8a',
-        'sha256sum': '8001e73b2458e94f65a606bb558a645ba5bca553b57fe416001f6c2175675a8a',
-        'size_bytes': 5323017,
-        'generation': 1671708979893186,
-        'output_file': 'top_level_java_choreographer_slices'
-      },
-      {
-        'object_name': 'test_data/chrome_page_load_all_categories_not_extended.pftrace.gz-6586e9e2bbc0c996caddb321a0374328654983733e6ffd7f4635ac07db32a493',
-        'sha256sum': '6586e9e2bbc0c996caddb321a0374328654983733e6ffd7f4635ac07db32a493',
-        'size_bytes': 1277750,
-        'generation': 1652442088902445,
-        'output_file': 'chrome_page_load_all_categories_not_extended.pftrace.gz'
-      },
-      {
-        'object_name': 'test_data/speedometer.perfetto_trace.gz-8a159b354d74a3ca0d38ce9cd071ef47de322db4261ee266bfafe04d70310529',
-        'sha256sum': '8a159b354d74a3ca0d38ce9cd071ef47de322db4261ee266bfafe04d70310529',
-        'size_bytes': 891088,
-        'generation': 1684336047660953,
-        'output_file': 'speedometer.perfetto_trace.gz'
-      },
-      {
-        'object_name': 'test_data/scroll_jank_with_pinch.pftrace-8587d2016fdb5d39b5f852704b6e3925e9e6527af01696396be767bed04d4a45',
-        'sha256sum': '8587d2016fdb5d39b5f852704b6e3925e9e6527af01696396be767bed04d4a45',
-        'size_bytes': 3914720,
-        'generation': 1717497788778335,
-        'output_file': 'scroll_jank_with_pinch.pftrace'
-      },
-      {
-        'object_name': 'test_data/cpu_powerups_1.pb-70f5511ba0cd6ce1359e3cadb4d1d9301fb6e26be85158e3384b06f41418d386',
-        'sha256sum': '70f5511ba0cd6ce1359e3cadb4d1d9301fb6e26be85158e3384b06f41418d386',
-        'size_bytes': 2033064,
-        'generation': 1669652389509708,
-        'output_file': 'cpu_powerups_1.pb'
-      },
-      {
-        'object_name': 'test_data/chrome_5672_histograms.pftrace.gz-a09bd44078ac71bcfbc901b0544750e8344d0d0f6f96e220f700a5a53fa932ee',
-        'sha256sum': 'a09bd44078ac71bcfbc901b0544750e8344d0d0f6f96e220f700a5a53fa932ee',
-        'size_bytes': 1127472,
-        'generation': 1684946598804577,
-        'output_file': 'chrome_5672_histograms.pftrace.gz'
-      },
-      {
-        'object_name': 'test_data/chrome_custom_navigation_trace.gz-ff68279e3cec94076b69259d756eed181a63eaf834d8b956a7f4ba665fabf939',
-        'sha256sum': 'ff68279e3cec94076b69259d756eed181a63eaf834d8b956a7f4ba665fabf939',
-        'size_bytes': 7572484,
-        'generation': 1666713705258900,
-        'output_file': 'chrome_custom_navigation_trace.gz'
-      },
-      {
-        'object_name': 'test_data/scroll_offsets.pftrace-62101edb5204fec8bea30124f65d4e49bda0808d7b036e95f89445aaad6d8d98',
-        'sha256sum': '62101edb5204fec8bea30124f65d4e49bda0808d7b036e95f89445aaad6d8d98',
-        'size_bytes': 769741,
-        'generation': 1693402148909129,
-        'output_file': 'scroll_offsets.pftrace'
-      }
-    ],
-    'dep_type': 'gcs'
-  },
-
   'src/third_party/perl': {
       'url': Var('chromium_git') + '/chromium/deps/perl.git' + '@' + '8ef97ff3b7332e38e61b347a2fbed425a4617151',
       'condition': 'checkout_win',
@@ -2436,20 +2144,6 @@
       'dep_type': 'cipd',
   },
 
-  'src/third_party/test_fonts/test_fonts': {
-      'dep_type': 'gcs',
-      'condition': 'non_git_source',
-      'bucket': 'chromium-fonts',
-      'objects': [
-          {
-              'object_name': 'f26f29c9d3bfae588207bbc9762de8d142e58935c62a86f67332819b15203b35',
-              'sha256sum': 'f26f29c9d3bfae588207bbc9762de8d142e58935c62a86f67332819b15203b35',
-              'size_bytes': 32750602,
-              'generation': 1717109450425063,
-          },
-      ],
-  },
-
   'src/third_party/text-fragments-polyfill/src':
     Var('chromium_git') + '/external/github.com/GoogleChromeLabs/text-fragments-polyfill.git' + '@' + 'c036420683f672d685e27415de0a5f5e85bdc23f',
 
@@ -2535,30 +2229,12 @@
       'condition': 'checkout_chromeos',
   },
 
-  'src/third_party/xdg-utils': {
-      'url': Var('chromium_git') + '/chromium/deps/xdg-utils.git' + '@' + 'cb54d9db2e535ee4ef13cc91b65a1e2741a94a44',
-      'condition': 'checkout_linux',
-  },
-
   'src/third_party/xnnpack/src':
     Var('chromium_git') + '/external/github.com/google/XNNPACK.git' + '@' + 'c2aef747585d1453d682e82ee431317a1e35c7f3',
 
   'src/tools/page_cycler/acid3':
     Var('chromium_git') + '/chromium/deps/acid3.git' + '@' + 'a926d0a32e02c4c03ae95bb798e6c780e0e184ba',
 
-  'src/third_party/libei/cipd': {
-
-    'packages': [
-      {
-        'package': 'chromium/third_party/libei/linux-amd64',
-        'version': 'PQz4zG5Q3SXoAaCYq3RK99W3wg_v0NoOu1OzTSvA_oIC',
-      },
-    ],
-
-    'condition': 'checkout_linux and non_git_source',
-    'dep_type': 'cipd',
-  },
-
   'src/third_party/zstd/src':
     Var('chromium_git') + '/external/github.com/facebook/zstd.git' + '@' + '0e2ceb2d5061f3a8357d124029ebaae16d915a3d',
 
@@ -4874,17 +4550,6 @@
     'dep_type': 'cipd',
   },
 
-  'src/third_party/screen-ai/linux': {
-      'packages': [
-          {
-              'package': 'chromium/third_party/screen-ai/linux',
-              'version': Var('screen_ai_linux'),
-          },
-      ],
-      'condition': 'checkout_linux and non_git_source',
-      'dep_type': 'cipd',
-  },
-
   'src/third_party/screen-ai/macos_amd64': {
       'packages': [
           {
@@ -5166,17 +4831,6 @@
     ],
   },
   {
-    # Ensure we remove any file from disk that is no longer needed (e.g. after
-    # hooks to native GCS deps migration).
-    'name': 'remove_stale_files',
-    'pattern': '.',
-    'action': [
-        'python3',
-        'src/tools/remove_stale_files.py',
-        'src/third_party/test_fonts/test_fonts.tar.gz', # Remove after 20240901
-    ],
-  },
-  {
     # Ensure that we don't accidentally reference any .pyc files whose
     # corresponding .py files have since been deleted.
     # We could actually try to avoid generating .pyc files, crbug.com/500078.
@@ -5972,18 +5626,6 @@
                'gs://aom-test-data',
                'src/third_party/libaom/testdata']
   },
-  {
-    'name': 'libvpx_testdata',
-    'pattern': '.',
-    'condition': 'download_libvpx_testdata',
-    'action': ['python3',
-               'src/third_party/depot_tools/gsutil.py',
-               '-q',
-               '-m',
-               'rsync',
-               'gs://downloads.webmproject.org/test_data/libvpx',
-               'src/third_party/libvpx/testdata'],
-  },
 ]
 
 # Add any corresponding DEPS files from this list to chromium.exclusions in
