--- a/DEPS
+++ b/DEPS
@@ -936,7 +936,7 @@
         'sha256sum': 'd95be7651ce879e92b18949e1228775e01352960600ed4ff389815ce8848676d',
         'size_bytes': 124379804,
         'generation': 1734710923416010,
-        'condition': 'host_os == "linux" and non_git_source',
+        'condition': 'host_os != "linux" and non_git_source',
       },
       {
         'object_name': 'Mac/rust-toolchain-8a1f8039a7ded79d3d4fe97b110016d89f2b11e2-1-llvmorg-20-init-16062-g091448e3.tar.xz',
