--- a/DEPS
+++ b/DEPS
@@ -2311,7 +2311,7 @@
           'version': Var('cpython3_version'),
         },
       ],
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -2324,7 +2324,7 @@
           'version': Var('cpython3_version'),
         },
       ],
-      'condition': 'non_git_source',
+      'condition': 'checkout_src_internal and non_git_source',
       'dep_type': 'cipd',
   },
 
@@ -4341,6 +4341,7 @@
   {
     'name': 'vpython3_common',
     'pattern': '.',
+    'condition': 'checkout_src_internal',
     'action': [ 'vpython3',
                 '-vpython-spec', 'src/vpython.toml',
                 '-vpython-tool', 'install',
