--- a/DEPS
+++ b/DEPS
@@ -764,7 +764,7 @@
   },
   'src/buildtools/linux64-format': {
     'bucket': 'chromium-clang-format',
-    'condition': 'host_os == "linux" and non_git_source',
+    'condition': 'host_os != "linux" and non_git_source',
     'dep_type': 'gcs',
     'objects': [
       {
@@ -878,7 +878,7 @@
   'src/third_party/llvm-build/Release+Asserts': {
     'dep_type': 'gcs',
     'bucket': 'chromium-browser-clang',
-    'condition': 'not llvm_force_head_revision',
+    'condition': 'llvm_force_head_revision',
     'objects': [
       {
         'object_name': 'Linux_x64/clang-android-runtime-library-llvmorg-24-init-3796-g20e97c4b-2.tar.xz',
