--- a/build/config/features.gni
+++ b/build/config/features.gni
@@ -40,6 +40,10 @@
 
   use_blink = !is_ios
 }
+
+declare_args() {
+  use_bluez = (is_linux && !is_castos && use_dbus) || is_chromeos
+}
 #
 # =============================================
 #   PLEASE DO NOT ADD MORE FLAGS TO THIS FILE
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1597,6 +1597,9 @@
     deps += [ "//third_party/fontconfig" ]
     if (use_dbus) {
       deps += [ "//dbus" ]
+      if (use_bluez) {
+        defines += [ "PLATFORM_BLUETOOTH" ]
+      }
     }
   }
 
--- a/chrome/browser/chrome_browser_main_linux.cc
+++ b/chrome/browser/chrome_browser_main_linux.cc
@@ -66,8 +66,10 @@
 
 #if !BUILDFLAG(IS_CHROMEOS)
 #if BUILDFLAG(USE_DBUS)
+#if defined(PLATFORM_BLUETOOTH)
   bluez::BluezDBusManager::Initialize(
       dbus_thread_linux::GetSharedSystemBus().get());
+#endif  // defined(PLATFORM_BLUETOOTH)
   session_end_listener_ = SessionEndListenerLinux::Create();
 #endif  // BUILDFLAG(USE_DBUS)
 #endif  // !BUILDFLAG(IS_CHROMEOS)
@@ -114,8 +116,10 @@
 #else
 #if BUILDFLAG(USE_DBUS)
   session_end_listener_.reset();
+#if defined(PLATFORM_BLUETOOTH)
   bluez::BluezDBusManager::Shutdown();
-#endif
+#endif  // defined(PLATFORM_BLUETOOTH)
+#endif  // BUILDFLAG(USE_DBUS)
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
   ChromeBrowserMainPartsPosix::PostDestroyThreads();
--- a/chrome/browser/metrics/bluetooth_metrics_provider.cc
+++ b/chrome/browser/metrics/bluetooth_metrics_provider.cc
@@ -15,11 +15,15 @@
 BluetoothMetricsProvider::~BluetoothMetricsProvider() = default;
 
 bool BluetoothMetricsProvider::ProvideHistograms() {
+#if defined(PLATFORM_BLUETOOTH)
   base::UmaHistogramEnumeration("Bluetooth.StackName",
                                 floss::features::IsFlossEnabled()
                                     ? BluetoothStackName::kFloss
                                     : BluetoothStackName::kBlueZ);
   return true;
+#else
+  return false;
+#endif
 }
 
 }  // namespace metrics
--- a/device/bluetooth/BUILD.gn
+++ b/device/bluetooth/BUILD.gn
@@ -6,10 +6,6 @@
 import("//build/config/features.gni")
 import("//device/bluetooth/emulation/buildflags.gni")
 
-declare_args() {
-  use_bluez = (is_linux && !is_castos && use_dbus) || is_chromeos
-}
-
 if (is_android) {
   import("//build/config/android/rules.gni")
   import("//third_party/jni_zero/jni_zero.gni")
