--- a/chrome/browser/ui/webui/settings/settings_ui.cc
+++ b/chrome/browser/ui/webui/settings/settings_ui.cc
@@ -24,7 +24,6 @@
 #include "chrome/browser/browser_features.h"
 #include "chrome/browser/commerce/shopping_service_factory.h"
 #include "chrome/browser/companion/core/features.h"
-#include "chrome/browser/compose/compose_enabling.h"
 #include "chrome/browser/download/bubble/download_bubble_prefs.h"
 #include "chrome/browser/history_embeddings/history_embeddings_utils.h"
 #include "chrome/browser/optimization_guide/optimization_guide_keyed_service.h"
@@ -96,7 +95,6 @@
 #include "components/commerce/core/commerce_feature_list.h"
 #include "components/commerce/core/feature_utils.h"
 #include "components/commerce/core/shopping_service.h"
-#include "components/compose/core/browser/compose_features.h"
 #include "components/content_settings/core/common/features.h"
 #include "components/favicon_base/favicon_url_parser.h"
 #include "components/optimization_guide/core/optimization_guide_features.h"
@@ -125,6 +123,11 @@
 #include "third_party/blink/public/common/features.h"
 #include "ui/base/interaction/element_identifier.h"
 
+#if BUILDFLAG(ENABLE_COMPOSE)
+#include "chrome/browser/compose/compose_enabling.h"
+#include "components/compose/core/browser/compose_features.h"
+#endif
+
 #if !BUILDFLAG(OPTIMIZE_WEBUI)
 #include "chrome/grit/settings_shared_resources.h"
 #include "chrome/grit/settings_shared_resources_map.h"
@@ -395,9 +398,11 @@
       "enableLinkedServicesSetting",
       base::FeatureList::IsEnabled(features::kLinkedServicesSetting));
 
+#if BUILDFLAG(ENABLE_COMPOSE)
   html_source->AddBoolean("enableComposeProactiveNudge",
                           base::FeatureList::IsEnabled(
                               compose::features::kEnableComposeProactiveNudge));
+#endif
 
   html_source->AddBoolean(
       "enablePageContentSetting",
@@ -584,7 +589,9 @@
       optimization_guide::features::kAiSettingsPageRefresh);
 
   if (ai_settings_refresh_enabled) {
+#if BUILDFLAG(ENABLE_COMPOSE)
     const bool compose_enabled = ComposeEnabling::IsEnabledForProfile(profile);
+#endif
     const bool tab_organization_enabled =
         TabOrganizationUtils::GetInstance()->IsEnabled(profile);
     const bool wallpaper_search_enabled =
@@ -594,7 +601,9 @@
     const bool compare_enabled = commerce::CanFetchProductSpecificationsData(
         shopping_service->GetAccountChecker());
 
+#if BUILDFLAG(ENABLE_COMPOSE)
     html_source->AddBoolean("showComposeControl", compose_enabled);
+#endif
     html_source->AddBoolean("showTabOrganizationControl",
                             tab_organization_enabled);
     html_source->AddBoolean("showWallpaperSearchControl",
@@ -602,9 +611,15 @@
     html_source->AddBoolean("showHistorySearchControl", history_search_enabled);
     html_source->AddBoolean("showCompareControl", compare_enabled);
 
+#if BUILDFLAG(ENABLE_COMPOSE)
     const bool show_ai_page = compose_enabled || tab_organization_enabled ||
                               wallpaper_search_enabled ||
                               history_search_enabled || compare_enabled;
+#else
+    const bool show_ai_page = tab_organization_enabled ||
+                              wallpaper_search_enabled ||
+                              history_search_enabled || compare_enabled;
+#endif
     // "showAdvancedFeaturesMainControl", despite the name, controls whether the
     // AI subpage is shown. We want to show the page if any of the AI features
     // are enabled.
@@ -612,7 +627,9 @@
     html_source->AddBoolean("showAdvancedFeaturesMainControl", show_ai_page);
   } else {
     optimization_guide::UserVisibleFeatureKey optimization_guide_features[4] = {
+#if BUILDFLAG(ENABLE_COMPOSE)
         optimization_guide::UserVisibleFeatureKey::kCompose,
+#endif
         optimization_guide::UserVisibleFeatureKey::kTabOrganization,
         optimization_guide::UserVisibleFeatureKey::kWallpaperSearch,
         optimization_guide::UserVisibleFeatureKey::kHistorySearch,
@@ -635,8 +652,10 @@
 
     html_source->AddBoolean("showAdvancedFeaturesMainControl",
                             optimization_guide_feature_visible[0]);
+#if BUILDFLAG(ENABLE_COMPOSE)
     html_source->AddBoolean("showComposeControl",
                             optimization_guide_feature_visible[1]);
+#endif
     html_source->AddBoolean("showTabOrganizationControl",
                             optimization_guide_feature_visible[2]);
     html_source->AddBoolean("showWallpaperSearchControl",
