--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -33,6 +33,7 @@
 #include "components/browsing_data/core/pref_names.h"
 #include "components/commerce/core/pref_names.h"
 #include "components/component_updater/pref_names.h"
+#include "components/compose/buildflags.h"
 #include "components/content_settings/core/common/pref_names.h"
 #include "components/dom_distiller/core/pref_names.h"
 #include "components/drive/drive_pref_names.h"
@@ -455,10 +456,12 @@
   (*s_allowlist)
       [::permissions::prefs::kUnusedSitePermissionsRevocationEnabled] =
           settings_api::PrefType::kBoolean;
+#if BUILDFLAG(ENABLE_COMPOSE)
   (*s_allowlist)[prefs::kEnableProactiveNudge] =
       settings_api::PrefType::kBoolean;
   (*s_allowlist)[prefs::kProactiveNudgeDisabledSitesWithTime] =
       settings_api::PrefType::kDictionary;
+#endif
 
   // Clear browsing data settings.
   (*s_allowlist)[browsing_data::prefs::kDeleteBrowsingHistory] =
