--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -4086,8 +4086,6 @@
       "new_tab_page/chrome_colors/chrome_colors_service.h",
       "new_tab_page/feature_promo_helper/new_tab_page_feature_promo_helper.cc",
       "new_tab_page/feature_promo_helper/new_tab_page_feature_promo_helper.h",
-      "new_tab_page/modules/feed/feed_handler.cc",
-      "new_tab_page/modules/feed/feed_handler.h",
       "new_tab_page/modules/file_suggestion/drive_service.cc",
       "new_tab_page/modules/file_suggestion/drive_service.h",
       "new_tab_page/modules/file_suggestion/drive_service_factory.cc",
@@ -4543,6 +4541,14 @@
       "webauthn/webauthn_switches.cc",
       "webauthn/webauthn_switches.h",
     ]
+
+    if (enable_feed_v2) {
+      sources += [
+        "new_tab_page/modules/feed/feed_handler.cc",
+        "new_tab_page/modules/feed/feed_handler.h",
+      ]
+    }
+
     deps += [
       "//chrome/app:command_ids",
       "//chrome/app/theme:chrome_unscaled_resources_grit",
@@ -4752,6 +4758,10 @@
       ]
     }
 
+    if (enable_feed_v2) {
+      deps += [ "//chrome/browser/new_tab_page/modules/feed:mojo_bindings" ]
+    }
+
     # On Windows, the hashes are embedded in //chrome:chrome_initial rather
     # than here in :chrome_dll.
     if (!is_win) {
@@ -8660,7 +8670,6 @@
       "//chrome/browser/lens/core/mojom:mojo_bindings",
       "//chrome/browser/new_tab_page/chrome_colors:generate_chrome_colors_info",
       "//chrome/browser/new_tab_page/chrome_colors:generate_colors_info",
-      "//chrome/browser/new_tab_page/modules/feed:mojo_bindings",
       "//chrome/browser/new_tab_page/modules/file_suggestion:mojo_bindings",
       "//chrome/browser/new_tab_page/modules/history_clusters:mojo_bindings",
       "//chrome/browser/new_tab_page/modules/history_clusters/cart:mojo_bindings",
--- a/chrome/browser/chrome_browser_interface_binders.cc
+++ b/chrome/browser/chrome_browser_interface_binders.cc
@@ -1312,10 +1312,12 @@
         file_suggestion::mojom::FileSuggestionHandler, NewTabPageUI>(map);
   }
 
+#if BUILDFLAG(ENABLE_FEED_V2)
   if (base::FeatureList::IsEnabled(ntp_features::kNtpFeedModule)) {
     RegisterWebUIControllerInterfaceBinder<ntp::feed::mojom::FeedHandler,
                                            NewTabPageUI>(map);
   }
+#endif
 
   if (base::FeatureList::IsEnabled(ntp_features::kNtpTabResumptionModule)) {
     RegisterWebUIControllerInterfaceBinder<
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -6834,9 +6834,6 @@
     "//components/favicon/core/test:test_support",
     "//components/feature_engagement/public",
     "//components/feed:buildflags",
-    "//components/feed/core/v2:feed_core_stubs",
-    "//components/feed/core/v2:feed_core_v2",
-    "//components/feed/core/v2:test_helpers",
     "//components/feed/mojom:mojo_bindings",
     "//components/feedback",
     "//components/feedback/redaction_tool",
@@ -7117,6 +7114,14 @@
     "//base:path_service_unittest",
   ]
 
+  if (enable_feed_v2) {
+    deps += [
+      "//components/feed/core/v2:feed_core_stubs",
+      "//components/feed/core/v2:feed_core_v2",
+      "//components/feed/core/v2:test_helpers",
+    ]
+  }
+
   if (is_chromeos) {
     sources += [ "../browser/device_reauth/chromeos/device_authenticator_chromeos_unittest.cc" ]
 
