--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -4156,8 +4156,6 @@
       "new_tab_page/chrome_colors/chrome_colors_service.h",
       "new_tab_page/feature_promo_helper/new_tab_page_feature_promo_helper.cc",
       "new_tab_page/feature_promo_helper/new_tab_page_feature_promo_helper.h",
-      "new_tab_page/modules/feed/feed_handler.cc",
-      "new_tab_page/modules/feed/feed_handler.h",
       "new_tab_page/modules/file_suggestion/drive_service.cc",
       "new_tab_page/modules/file_suggestion/drive_service.h",
       "new_tab_page/modules/file_suggestion/drive_service_factory.cc",
@@ -4663,6 +4661,14 @@
       "webauthn/webauthn_switches.cc",
       "webauthn/webauthn_switches.h",
     ]
+
+    if (enable_feed_v2) {
+      sources += [
+        "new_tab_page/modules/feed/feed_handler.cc",
+        "new_tab_page/modules/feed/feed_handler.h",
+      ]
+    }
+
     deps += [
       ":browser_themes",
       ":theme_properties",
@@ -4680,7 +4686,6 @@
       "//chrome/browser/media/router/discovery/access_code:access_code_sink_service",
       "//chrome/browser/new_tab_page/chrome_colors:generate_chrome_colors_info",
       "//chrome/browser/new_tab_page/chrome_colors:generate_colors_info",
-      "//chrome/browser/new_tab_page/modules/feed:mojo_bindings",
       "//chrome/browser/new_tab_page/modules/file_suggestion:mojo_bindings",
       "//chrome/browser/new_tab_page/modules/history_clusters:mojo_bindings",
       "//chrome/browser/new_tab_page/modules/history_clusters/cart:mojo_bindings",
@@ -4906,6 +4911,10 @@
       ]
     }
 
+    if (enable_feed_v2) {
+      deps += [ "//chrome/browser/new_tab_page/modules/feed:mojo_bindings" ]
+    }
+
     # On Windows, the hashes are embedded in //chrome:chrome_initial rather
     # than here in :chrome_dll.
     if (!is_win) {
--- a/chrome/browser/chrome_browser_interface_binders.cc
+++ b/chrome/browser/chrome_browser_interface_binders.cc
@@ -1306,10 +1306,12 @@
                                            NewTabPageUI>(map);
   }
 
+#if BUILDFLAG(ENABLE_FEED_V2)
   if (base::FeatureList::IsEnabled(ntp_features::kNtpFeedModule)) {
     RegisterWebUIControllerInterfaceBinder<ntp::feed::mojom::FeedHandler,
                                            NewTabPageUI>(map);
   }
+#endif
 
   if (base::FeatureList::IsEnabled(ntp_features::kNtpHistoryClustersModule) ||
       base::FeatureList::IsEnabled(
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -6957,9 +6957,6 @@
     "//components/favicon/core/test:test_support",
     "//components/feature_engagement/public",
     "//components/feed:buildflags",
-    "//components/feed/core/v2:feed_core_stubs",
-    "//components/feed/core/v2:feed_core_v2",
-    "//components/feed/core/v2:test_helpers",
     "//components/feed/mojom:mojo_bindings",
     "//components/feedback",
     "//components/feedback/redaction_tool",
@@ -7229,6 +7226,14 @@
     "//base:path_service_unittest",
   ]
 
+  if (enable_feed_v2) {
+    deps += [
+      "//components/feed/core/v2:feed_core_stubs",
+      "//components/feed/core/v2:feed_core_v2",
+      "//components/feed/core/v2:test_helpers",
+    ]
+  }
+
   if (is_chromeos) {
     sources += [
       "../browser/chromeos/office_web_app/office_web_app_unittest.cc",
