--- a/BUILD.gn
+++ b/BUILD.gn
@@ -804,7 +804,7 @@
       deps += [ "//native_client_sdk/src:nacl_core_sdk" ]
     }
 
-    if (((is_linux || is_chromeos) && use_ozone) || is_fuchsia) {
+    if ((is_chromeos && use_ozone) || is_fuchsia) {
       deps += [
         "//headless:headless_non_renderer",
         "//headless:headless_tests",
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1147,7 +1147,6 @@
       "//components/memory_system",
       "//components/policy:generated",
       "//content/public/app",
-      "//headless:headless_shell_lib",
       "//third_party/cld_3/src/src:cld_3",
     ]
 
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1847,7 +1847,6 @@
     # should be added to browser_generated_files.
     ":browser_public_dependencies",
     "//chrome/browser/google",
-    "//chrome/browser/headless",
     "//chrome/browser/profiles",
     "//chrome/browser/ui/tabs:tab_enums",
   ]
@@ -2150,7 +2149,6 @@
     "//components/gcm_driver",
     "//components/guest_view/buildflags",
     "//components/handoff",
-    "//components/headless/select_file_dialog",
     "//components/heap_profiling/in_process",
     "//components/heap_profiling/in_process:mojom",
     "//components/heavy_ad_intervention",
@@ -3755,8 +3753,6 @@
       "first_run/first_run_internal.h",
       "font_family_cache.cc",
       "font_family_cache.h",
-      "headless/chrome_browser_main_extra_parts_headless.cc",
-      "headless/chrome_browser_main_extra_parts_headless.h",
       "hid/chrome_hid_delegate.cc",
       "hid/chrome_hid_delegate.h",
       "hid/hid_chooser_context.cc",
@@ -4389,8 +4385,6 @@
       "//components/feedback",
       "//components/feedback/content:factory",
       "//components/feedback/redaction_tool",
-      "//components/headless/clipboard",
-      "//components/headless/policy",
       "//components/image_fetcher/core",
       "//components/keep_alive_registry",
       "//components/live_caption",
@@ -7066,7 +7060,7 @@
     deps += [ "//components/power_metrics" ]
   }
 
-  if (is_win || is_mac || is_linux) {
+  if (is_win || is_mac) {
     deps += [ "//chrome/browser/headless:metrics" ]
   }
 
@@ -7114,7 +7108,6 @@
     deps += [
       "//chrome/common/printing",
       "//components/printing/browser",
-      "//components/printing/browser/headless",
       "//components/printing/browser/print_to_pdf",
       "//components/services/print_compositor/public/cpp",
       "//components/services/print_compositor/public/mojom",
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -312,7 +312,6 @@
     "//chrome/browser:browser_public_dependencies",
     "//chrome/browser:primitives",
     "//chrome/browser/feedback",
-    "//chrome/browser/headless",
     "//chrome/browser/media/webrtc",
     "//chrome/browser/search",
     "//chrome/browser/task_manager",
@@ -1705,7 +1704,6 @@
       "//chrome/browser/download",
       "//chrome/browser/enterprise/watermark:watermark_view_lib",
       "//chrome/browser/feedback:feedback_enum",
-      "//chrome/browser/headless:command_processor",
       "//chrome/browser/hid",
       "//chrome/browser/image_editor",
       "//chrome/browser/image_editor:image_editor_component_util",
@@ -1817,7 +1815,6 @@
       "//components/endpoint_fetcher:endpoint_fetcher",
       "//components/enterprise/common:files_scan_data",
       "//components/feedback/proto",
-      "//components/headless/policy",
       "//components/history_embeddings",
       "//components/keep_alive_registry",
       "//components/manta",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -2192,7 +2192,6 @@
       "//chrome/browser/file_system_access",
       "//chrome/browser/file_system_access:browser_tests",
       "//chrome/browser/first_party_sets",
-      "//chrome/browser/headless:browser_tests",
       "//chrome/browser/icon_transcoder:browser_tests",
       "//chrome/browser/image_decoder:browser_tests",
       "//chrome/browser/image_fetcher",
@@ -3842,7 +3841,6 @@
             "../browser/devtools/protocol/devtools_printtopdf_browsertest.cc",
           ]
           deps += [
-            "//components/headless/test",
             "//components/printing/browser",
             "//pdf",
           ]
--- a/chrome/app/chrome_main.cc
+++ b/chrome/app/chrome_main.cc
@@ -16,7 +16,6 @@
 #include "build/build_config.h"
 #include "chrome/app/chrome_main_delegate.h"
 #include "chrome/app/startup_timestamps.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/common/buildflags.h"
 #include "chrome/common/chrome_result_codes.h"
 #include "chrome/common/chrome_switches.h"
@@ -56,8 +55,7 @@
 // Chrome is started with --headless=old switch. This should be deleted
 // sometime after old headless code is removed from Chrome.
 // See https://crbug.com/373672160.
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
 #define ENABLE_OLD_HEADLESS_INFO
 #endif
 
@@ -195,6 +193,7 @@
   // function, which are now redundant.
   base::PoissonAllocationSampler::Init();
 
+#ifdef ENABLE_OLD_HEADLESS_INFO
   // Chrome-specific process modes.
   std::unique_ptr<headless::HeadlessModeHandle> headless_mode_handle;
   if (headless::IsHeadlessMode()) {
@@ -204,13 +203,12 @@
     }
     headless_mode_handle = headless::InitHeadlessMode();
   } else {
-#ifdef ENABLE_OLD_HEADLESS_INFO
     if (headless::IsOldHeadlessMode()) {
       ShowOldHeadlessInfoMaybe(command_line);
       return EXIT_FAILURE;
     }
-#endif  // ENABLE_OLD_HEADLESS_INFO
   }
+#endif  // ENABLE_OLD_HEADLESS_INFO
 
 #if BUILDFLAG(IS_MAC)
   // Gracefully exit if a helper app was launched in an unexpected situation.
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -47,7 +47,6 @@
 #include "chrome/browser/chrome_content_browser_client.h"
 #include "chrome/browser/chrome_resource_bundle_helper.h"
 #include "chrome/browser/defaults.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/lifetime/browser_shutdown.h"
 #include "chrome/browser/metrics/chrome_feature_list_creator.h"
 #include "chrome/browser/startup_data.h"
@@ -1509,7 +1508,7 @@
   // before field trial initialization, so the values of
   // `kPersistentHistogramsFeature` and `kPersistentHistogramsStorage` will
   // not be used. Persist histograms to a memory-mapped file.
-  if (process_type.empty() && !headless::IsHeadlessMode()) {
+  if (process_type.empty()) {
     base::FilePath metrics_dir;
     if (base::PathService::Get(chrome::DIR_USER_DATA, &metrics_dir)) {
       InstantiatePersistentHistograms(
--- a/chrome/browser/chrome_browser_main.cc
+++ b/chrome/browser/chrome_browser_main.cc
@@ -181,12 +181,14 @@
 #endif
 
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
 #include "chrome/browser/headless/headless_mode_metrics.h"  // nogncheck
 #include "chrome/browser/headless/headless_mode_util.h"     // nogncheck
+#include "components/headless/select_file_dialog/headless_select_file_dialog.h"
+#endif  // BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
 #include "chrome/browser/metrics/desktop_session_duration/desktop_session_duration_tracker.h"
 #include "chrome/browser/metrics/desktop_session_duration/touch_mode_stats_tracker.h"
 #include "chrome/browser/profiles/profile_activity_metrics_recorder.h"
-#include "components/headless/select_file_dialog/headless_select_file_dialog.h"
 #include "ui/base/pointer/touch_ui_controller.h"
 #include "ui/gfx/switches.h"
 #endif
@@ -1311,7 +1313,7 @@
       *UrlLanguageHistogramFactory::GetForBrowserContext(profile));
 #endif
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
   if (headless::IsHeadlessMode()) {
     headless::ReportHeadlessActionMetrics();
   }
@@ -1420,7 +1422,7 @@
   // In headless mode provide alternate SelectFileDialog factory overriding
   // any platform specific SelectFileDialog implementation that may have been
   // set.
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
   if (headless::IsHeadlessMode()) {
     headless::HeadlessSelectFileDialogFactory::SetUp();
   }
@@ -1960,7 +1962,7 @@
 
   // Drop the request if headless mode is in effect or the request is from
   // a headless Chrome process.
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
   if (headless::IsHeadlessMode() ||
       command_line.HasSwitch(switches::kHeadless)) {
     return false;
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -86,7 +86,6 @@
 #include "chrome/browser/first_party_sets/first_party_sets_navigation_throttle.h"
 #include "chrome/browser/font_family_cache.h"
 #include "chrome/browser/gpu/chrome_browser_main_extra_parts_gpu.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/hid/chrome_hid_delegate.h"
 #include "chrome/browser/history/history_service_factory.h"
 #include "chrome/browser/interstitials/enterprise_util.h"
@@ -586,7 +585,6 @@
 #include "chrome/browser/devtools/chrome_devtools_manager_delegate.h"
 #include "chrome/browser/devtools/devtools_window.h"
 #include "chrome/browser/direct_sockets/chrome_direct_sockets_delegate.h"
-#include "chrome/browser/headless/chrome_browser_main_extra_parts_headless.h"
 #include "chrome/browser/media/unified_autoplay_config.h"
 #include "chrome/browser/metrics/usage_scenario/chrome_responsiveness_calculator_delegate.h"
 #include "chrome/browser/new_tab_page/new_tab_page_util.h"
@@ -1849,7 +1847,7 @@
           enterprise_util::ChromeBrowserMainExtraPartsEnterprise>());
 #endif
 
-#if !BUILDFLAG(IS_ANDROID)
+#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_LINUX)
   main_parts->AddParts(
       std::make_unique<headless::ChromeBrowserMainExtraPartsHeadless>());
 #endif
@@ -2091,7 +2089,7 @@
   static bool is_devtools_user = true;
 #else
   static bool is_devtools_user =
-      DetermineIfDevToolsUserForProcessPerSite() || headless::IsHeadlessMode();
+      DetermineIfDevToolsUserForProcessPerSite();
 #endif
 
   if (is_devtools_user && base::FeatureList::IsEnabled(
--- a/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
+++ b/chrome/browser/chrome_content_browser_client_receiver_bindings.cc
@@ -14,7 +14,6 @@
 #include "chrome/browser/chrome_content_browser_client.h"
 #include "chrome/browser/chrome_content_browser_client_parts.h"
 #include "chrome/browser/content_settings/content_settings_manager_delegate.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/net/net_error_tab_helper.h"
 #include "chrome/browser/net_benchmarking.h"
 #include "chrome/browser/password_manager/chrome_password_manager_client.h"
@@ -112,7 +111,6 @@
 
 #if BUILDFLAG(ENABLE_PRINTING)
 #include "chrome/browser/printing/print_view_manager_basic.h"
-#include "components/printing/browser/headless/headless_print_manager.h"
 #if BUILDFLAG(ENABLE_PRINT_PREVIEW)
 #include "chrome/browser/printing/print_view_manager.h"
 #endif  // BUILDFLAG(ENABLE_PRINT_PREVIEW)
@@ -555,10 +553,6 @@
           [](content::RenderFrameHost* render_frame_host,
              mojo::PendingAssociatedReceiver<printing::mojom::PrintManagerHost>
                  receiver) {
-            if (headless::IsHeadlessMode()) {
-              headless::HeadlessPrintManager::BindPrintManagerHost(
-                  std::move(receiver), render_frame_host);
-            } else {
 #if BUILDFLAG(IS_CHROMEOS)
               if (base::FeatureList::IsEnabled(
                       ::features::kPrintPreviewCrosPrimary)) {
@@ -580,7 +574,6 @@
               printing::PrintViewManagerBasic::BindPrintManagerHost(
                   std::move(receiver), render_frame_host);
 #endif  // BUILDFLAG(ENABLE_PRINT_PREVIEW)
-            }
           },
           &render_frame_host));
 #endif  // BUILDFLAG(ENABLE_PRINTING)
--- a/chrome/browser/first_run/first_run.cc
+++ b/chrome/browser/first_run/first_run.cc
@@ -26,7 +26,6 @@
 #include "chrome/browser/extensions/updater/extension_updater.h"
 #include "chrome/browser/first_run/first_run_internal.h"
 #include "chrome/browser/google/google_brand.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/importer/external_process_importer_host.h"
 #include "chrome/browser/importer/importer_list.h"
 #include "chrome/browser/importer/importer_progress_observer.h"
@@ -123,8 +122,6 @@
   ExternalProcessImporterHost* importer_host =
       new ExternalProcessImporterHost;
   // Don't show the warning dialog if import fails.
-  importer_host->set_headless();
-
   ImportEndedObserver observer;
   importer_host->set_observer(&observer);
   importer_host->StartImportSettings(source_profile,
@@ -396,10 +393,7 @@
   DCHECK(!user_data_dir.empty());
 
   if (initial_prefs.get()) {
-    // Don't show EULA when running in headless mode since this would
-    // effectively block the UI because there is no one to accept it.
-    if (!headless::IsHeadlessMode() &&
-        !internal::ShowPostInstallEULAIfNeeded(initial_prefs.get())) {
+    if (!internal::ShowPostInstallEULAIfNeeded(initial_prefs.get())) {
       return EULA_EXIT_NOW;
     }
 
--- a/chrome/browser/policy/configuration_policy_handler_list_factory.cc
+++ b/chrome/browser/policy/configuration_policy_handler_list_factory.cc
@@ -160,7 +160,6 @@
 #include "chrome/browser/policy/managed_account_policy_handler.h"
 #include "chrome/browser/web_applications/policy/web_app_settings_policy_handler.h"
 #include "components/enterprise/connectors/core/connectors_prefs.h"
-#include "components/headless/policy/headless_mode_policy_handler.h"
 #include "components/lens/lens_overlay_permission_utils.h"
 #include "components/media_router/common/pref_names.h"
 #endif  // BUILDFLAG(IS_ANDROID)
@@ -2552,7 +2551,6 @@
                                             prefs::kPromotionsEnabled,
                                             base::Value::Type::BOOLEAN)));
 
-  handlers->AddHandler(std::make_unique<headless::HeadlessModePolicyHandler>());
   handlers->AddHandler(
       std::make_unique<DownloadAutoOpenPolicyHandler>(chrome_schema));
 
--- a/chrome/browser/prefs/browser_prefs.cc
+++ b/chrome/browser/prefs/browser_prefs.cc
@@ -310,7 +310,6 @@
 #include "chrome/browser/ui/webui/side_panel/read_anything/read_anything_prefs.h"
 #include "chrome/browser/ui/webui/tab_search/tab_search_prefs.h"
 #include "chrome/browser/upgrade_detector/upgrade_detector.h"
-#include "components/headless/policy/headless_mode_prefs.h"
 #include "components/lens/lens_overlay_permission_utils.h"
 #include "components/live_caption/live_caption_controller.h"
 #include "components/live_caption/live_translate_controller.h"
@@ -1657,7 +1656,6 @@
   registry->RegisterBooleanPref(lens::kLensCameraAssistedSearchEnabled, true);
 #else  // BUILDFLAG(IS_ANDROID)
   gcm::RegisterPrefs(registry);
-  headless::RegisterPrefs(registry);
   IntranetRedirectDetector::RegisterPrefs(registry);
   media_router::RegisterLocalStatePrefs(registry);
   metrics::TabStatsTracker::RegisterPrefs(registry);
--- a/chrome/browser/printing/printing_init.cc
+++ b/chrome/browser/printing/printing_init.cc
@@ -4,9 +4,7 @@
 
 #include "chrome/browser/printing/printing_init.h"
 
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "components/embedder_support/user_agent_utils.h"
-#include "components/printing/browser/headless/headless_print_manager.h"
 #include "components/printing/browser/print_manager_utils.h"
 #include "content/public/browser/web_contents.h"
 #include "printing/buildflags/buildflags.h"
@@ -35,19 +33,12 @@
 #endif
 
 void InitializePrintingForWebContents(content::WebContents* web_contents) {
-  // Headless mode uses a minimalistic Print Manager implementation that
-  // shortcuts most of the callbacks providing only print to PDF
-  // functionality.
-  if (headless::IsHeadlessMode()) {
-    headless::HeadlessPrintManager::CreateForWebContents(web_contents);
-  } else {
 #if BUILDFLAG(ENABLE_PRINT_PREVIEW)
-    PrintViewManager::CreateForWebContents(web_contents);
-    PdfNupConverterClient::CreateForWebContents(web_contents);
+  PrintViewManager::CreateForWebContents(web_contents);
+  PdfNupConverterClient::CreateForWebContents(web_contents);
 #else
-    PrintViewManagerBasic::CreateForWebContents(web_contents);
+  PrintViewManagerBasic::CreateForWebContents(web_contents);
 #endif  // BUILDFLAG(ENABLE_PRINT_PREVIEW)
-  }
   CreateCompositeClientIfNeeded(web_contents, embedder_support::GetUserAgent());
 }
 
--- a/chrome/browser/ui/startup/startup_browser_creator.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator.cc
@@ -135,9 +135,11 @@
 #endif  // BUILDFLAG(IS_WIN)
 
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
 #include "chrome/browser/headless/headless_mode_util.h"
-#include "chrome/browser/ui/startup/web_app_info_recorder_utils.h"
 #include "components/headless/policy/headless_mode_policy.h"
+#endif  // if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#include "chrome/browser/ui/startup/web_app_info_recorder_utils.h"
 #endif
 
 #if !BUILDFLAG(IS_CHROMEOS)
@@ -978,7 +980,7 @@
   TRACE_EVENT0("startup", "StartupBrowserCreator::ProcessCmdLineImpl");
   ComputeAndRecordLaunchMode(command_line);
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
   if (headless::IsHeadlessMode() &&
       headless::HeadlessModePolicy::IsHeadlessModeDisabled(
           g_browser_process->local_state())) {
--- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
@@ -22,7 +22,6 @@
 #include "chrome/browser/custom_handlers/protocol_handler_registry_factory.h"
 #include "chrome/browser/defaults.h"
 #include "chrome/browser/first_run/first_run.h"
-#include "chrome/browser/headless/headless_command_processor.h"
 #include "chrome/browser/prefs/session_startup_pref.h"
 #include "chrome/browser/privacy_sandbox/privacy_sandbox_service.h"
 #include "chrome/browser/privacy_sandbox/privacy_sandbox_service_factory.h"
@@ -222,7 +221,7 @@
   CHECK(profile_);
 
   bool first_tab = true;
-  bool process_headless_commands = headless::ShouldProcessHeadlessCommands();
+  bool process_headless_commands = false;
   custom_handlers::ProtocolHandlerRegistry* registry =
       ProtocolHandlerRegistryFactory::GetForBrowserContext(profile_);
   for (auto& tab : tabs) {
@@ -249,34 +248,6 @@
     }
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
 
-    // Headless mode is restricted to only one url in the command line, so
-    // just grab the first one assuming it's the target.
-    if (first_tab && process_headless_commands) {
-      std::unique_ptr<ScopedProfileKeepAlive> profile_keepalive;
-      if (!profile_->IsOffTheRecord()) {
-        profile_keepalive = std::make_unique<ScopedProfileKeepAlive>(
-            profile_, ProfileKeepAliveOrigin::kHeadlessCommand);
-      }
-      headless::ProcessHeadlessCommands(
-          profile_, tab.url,
-          base::BindOnce(
-              [](base::WeakPtr<Browser> browser,
-                 std::unique_ptr<ScopedProfileKeepAlive> profile_keepalive,
-                 headless::HeadlessCommandHandler::Result result) {
-                if (browser && browser->window()) {
-#if BUILDFLAG(IS_MAC)
-                  // On Macs Chrome keeps running after the last browser
-                  // window is closed which is not expected for headless
-                  // command execution, so explicitly allow application
-                  // to terminate after the browser window is closed.
-                  app_controller_mac::AllowApplicationToTerminate();
-#endif
-                  browser->window()->Close();
-                }
-              },
-              browser->AsWeakPtr(), std::move(profile_keepalive)));
-      continue;
-    }
     // Active tab overwrites apply only to one tab per launch, and can only
     // happen if there is already a tab open to replace
     if (first_tab && browser->tab_strip_model()->count() &&
--- a/chrome/browser/ui/startup/startup_tab_provider.cc
+++ b/chrome/browser/ui/startup/startup_tab_provider.cc
@@ -62,7 +62,7 @@
 #include "extensions/browser/extension_registry.h"
 #endif  // !BUILDFLAG(IS_ANDROID)
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
 #include "chrome/browser/headless/headless_mode_util.h"
 #endif
 
@@ -101,7 +101,7 @@
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
   bool url_scheme_is_chrome = false;
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
   // In Headless mode, allow any URL pattern that matches chrome:// scheme if
   // the user explicitly allowed it.
   if (headless::IsHeadlessMode() && url.SchemeIs(content::kChromeUIScheme)) {
--- a/chrome/browser/ui/views/frame/browser_frame.cc
+++ b/chrome/browser/ui/views/frame/browser_frame.cc
@@ -14,7 +14,6 @@
 #include "base/i18n/rtl.h"
 #include "build/build_config.h"
 #include "chrome/browser/app_mode/app_mode_utils.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/themes/custom_theme_supplier.h"
 #include "chrome/browser/themes/theme_service.h"
@@ -132,7 +131,7 @@
       views::Widget::InitParams::NATIVE_WIDGET_OWNS_WIDGET);
   params.name = "BrowserFrame";
   params.delegate = browser_view_;
-  params.headless_mode = headless::IsHeadlessMode();
+  params.headless_mode = false;
 
   Browser* browser = browser_view_->browser();
   if (browser->is_type_picture_in_picture()) {
--- a/chrome/browser/ui/views/policy/enterprise_startup_dialog_view.cc
+++ b/chrome/browser/ui/views/policy/enterprise_startup_dialog_view.cc
@@ -14,7 +14,6 @@
 #include "build/branding_buildflags.h"
 #include "build/build_config.h"
 #include "chrome/app/vector_icons/vector_icons.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/ui/views/chrome_layout_provider.h"
 #include "chrome/grit/branded_strings.h"
 #include "chrome/grit/theme_resources.h"
@@ -98,55 +97,6 @@
 END_METADATA
 #endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING)
 
-// Alternate implementation of the EnterpriseStartupDialog which is used when
-// the headless mode is in effect. It does not display anything and when error
-// is set immediately calls back with not accepted condition.
-class HeadlessEnterpriseStartupDialogImpl : public EnterpriseStartupDialog {
- public:
-  explicit HeadlessEnterpriseStartupDialogImpl(DialogResultCallback callback)
-      : callback_(std::move(callback)) {}
-
-  HeadlessEnterpriseStartupDialogImpl(
-      const HeadlessEnterpriseStartupDialogImpl&) = delete;
-  HeadlessEnterpriseStartupDialogImpl& operator=(
-      const HeadlessEnterpriseStartupDialogImpl&) = delete;
-
-  ~HeadlessEnterpriseStartupDialogImpl() override {
-    if (callback_) {
-      // ChromeBrowserCloudManagementRegisterWatcher dismisses the dialog
-      // without displaying an error messgae (in which case we would not
-      // have the outstanding callback) in case of successful enrollment,
-      // so allow it to show the browser window using the callback.
-      std::move(callback_).Run(/*was_accepted=*/false,
-                               /*can_show_browser_window_=*/true);
-    }
-  }
-
-  // Override EnterpriseStartupDialog
-  void DisplayLaunchingInformationWithThrobber(
-      const std::u16string& information) override {}
-
-  void DisplayErrorMessage(
-      const std::u16string& error_message,
-      const std::optional<std::u16string>& accept_button) override {
-    if (callback_) {
-      // In headless mode the dialog is invisible, therefore there is
-      // no one to accept or dismiss it. So just dismiss the dialog
-      // right away without accepting the prompt and not allowing
-      // browser to show its window.
-      base::SingleThreadTaskRunner::GetCurrentDefault()->PostTask(
-          FROM_HERE,
-          base::BindOnce(std::move(callback_), /*was_accepted=*/false,
-                         /*can_show_browser_window_=*/false));
-    }
-  }
-
-  bool IsShowing() override { return true; }
-
- private:
-  DialogResultCallback callback_;
-};
-
 }  // namespace
 
 EnterpriseStartupDialogView::EnterpriseStartupDialogView(
@@ -339,13 +289,6 @@
 // static
 std::unique_ptr<EnterpriseStartupDialog>
 EnterpriseStartupDialog::CreateAndShowDialog(DialogResultCallback callback) {
-  // If running in headless mode use an alternate version of the enterprise
-  // startup dialog.
-  if (headless::IsHeadlessMode()) {
-    return std::make_unique<HeadlessEnterpriseStartupDialogImpl>(
-        std::move(callback));
-  }
-
   return std::make_unique<EnterpriseStartupDialogImpl>(std::move(callback));
 }
 
--- a/chrome/browser/user_education/user_education_service_factory.cc
+++ b/chrome/browser/user_education/user_education_service_factory.cc
@@ -9,7 +9,6 @@
 
 #include "build/branding_buildflags.h"
 #include "build/build_config.h"
-#include "chrome/browser/headless/headless_mode_util.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/profiles/profiles_state.h"
 #include "chrome/browser/user_education/browser_user_education_storage_service.h"
@@ -124,9 +123,6 @@
     return false;
   }
 #endif
-  if (headless::IsHeadlessMode()) {
-    return false;
-  }
   return true;
 #endif  // BUILDFLAG(CHROME_FOR_TESTING)
 }
--- a/components/BUILD.gn
+++ b/components/BUILD.gn
@@ -117,7 +117,6 @@
       "test/data/explore_sites/",
       "test/data/feed/",
       "test/data/fenced_frames/",
-      "test/data/headless/",
       "test/data/history/",
       "test/data/html_viewer/",
       "test/data/json_schema/",
@@ -590,7 +589,6 @@
       "//components/fingerprinting_protection_filter/common:unit_tests",
       "//components/fingerprinting_protection_filter/interventions/browser:unit_tests",
       "//components/fingerprinting_protection_filter/renderer:unit_tests",
-      "//components/headless/test:unit_tests",
       "//components/ip_protection:unit_tests",
       "//components/media_router/browser:unit_tests",
       "//components/media_router/common:unit_tests",
