--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1678,7 +1678,7 @@
       if (is_linux && enable_swiftshader_vulkan) {
         deps += [ ":swiftshader_vk_symbols" ]
       }
-      if (!is_chromeos && angle_shared_libvulkan) {
+      if (!is_chromeos && angle_shared_libvulkan && angle_enable_vulkan) {
         deps += [ ":angle_libvulkan_symbols" ]
       }
       if (build_with_internal_optimization_guide) {
@@ -1760,7 +1760,7 @@
         deps = [ "//third_party/angle:libGLESv2" ]
       }
     }
-    if (!is_chromeos && angle_shared_libvulkan) {
+    if (!is_chromeos && angle_shared_libvulkan && angle_enable_vulkan) {
       extract_symbols("angle_libvulkan_symbols") {
         binary = "$root_out_dir/libvulkan.so.1"
 
--- a/components/exo/wayland/BUILD.gn
+++ b/components/exo/wayland/BUILD.gn
@@ -377,12 +377,17 @@
 
   deps = [
     "//base",
-    "//gpu/vulkan",
-    "//gpu/vulkan/init",
     "//ui/gl",
     "//ui/gl/init",
   ]
 
+  if (enable_vulkan) {
+    deps += [
+      "//gpu/vulkan",
+      "//gpu/vulkan/init",
+    ]
+  }
+
   public_deps = [
     "//components/exo/wayland/protocol:aura_output_management_protocol",
     "//components/exo/wayland/protocol:aura_shell_protocol",
--- a/ui/gl/gl_switches.cc
+++ b/ui/gl/gl_switches.cc
@@ -19,6 +19,7 @@
 #if BUILDFLAG(ENABLE_VULKAN) && \
     (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID))
 #include <vulkan/vulkan_core.h>
+#include "gpu/vulkan/buildflags.h"
 #include "third_party/angle/src/gpu_info_util/SystemInfo.h"  // nogncheck
 #endif  // BUILDFLAG(ENABLE_VULKAN) && (BUILDFLAG(IS_LINUX) ||
         // BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID))
@@ -306,6 +307,7 @@
 #endif
 
 bool IsDefaultANGLEVulkan() {
+#if BUILDFLAG(ENABLE_VULKAN)
   // Force on if DefaultANGLEVulkan feature is enabled from command line.
   base::FeatureList* feature_list = base::FeatureList::GetInstance();
   if (feature_list && feature_list->IsFeatureOverriddenFromCommandLine(
@@ -442,6 +444,9 @@
 
   return base::FeatureList::IsEnabled(kDefaultANGLEVulkan);
 #endif  // !defined(MEMORY_SANITIZER)
+#else   // !BUILDFLAG(ENABLE_VULKAN)
+  return false;
+#endif  // BUILDFLAG(ENABLE_VULKAN)
 }
 
 // Use waitable swap chain on Windows to reduce display latency.
