--- a/content/utility/on_device_model/on_device_model_sandbox_init.cc
+++ b/content/utility/on_device_model/on_device_model_sandbox_init.cc
@@ -26,7 +26,7 @@
 #include "sandbox/policy/linux/sandbox_linux.h"
 #endif
 
-#if !BUILDFLAG(IS_FUCHSIA) && \
+#if !BUILDFLAG(IS_FUCHSIA) && BUILDFLAG(USE_DAWN) && \
     !(BUILDFLAG(IS_LINUX) && BUILDFLAG(ENABLE_CAST_RECEIVER))
 #include "base/feature_list.h"
 #include "third_party/dawn/include/dawn/dawn_proc.h"          // nogncheck
@@ -72,7 +72,8 @@
 // adapter. This makes sure any relevant drivers or other libs are loaded before
 // enabling the sandbox.
 BASE_FEATURE(kOnDeviceModelWarmDrivers,
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(USE_DAWN) && \
+    (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN))
              base::FEATURE_ENABLED_BY_DEFAULT
 #else
              base::FEATURE_DISABLED_BY_DEFAULT
@@ -121,7 +122,7 @@
     // Warm any relevant drivers before attempting to bring up the sandbox. For
     // good measure we initialize a device instance for any adapter with an
     // appropriate backend on top of any integrated or discrete GPU.
-#if !BUILDFLAG(IS_FUCHSIA) && \
+#if !BUILDFLAG(IS_FUCHSIA) && BUILDFLAG(USE_DAWN) && \
     !(BUILDFLAG(IS_LINUX) && BUILDFLAG(ENABLE_CAST_RECEIVER))
     dawnProcSetProcs(&dawn::native::GetProcs());
     auto instance = std::make_unique<dawn::native::Instance>();
--- a/gpu/BUILD.gn
+++ b/gpu/BUILD.gn
@@ -109,7 +109,6 @@
     "//gpu/command_buffer/service",
     "//gpu/ipc:gpu_thread_holder",
     "//third_party/dawn/include/dawn:cpp_headers",
-    "//third_party/dawn/src/dawn:proc",
   ]
   deps = [
     "//base/test:test_support",
@@ -135,6 +134,10 @@
   }
   deps += [ "//third_party/angle:translator" ]
 
+  if (use_dawn) {
+    public_deps += [ "//third_party/dawn/src/dawn:proc" ]
+  }
+
   # TODO(crbug.com/40031409): Fix code that adds exit-time destructors and
   # enable the diagnostic by removing this line.
   configs += [ "//build/config/compiler:no_exit_time_destructors" ]
--- a/gpu/command_buffer/service/BUILD.gn
+++ b/gpu/command_buffer/service/BUILD.gn
@@ -328,7 +328,10 @@
   ]
 
   if (use_dawn) {
-    deps += [ "//gpu/webgpu:common" ]
+    deps += [
+      "//gpu/webgpu:common",
+      "//third_party/dawn/src/dawn:proc",
+    ]
     sources += [
       "dawn_caching_interface.cc",
       "dawn_caching_interface.h",
@@ -396,7 +399,6 @@
     "//third_party/angle:angle_version_info",
     "//third_party/angle:translator",
     "//third_party/dawn/include/dawn:cpp_headers",
-    "//third_party/dawn/src/dawn:proc",
     "//third_party/libyuv",
     "//third_party/protobuf:protobuf_lite",
     "//third_party/re2",
--- a/services/on_device_model/BUILD.gn
+++ b/services/on_device_model/BUILD.gn
@@ -5,6 +5,7 @@
 import("//build/buildflag_header.gni")
 import("//services/on_device_model/on_device_model.gni")
 import("//third_party/dawn/scripts/dawn_features.gni")
+import("//ui/gl/features.gni")
 
 component("backend_interfaces") {
   sources = [
@@ -65,7 +66,8 @@
   if (is_android) {
     deps += [ "//services/on_device_model/android" ]
   }
-  if (!is_fuchsia && !(is_linux && enable_cast_receiver)) {
+  if (!is_fuchsia && use_dawn &&
+      !(is_linux && enable_cast_receiver)) {
     deps += [
       "//third_party/dawn/include/dawn:cpp_headers",
       "//third_party/dawn/src/dawn:proc",
--- a/services/on_device_model/ml/BUILD.gn
+++ b/services/on_device_model/ml/BUILD.gn
@@ -5,6 +5,7 @@
 import("//build/config/features.gni")
 import("//components/optimization_guide/features.gni")
 import("//services/on_device_model/on_device_model.gni")
+import("//ui/gl/features.gni")
 
 source_set("api") {
   public = [
@@ -74,10 +75,15 @@
       "//services/on_device_model/safety:safety_util",
       "//third_party/abseil-cpp:absl",
       "//third_party/angle:includes",
-      "//third_party/dawn/src/dawn:proc",
-      "//third_party/dawn/src/dawn/native",
     ]
 
+    if (use_dawn) {
+      deps += [
+        "//third_party/dawn/src/dawn:proc",
+        "//third_party/dawn/src/dawn/native",
+      ]
+    }
+
     if (enable_constraints) {
       defines += [ "ENABLE_ON_DEVICE_CONSTRAINTS" ]
       deps += [ "//third_party/rust/llguidance/v1:lib" ]
