--- a/services/on_device_model/BUILD.gn
+++ b/services/on_device_model/BUILD.gn
@@ -5,6 +5,7 @@
 import("//build/buildflag_header.gni")
 import("//services/on_device_model/on_device_model.gni")
 import("//third_party/dawn/scripts/dawn_features.gni")
+import("//ui/gl/features.gni")
 
 component("on_device_model_service") {
   sources = [
@@ -44,7 +45,7 @@
   if (is_linux || is_chromeos) {
     deps += [ "//gpu/config" ]
   }
-  if (!is_fuchsia) {
+  if (!is_fuchsia && use_dawn) {
     deps += [
       "//third_party/dawn/include/dawn:cpp_headers",
       "//third_party/dawn/src/dawn:proc",
--- a/services/on_device_model/ml/BUILD.gn
+++ b/services/on_device_model/ml/BUILD.gn
@@ -4,6 +4,7 @@
 
 import("//build/config/features.gni")
 import("//services/on_device_model/on_device_model.gni")
+import("//ui/gl/features.gni")
 
 source_set("api") {
   public = [
@@ -66,9 +67,13 @@
       "//components/translate/core/language_detection",
       "//third_party/abseil-cpp:absl",
       "//third_party/angle:includes",
-      "//third_party/dawn/src/dawn:proc",
-      "//third_party/dawn/src/dawn/native",
     ]
+    if (use_dawn) {
+      deps += [
+        "//third_party/dawn/src/dawn:proc",
+        "//third_party/dawn/src/dawn/native",
+      ]
+    }
     if (enable_constraints) {
       defines += [ "ENABLE_ON_DEVICE_CONSTRAINTS" ]
       deps += [ "//third_party/rust/llguidance/v0_7:lib" ]
--- a/services/on_device_model/pre_sandbox_init.cc
+++ b/services/on_device_model/pre_sandbox_init.cc
@@ -18,7 +18,7 @@
 #include "gpu/config/gpu_info_collector.h"                    // nogncheck
 #endif
 
-#if !BUILDFLAG(IS_FUCHSIA)
+#if !BUILDFLAG(IS_FUCHSIA) && BUILDFLAG(USE_DAWN)
 #include "third_party/dawn/include/dawn/dawn_proc.h"          // nogncheck
 #include "third_party/dawn/include/dawn/native/DawnNative.h"  // nogncheck
 #include "third_party/dawn/include/dawn/webgpu_cpp.h"         // nogncheck
@@ -91,7 +91,7 @@
   }
 #endif
 
-#if !BUILDFLAG(IS_FUCHSIA)
+#if !BUILDFLAG(IS_FUCHSIA) && BUILDFLAG(USE_DAWN)
   if (base::FeatureList::IsEnabled(kOnDeviceModelWarmDrivers)) {
     // Warm any relevant drivers before attempting to bring up the sandbox. For
     // good measure we initialize a device instance for any adapter with an
