--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -712,9 +712,7 @@
           "-Wcrl,object_path_lto",
         ]
       } else {
-        ldflags +=
-            [ "-Wl,--thinlto-cache-dir=" +
-              rebase_path("$root_out_dir/thinlto-cache", root_build_dir) ]
+        ldflags += [ "-Wl,--thinlto-cache-dir=/custom/path/to/thinlto-cache" ]
       }
 
       ldflags += [ "-Wl,--thinlto-cache-policy=$cache_policy" ]
