--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -657,9 +657,15 @@
     # available disk space, 40GB and 100000 files.
     cache_policy = "cache_size=10%:cache_size_bytes=40g:cache_size_files=100000"
 
-    # TODO(gbiv): We ideally shouldn't need to specify this; ThinLTO
-    # should be able to better manage binary size increases on its own.
-    import_instr_limit = 5
+    # Use the size savings of PGO to increase speed. Linux firefox uses a value
+    # of 10 with an import-hot-multiplier of 30. The defaults in this build are
+    # 5 and 60 respectively, while vanilla chromium uses 5 and 10.
+    if (is_linux && is_clang && chrome_pgo_phase != 0 &&
+        sample_profile_is_accurate) {
+      import_instr_limit = 10
+    } else {
+      import_instr_limit = 5
+    }
 
     if (is_win) {
       ldflags += [
