--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2197,6 +2197,15 @@
       "-Wl,--gc-sections",
     ]
   }
+
+  if (is_clang) {
+    polly_cflags = [
+      "-mllvm",
+      "-polly",
+      "-mllvm",
+      "-polly-vectorizer=stripmine",
+    ]
+  }
 }
 
 config("default_stack_frames") {
@@ -2274,7 +2283,7 @@
     # vectorization while otherwise optimizing for size.
     rustflags = [ "-Copt-level=s" ]
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
@@ -2345,7 +2354,7 @@
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O2" ] + common_optimize_on_cflags
+      cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
     }
     rustflags = [ "-Copt-level=3" ]
   }
@@ -2378,7 +2387,7 @@
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O3" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags + polly_cflags
     }
     rustflags = [ "-Copt-level=3" ]
   }
