--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2068,6 +2068,15 @@
       # smaller.
       common_optimize_on_cflags += [ "-fno-unique-section-names" ]
     }
+    if (is_clang) {
+      # Add llmv polly vectorization
+      common_optimize_on_cflags += [
+        "-mllvm",
+        "-polly",
+        "-mllvm",
+        "-polly-vectorizer=stripmine",
+      ]
+    }
 
     common_optimize_on_ldflags += [
       # Specifically tell the linker to perform optimizations.
