--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2114,7 +2114,11 @@
       "-polly-run-inliner",
       "-mllvm",
       "-polly-invariant-load-hoisting",
+      "-mllvm",
+      "-polly-parallel",
+      "-fopenmp",
     ]
+    polly_ldflags = [ "-lgomp" ]
   }
 }
 
@@ -2146,6 +2150,7 @@
 
 # Default "optimization on" config.
 config("optimize") {
+  ldflags = []
   if (is_win) {
     if (chrome_pgo_phase != 2) {
       # Favor size over speed, /O1 must be before the common flags.
@@ -2194,13 +2199,14 @@
     rustflags = [ "-Copt-level=s" ]
   } else {
     cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
+    ldflags += polly_ldflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
     # to use.
     rustflags = [ "-Copt-level=3" ]
   }
-  ldflags = common_optimize_on_ldflags
+  ldflags += common_optimize_on_ldflags
 }
 
 # Turn off optimizations.
@@ -2245,6 +2251,7 @@
 # optimization and link-time code generation which is very expensive and should
 # be used sparingly.
 config("optimize_max") {
+  ldflags = []
   if (is_nacl && is_nacl_irt) {
     # The NaCl IRT is a special case and always wants its own config.
     # Various components do:
@@ -2256,7 +2263,6 @@
     # "default_optimization", below.
     configs = [ "//build/config/nacl:irt_optimize" ]
   } else {
-    ldflags = common_optimize_on_ldflags
     if (is_win) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
@@ -2265,7 +2271,9 @@
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
+      ldflags += polly_ldflags
     }
+    ldflags += common_optimize_on_ldflags
     rustflags = [ "-Copt-level=3" ]
   }
 }
@@ -2278,6 +2286,7 @@
 # TODO(crbug.com/621335) - rework how all of these configs are related
 # so that we don't need this disclaimer.
 config("optimize_speed") {
+  ldflags = []
   if (is_nacl && is_nacl_irt) {
     # The NaCl IRT is a special case and always wants its own config.
     # Various components do:
@@ -2289,7 +2298,6 @@
     # "default_optimization", below.
     configs = [ "//build/config/nacl:irt_optimize" ]
   } else {
-    ldflags = common_optimize_on_ldflags
     if (is_win) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
@@ -2298,7 +2306,9 @@
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O3" ] + common_optimize_on_cflags + polly_cflags
+      ldflags += polly_ldflags
     }
+    ldflags += common_optimize_on_ldflags
     rustflags = [ "-Copt-level=3" ]
   }
 }
