--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2118,7 +2118,11 @@
       "-polly-run-dce",
       "-mllvm",
       "-polly-invariant-load-hoisting",
+      "-mllvm",
+      "-polly-parallel",
+      "-fopenmp",
     ]
+    polly_ldflags = [ "-lgomp" ]
   }
 }
 
@@ -2150,6 +2154,7 @@
 
 # Default "optimization on" config.
 config("optimize") {
+  ldflags = []
   if (is_win) {
     if (chrome_pgo_phase != 2) {
       # Favor size over speed, /O1 must be before the common flags.
@@ -2198,13 +2203,14 @@
     rustflags = [ "-Copt-level=s" ]
   } else {
     cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
+    ldflags += polly_ldflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
     # to use.
     rustflags = [ "-Copt-level=3" ]
   }
-  ldflags = common_optimize_on_ldflags
+  ldflags += common_optimize_on_ldflags
 }
 
 # Turn off optimizations.
@@ -2249,6 +2255,7 @@
 # optimization and link-time code generation which is very expensive and should
 # be used sparingly.
 config("optimize_max") {
+  ldflags = []
   if (is_nacl && is_nacl_irt) {
     # The NaCl IRT is a special case and always wants its own config.
     # Various components do:
@@ -2260,7 +2267,6 @@
     # "default_optimization", below.
     configs = [ "//build/config/nacl:irt_optimize" ]
   } else {
-    ldflags = common_optimize_on_ldflags
     if (is_win) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
@@ -2269,7 +2275,9 @@
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O2" ] + common_optimize_on_cflags + polly_cflags
+      ldflags += polly_ldflags
     }
+    ldflags += common_optimize_on_ldflags
     rustflags = [ "-Copt-level=3" ]
   }
 }
@@ -2282,6 +2290,7 @@
 # TODO(crbug.com/621335) - rework how all of these configs are related
 # so that we don't need this disclaimer.
 config("optimize_speed") {
+  ldflags = []
   if (is_nacl && is_nacl_irt) {
     # The NaCl IRT is a special case and always wants its own config.
     # Various components do:
@@ -2293,7 +2302,6 @@
     # "default_optimization", below.
     configs = [ "//build/config/nacl:irt_optimize" ]
   } else {
-    ldflags = common_optimize_on_ldflags
     if (is_win) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
@@ -2302,7 +2310,9 @@
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O3" ] + common_optimize_on_cflags + polly_cflags
+      ldflags += polly_ldflags
     }
+    ldflags += common_optimize_on_ldflags
     rustflags = [ "-Copt-level=3" ]
   }
 }
