--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2314,6 +2314,15 @@
       "-Wl,-O2",
       "-Wl,--gc-sections",
     ]
+
+    if (is_clang) {
+      polly_cflags = [
+        "-mllvm",
+        "-polly",
+        "-mllvm",
+        "-polly-vectorizer=stripmine",
+      ]
+    }
   }
 
   # We cannot rely on errno being set after math functions,
@@ -2504,7 +2513,7 @@
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O3" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags + polly_cflags
     }
     rustflags = [ "-Copt-level=3" ]
   }
