--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1228,6 +1228,7 @@
 config("compiler_cpu_abi") {
   cflags = []
   ldflags = []
+  rustflags = []
   defines = []
 
   configs = []
@@ -1244,6 +1245,10 @@
         cflags += [
           "-maes",
           "-mpclmul",
+          "-march=x86-64-v2",
+        ]
+        rustflags += [
+          "-Ctarget-cpu=x86-64-v2",
         ]
       } else {
         cflags += [ "-m64" ]
