--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2564,6 +2564,7 @@
   if (is_clang) {
     extra_cflags += [
       "-mpclmul",
+      "-march=x86-64-v2",
     ]
   }
 }
@@ -2662,7 +2663,10 @@
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
     # to use.
-    rustflags = [ "-Copt-level=3" ]
+    rustflags = [
+      "-Copt-level=3",
+      "-Ctarget-cpu=x86-64-v2",
+    ]
   }
 }
 
@@ -2716,7 +2720,10 @@
       ldflags += [ "-Wl,--lto-CGO2" ]
     }
   }
-  rustflags = [ "-Copt-level=3" ]
+  rustflags = [
+    "-Copt-level=3",
+    "-Ctarget-cpu=x86-64-v2",
+  ]
 }
 
 # This config can be used to override the default settings for per-component
@@ -2743,7 +2750,10 @@
       ldflags += [ "-Wl,--lto-CGO3" ]
     }
   }
-  rustflags = [ "-Copt-level=3" ]
+  rustflags = [
+    "-Copt-level=3",
+    "-Ctarget-cpu=x86-64-v2",
+  ]
 }
 
 config("optimize_fuzzing") {
