--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1193,6 +1193,7 @@
 config("compiler_cpu_abi") {
   cflags = []
   ldflags = []
+  rustflags = []
   defines = []
 
   configs = []
@@ -1208,8 +1209,10 @@
         "-m64",
         "-maes",
         "-mpclmul",
+        "-march=x86-64-v2",
       ]
       ldflags += [ "-m64" ]
+      rustflags += [ "-Ctarget-cpu=x86-64-v2" ]
     } else if (current_cpu == "x86") {
       cflags += [ "-m32" ]
       ldflags += [ "-m32" ]
