--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2686,6 +2686,7 @@
     rustflags = [
       "-Copt-level=3",
       "-Ctarget-cpu=x86-64-v2",
+      "-Ctarget-feature=+pclmulqdq,+aes,+avx",
     ]
   }
 }
@@ -2743,6 +2744,7 @@
   rustflags = [
     "-Copt-level=3",
     "-Ctarget-cpu=x86-64-v2",
+    "-Ctarget-feature=+pclmulqdq,+aes,+avx",
   ]
 }
 
@@ -2773,6 +2775,7 @@
   rustflags = [
     "-Copt-level=3",
     "-Ctarget-cpu=x86-64-v2",
+    "-Ctarget-feature=+pclmulqdq,+aes,+avx",
   ]
 }
 
