--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2669,6 +2669,7 @@
     rustflags = [
       "-Copt-level=3",
       "-Ctarget-cpu=x86-64-v2",
+      "-Ctarget-feature=+pclmulqdq,+aes,+avx",
     ]
   }
 }
@@ -2726,6 +2727,7 @@
   rustflags = [
     "-Copt-level=3",
     "-Ctarget-cpu=x86-64-v2",
+    "-Ctarget-feature=+pclmulqdq,+aes,+avx",
   ]
 }
 
@@ -2756,6 +2758,7 @@
   rustflags = [
     "-Copt-level=3",
     "-Ctarget-cpu=x86-64-v2",
+    "-Ctarget-feature=+pclmulqdq,+aes,+avx",
   ]
 }
 
