--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -7,6 +7,7 @@
 import("//build/config/ozone.gni")
 import("//build/config/ui.gni")
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//components/enterprise/buildflags/buildflags.gni")
 import("//components/guest_view/buildflags/buildflags.gni")
 import("//components/safe_browsing/buildflags.gni")
@@ -767,32 +768,6 @@
       "api/file_system/chrome_file_system_delegate.h",
       "api/file_system/file_entry_picker.cc",
       "api/file_system/file_entry_picker.h",
-      "api/image_writer_private/destroy_partitions_operation.cc",
-      "api/image_writer_private/destroy_partitions_operation.h",
-      "api/image_writer_private/error_constants.cc",
-      "api/image_writer_private/error_constants.h",
-      "api/image_writer_private/extraction_properties.cc",
-      "api/image_writer_private/extraction_properties.h",
-      "api/image_writer_private/image_writer_private_api.cc",
-      "api/image_writer_private/image_writer_private_api.h",
-      "api/image_writer_private/image_writer_utility_client.cc",
-      "api/image_writer_private/image_writer_utility_client.h",
-      "api/image_writer_private/operation.cc",
-      "api/image_writer_private/operation.h",
-      "api/image_writer_private/operation_manager.cc",
-      "api/image_writer_private/operation_manager.h",
-      "api/image_writer_private/removable_storage_provider.cc",
-      "api/image_writer_private/removable_storage_provider.h",
-      "api/image_writer_private/tar_extractor.cc",
-      "api/image_writer_private/tar_extractor.h",
-      "api/image_writer_private/write_from_file_operation.cc",
-      "api/image_writer_private/write_from_file_operation.h",
-      "api/image_writer_private/write_from_url_operation.cc",
-      "api/image_writer_private/write_from_url_operation.h",
-      "api/image_writer_private/xz_extractor.cc",
-      "api/image_writer_private/xz_extractor.h",
-      "api/image_writer_private/zip_extractor.cc",
-      "api/image_writer_private/zip_extractor.h",
       "api/language_settings_private/language_settings_private_api.cc",
       "api/language_settings_private/language_settings_private_api.h",
       "api/language_settings_private/language_settings_private_delegate.cc",
@@ -851,6 +826,37 @@
       "webstore_install_with_prompt.h",
     ]
 
+    if (enable_extractors) {
+      sources += [
+        "api/image_writer_private/destroy_partitions_operation.cc",
+        "api/image_writer_private/destroy_partitions_operation.h",
+        "api/image_writer_private/error_constants.cc",
+        "api/image_writer_private/error_constants.h",
+        "api/image_writer_private/extraction_properties.cc",
+        "api/image_writer_private/extraction_properties.h",
+        "api/image_writer_private/image_writer_private_api.cc",
+        "api/image_writer_private/image_writer_private_api.h",
+        "api/image_writer_private/image_writer_utility_client.cc",
+        "api/image_writer_private/image_writer_utility_client.h",
+        "api/image_writer_private/operation.cc",
+        "api/image_writer_private/operation.h",
+        "api/image_writer_private/operation_manager.cc",
+        "api/image_writer_private/operation_manager.h",
+        "api/image_writer_private/removable_storage_provider.cc",
+        "api/image_writer_private/removable_storage_provider.h",
+        "api/image_writer_private/tar_extractor.cc",
+        "api/image_writer_private/tar_extractor.h",
+        "api/image_writer_private/write_from_file_operation.cc",
+        "api/image_writer_private/write_from_file_operation.h",
+        "api/image_writer_private/write_from_url_operation.cc",
+        "api/image_writer_private/write_from_url_operation.h",
+        "api/image_writer_private/xz_extractor.cc",
+        "api/image_writer_private/xz_extractor.h",
+        "api/image_writer_private/zip_extractor.cc",
+        "api/image_writer_private/zip_extractor.h",
+      ]
+    }
+
     if (safe_browsing_mode != 0) {
       sources += [
         "api/safe_browsing_private/safe_browsing_private_event_router.cc",
@@ -1233,7 +1239,6 @@
       sources += [
         "api/enterprise_reporting_private/chrome_desktop_report_request_helper.cc",
         "api/enterprise_reporting_private/chrome_desktop_report_request_helper.h",
-        "api/image_writer_private/operation_nonchromeos.cc",
       ]
       deps += [
         "//components/enterprise/common/proto:connectors_proto",
@@ -1242,6 +1247,9 @@
       if (use_gio) {
         configs += [ "//build/linux:gio_config" ]
       }
+      if (enable_extractors) {
+        sources += [ "api/image_writer_private/operation_nonchromeos.cc" ]
+      }
     }
 
     if (is_chromeos) {
@@ -1479,7 +1487,7 @@
       ]
     }
 
-    if (is_linux) {
+    if (is_linux && enable_extractors) {
       sources +=
           [ "api/image_writer_private/removable_storage_provider_linux.cc" ]
     }
--- a/chrome/browser/extensions/keyed_services/chrome_browser_context_keyed_service_factories.cc
+++ b/chrome/browser/extensions/keyed_services/chrome_browser_context_keyed_service_factories.cc
@@ -38,9 +38,14 @@
 #if BUILDFLAG(ENABLE_EXTENSIONS)
 #include "chrome/browser/extensions/plugin_manager.h"
 #include "chrome/browser/extensions/warning_badge_service_factory.h"
+#include "chrome/services/file_util/buildflags.h"
 #include "content/public/common/buildflags.h"
 #endif  // BUILDFLAG(ENABLE_EXTENSIONS)
 
+#if BUILDFLAG(ENABLE_EXTRACTORS)
+#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
+#endif
+
 #if BUILDFLAG(IS_CHROMEOS)
 #include "chrome/browser/chromeos/extensions/component_extension_content_settings/component_extension_content_settings_allowlist_factory.h"
 #include "chrome/browser/extensions/forced_extensions/assessment_assistant_tracker.h"
--- a/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
+++ b/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
@@ -37,7 +37,6 @@
 #include "chrome/browser/extensions/api/bookmark_manager_private/bookmark_manager_private_api.h"
 #include "chrome/browser/extensions/api/braille_display_private/braille_display_private_api.h"
 #include "chrome/browser/extensions/api/enterprise_reporting_private/enterprise_reporting_private_event_router.h"
-#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
 #include "chrome/browser/extensions/api/language_settings_private/language_settings_private_delegate_factory.h"
 #include "chrome/browser/extensions/api/networking_private/networking_private_ui_delegate_factory_impl.h"
 #include "chrome/browser/extensions/api/passwords_private/passwords_private_delegate_factory.h"
@@ -45,6 +44,7 @@
 #include "chrome/browser/extensions/api/settings_overrides/settings_overrides_api.h"
 #include "chrome/browser/extensions/api/settings_private/settings_private_event_router_factory.h"
 #include "chrome/browser/extensions/api/side_panel/side_panel_service.h"
+#include "chrome/services/file_util/buildflags.h"
 #include "components/safe_browsing/buildflags.h"
 #include "extensions/browser/api/networking_private/networking_private_delegate_factory.h"
 #include "pdf/buildflags.h"
@@ -69,6 +69,10 @@
 #include "extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.h"
 #endif
 
+#if BUILDFLAG(ENABLE_EXTRACTORS)
+#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
+#endif
+
 #if BUILDFLAG(ENABLE_SERVICE_DISCOVERY)
 #include "chrome/browser/extensions/api/mdns/mdns_api.h"
 #endif
@@ -121,7 +125,9 @@
 #if BUILDFLAG(IS_CHROMEOS)
   extensions::InputImeAPI::GetFactoryInstance();
 #endif
+#if BUILDFLAG(ENABLE_EXTRACTORS)
   extensions::image_writer::OperationManager::GetFactoryInstance();
+#endif
   extensions::LanguageSettingsPrivateDelegateFactory::GetInstance();
 #if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
   auto networking_private_ui_delegate_factory =
--- a/chrome/common/extensions/api/api_sources.gni
+++ b/chrome/common/extensions/api/api_sources.gni
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//components/guest_view/buildflags/buildflags.gni")
 import("//pdf/features.gni")
 
@@ -85,7 +86,6 @@
     "enterprise_reporting_private.webidl",
     "experimental_actor.webidl",
     "experimental_ai_data.webidl",
-    "image_writer_private.webidl",
     "indigo_private.webidl",
     "language_settings_private.idl",
     "passwords_private.idl",
@@ -153,6 +153,10 @@
         "printing_metrics.idl",
       ]
     }
+
+    if (enable_extractor) {
+      schema_sources_ += [ "image_writer_private.idl" ]
+    }
   }
 
   if (is_linux || is_win || is_chromeos) {
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -15,6 +15,7 @@
 import("//chrome/browser/downgrade/buildflags.gni")
 import("//chrome/common/features.gni")
 import("//chrome/common/request_header_integrity/buildflags.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//chrome/services/speech/buildflags/buildflags.gni")
 import("//chrome/test/include_js_tests.gni")
 import("//components/allocation_recorder/testing/testing.gni")
@@ -7361,7 +7362,6 @@
       "//chrome/browser/web_applications:web_app_test",
       "//chrome/browser/web_applications:web_applications_unit_tests",
       "//chrome/browser/web_applications/extensions:unit_tests",
-      "//chrome/utility/image_writer:unit_tests",
       "//components/services/unzip:in_process",
       "//extensions/browser/mime_handler",
       "//extensions/browser/mime_handler:stream_container",
@@ -7369,6 +7369,10 @@
       "//extensions/renderer:unit_test_support",
     ]
 
+    if (enable_extractors) {
+      deps += [ "//chrome/utility/image_writer:unit_tests" ]
+    }
+
     if (is_mac) {
       deps += [ "//chrome/browser/apps/app_shim:unit_tests" ]
     }
