--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -866,7 +866,6 @@
     "//chrome/common/safe_browsing:proto",
     "//chrome/services/file_util:buildflags",
     "//chrome/services/file_util/public/mojom:mojom",
-    "//chrome/services/removable_storage_writer/public/mojom",
     "//components/affiliations/core/browser:affiliations",
     "//components/app_constants",
     "//components/autofill/content/browser",
@@ -1029,6 +1028,10 @@
     "//url",
   ]
 
+  if (enable_extractors) {
+    deps += [ "//chrome/services/removable_storage_writer/public/mojom" ]
+  }
+
   if (is_linux || is_mac || is_win) {
     sources += [
       "api/system_indicator/system_indicator_api.cc",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -1855,7 +1855,6 @@
       "//chrome/common/privacy_budget:test_support",
       "//chrome/renderer",
       "//chrome/renderer/companion/visual_query:browser_tests",
-      "//chrome/services/removable_storage_writer:lib",
       "//chrome/test/data/webui:resources_grit",
       "//chrome/test/data/webui/mojo:mojo_bindings",
       "//chrome/test/media_router/access_code_cast:access_code_cast_integration_base",
@@ -2209,6 +2208,10 @@
       "//v8",
     ]
 
+    if (enable_extractors) {
+      deps += [ "//chrome/services/removable_storage_writer:lib" ]
+    }
+
     allow_circular_includes_from = []
 
     if (is_chromeos) {
@@ -9205,7 +9208,6 @@
       "//chrome/common/apps/platform_apps",
       "//chrome/common/extensions:extension_features_unittest",
       "//chrome/common/extensions/api",
-      "//chrome/utility/image_writer:unit_tests",
       "//components/enterprise:test_support",
       "//components/password_manager/core/browser/features:password_features",
       "//components/password_manager/core/browser/sharing",
@@ -9237,6 +9239,10 @@
       "//tools/json_schema_compiler/test:unit_tests",
     ]
 
+    if (enable_extractors) {
+      deps += [ "//chrome/utility/image_writer:unit_tests" ]
+    }
+
     if (enable_service_discovery) {
       sources += [ "../browser/extensions/api/mdns/mdns_api_unittest.cc" ]
     }
--- a/chrome/utility/BUILD.gn
+++ b/chrome/utility/BUILD.gn
@@ -6,6 +6,7 @@
 import("//build/config/features.gni")
 import("//build/config/sysroot.gni")
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//chrome/services/speech/buildflags/buildflags.gni")
 import("//chromeos/ash/components/assistant/assistant.gni")
 import("//crypto/features.gni")
@@ -151,16 +152,24 @@
   if (enable_extensions) {
     deps += [
       "//chrome/common/extensions/api",
-      "//chrome/services/removable_storage_writer:lib",
-      "//chrome/utility/image_writer",
     ]
 
     public_deps += [
       "//chrome/common/extensions/api",
-      "//chrome/services/removable_storage_writer/public/mojom",
       "//mojo/public/cpp/bindings",
     ]
 
+    if (enable_extractors) {
+      deps += [
+        "//chrome/services/removable_storage_writer:lib",
+        "//chrome/utility/image_writer",
+      ]
+
+      public_deps += [
+        "//chrome/services/removable_storage_writer/public/mojom",
+      ]
+    }
+
     # Prevent wininet from loading in the renderer. http://crbug.com/460679
     if (is_win) {
       ldflags = [ "/DELAYLOAD:wininet.dll" ]
--- a/chrome/utility/services.cc
+++ b/chrome/utility/services.cc
@@ -9,6 +9,7 @@
 
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
+#include "chrome/services/file_util/buildflags.h"
 #include "chrome/services/on_device_translation/on_device_translation_service.h"
 #include "chrome/services/speech/buildflags/buildflags.h"
 #include "components/paint_preview/buildflags/buildflags.h"
@@ -73,7 +74,7 @@
 #include "chrome/services/file_util/file_util_service.h"  // nogncheck
 #endif
 
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(ENABLE_EXTRACTORS)
 #include "chrome/services/removable_storage_writer/public/mojom/removable_storage_writer.mojom.h"
 #include "chrome/services/removable_storage_writer/removable_storage_writer.h"
 #endif
@@ -276,7 +277,7 @@
 }
 #endif
 
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(ENABLE_EXTRACTORS)
 auto RunRemovableStorageWriter(
     mojo::PendingReceiver<chrome::mojom::RemovableStorageWriter> receiver) {
   return std::make_unique<RemovableStorageWriter>(std::move(receiver));
@@ -481,7 +482,7 @@
   services.Add(RunFileUtil);
 #endif
 
-#if BUILDFLAG(ENABLE_EXTENSIONS) && !BUILDFLAG(IS_WIN)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(ENABLE_EXTRACTORS) && !BUILDFLAG(IS_WIN)
   // On Windows, this service runs in an elevated utility process.
   services.Add(RunRemovableStorageWriter);
 #endif
