--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -572,6 +572,7 @@
     "//chrome/browser/undo",
     "//chrome/browser/update_client",
     "//chrome/common",
+    "//chrome/services/file_util:buildflags",
     "//components/bookmarks/browser",
     "//components/bookmarks/managed",
     "//components/content_settings/core/browser",
@@ -1009,7 +1010,6 @@
       "//chrome/browser/web_applications/extensions",
       "//chrome/common/extensions/api:extensions_features",
       "//chrome/services/file_util/public/mojom",
-      "//chrome/services/removable_storage_writer/public/mojom",
       "//components/affiliations/core/browser:affiliations",
       "//components/app_constants",
       "//components/autofill/content/browser",
@@ -1130,6 +1130,10 @@
       "//url",
     ]
 
+    if (enable_extractors) {
+      deps += [ "//chrome/services/removable_storage_writer/public/mojom" ]
+    }
+
     if (!is_chromeos) {
       sources += [
         "api/enterprise_reporting_private/chrome_desktop_report_request_helper.cc",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -2207,7 +2207,6 @@
       "//chrome/renderer",
       "//chrome/renderer/actor:browser_tests",
       "//chrome/renderer/webui_browser:browser_tests",
-      "//chrome/services/removable_storage_writer:lib",
       "//chrome/test/base:browser_tests",
       "//chrome/test/base:browser_tests_runner",
       "//chrome/test/components_tests:browser_tests",
@@ -2578,6 +2577,10 @@
       "//ui/webui/resources/cr_components/history_clusters:mojo_bindings",
     ]
 
+    if (enable_extractors) {
+      deps += [ "//chrome/services/removable_storage_writer:lib" ]
+    }
+
     # TODO(crbug.com/428666921): Remove this dependency when
     # chrome/browser/ui/tab_ui_helper_browsertest.cc gets modularized.
     deps += [ "//chrome/browser/ui/performance_controls/test_support" ]
--- a/chrome/utility/BUILD.gn
+++ b/chrome/utility/BUILD.gn
@@ -157,16 +157,24 @@
   if (enable_extensions) {
     deps += [
       "//chrome/common/extensions/api",
-      "//chrome/services/removable_storage_writer:lib",
-      "//chrome/utility/image_writer",
     ]
 
     public_deps += [
       "//chrome/common/extensions/api",
-      "//chrome/services/removable_storage_writer/public/mojom",
       "//mojo/public/cpp/bindings",
     ]
 
+    if (enable_extractors) {
+      deps += [
+        "//chrome/services/removable_storage_writer:lib",
+        "//chrome/utility/image_writer",
+      ]
+
+      public_deps += [
+        "//chrome/services/removable_storage_writer/public/mojom",
+      ]
+    }
+
     # Prevent wininet from loading in the renderer. http://crbug.com/40406692
     if (is_win) {
       ldflags = [ "/DELAYLOAD:wininet.dll" ]
--- a/chrome/utility/services.cc
+++ b/chrome/utility/services.cc
@@ -86,7 +86,7 @@
 #include "chrome/services/file_util/file_util_service.h"  // nogncheck
 #endif
 
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(ENABLE_EXTRACTORS)
 #include "chrome/services/removable_storage_writer/public/mojom/removable_storage_writer.mojom.h"
 #include "chrome/services/removable_storage_writer/removable_storage_writer.h"
 #endif
@@ -304,7 +304,7 @@
 }
 #endif
 
-#if BUILDFLAG(ENABLE_EXTENSIONS)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(ENABLE_EXTRACTORS)
 auto RunRemovableStorageWriter(
     mojo::PendingReceiver<chrome::mojom::RemovableStorageWriter> receiver) {
   return std::make_unique<RemovableStorageWriter>(std::move(receiver));
@@ -522,7 +522,7 @@
   services.Add(RunFileUtil);
 #endif
 
-#if BUILDFLAG(ENABLE_EXTENSIONS) && !BUILDFLAG(IS_WIN)
+#if BUILDFLAG(ENABLE_EXTENSIONS) && BUILDFLAG(ENABLE_EXTRACTORS) && !BUILDFLAG(IS_WIN)
   // On Windows, this service runs in an elevated utility process.
   services.Add(RunRemovableStorageWriter);
 #endif
