--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -8,6 +8,7 @@
 import("//build/config/ozone.gni")
 import("//build/config/ui.gni")
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//components/enterprise/buildflags/buildflags.gni")
 import("//components/safe_browsing/buildflags.gni")
 import("//content/public/common/features.gni")
@@ -702,32 +703,6 @@
       "api/identity/identity_get_auth_token_function.h",
       "api/identity/identity_remove_cached_auth_token_function.cc",
       "api/identity/identity_remove_cached_auth_token_function.h",
-      "api/image_writer_private/destroy_partitions_operation.cc",
-      "api/image_writer_private/destroy_partitions_operation.h",
-      "api/image_writer_private/error_constants.cc",
-      "api/image_writer_private/error_constants.h",
-      "api/image_writer_private/extraction_properties.cc",
-      "api/image_writer_private/extraction_properties.h",
-      "api/image_writer_private/image_writer_private_api.cc",
-      "api/image_writer_private/image_writer_private_api.h",
-      "api/image_writer_private/image_writer_utility_client.cc",
-      "api/image_writer_private/image_writer_utility_client.h",
-      "api/image_writer_private/operation.cc",
-      "api/image_writer_private/operation.h",
-      "api/image_writer_private/operation_manager.cc",
-      "api/image_writer_private/operation_manager.h",
-      "api/image_writer_private/removable_storage_provider.cc",
-      "api/image_writer_private/removable_storage_provider.h",
-      "api/image_writer_private/tar_extractor.cc",
-      "api/image_writer_private/tar_extractor.h",
-      "api/image_writer_private/write_from_file_operation.cc",
-      "api/image_writer_private/write_from_file_operation.h",
-      "api/image_writer_private/write_from_url_operation.cc",
-      "api/image_writer_private/write_from_url_operation.h",
-      "api/image_writer_private/xz_extractor.cc",
-      "api/image_writer_private/xz_extractor.h",
-      "api/image_writer_private/zip_extractor.cc",
-      "api/image_writer_private/zip_extractor.h",
       "api/language_settings_private/language_settings_private_api.cc",
       "api/language_settings_private/language_settings_private_api.h",
       "api/language_settings_private/language_settings_private_delegate.cc",
@@ -818,6 +793,37 @@
       "webstore_install_with_prompt.h",
     ]
 
+    if (enable_extractors) {
+      sources += [
+        "api/image_writer_private/destroy_partitions_operation.cc",
+        "api/image_writer_private/destroy_partitions_operation.h",
+        "api/image_writer_private/error_constants.cc",
+        "api/image_writer_private/error_constants.h",
+        "api/image_writer_private/extraction_properties.cc",
+        "api/image_writer_private/extraction_properties.h",
+        "api/image_writer_private/image_writer_private_api.cc",
+        "api/image_writer_private/image_writer_private_api.h",
+        "api/image_writer_private/image_writer_utility_client.cc",
+        "api/image_writer_private/image_writer_utility_client.h",
+        "api/image_writer_private/operation.cc",
+        "api/image_writer_private/operation.h",
+        "api/image_writer_private/operation_manager.cc",
+        "api/image_writer_private/operation_manager.h",
+        "api/image_writer_private/removable_storage_provider.cc",
+        "api/image_writer_private/removable_storage_provider.h",
+        "api/image_writer_private/tar_extractor.cc",
+        "api/image_writer_private/tar_extractor.h",
+        "api/image_writer_private/write_from_file_operation.cc",
+        "api/image_writer_private/write_from_file_operation.h",
+        "api/image_writer_private/write_from_url_operation.cc",
+        "api/image_writer_private/write_from_url_operation.h",
+        "api/image_writer_private/xz_extractor.cc",
+        "api/image_writer_private/xz_extractor.h",
+        "api/image_writer_private/zip_extractor.cc",
+        "api/image_writer_private/zip_extractor.h",
+      ]
+    }
+
     if (safe_browsing_mode != 0) {
       sources += [
         "api/safe_browsing_private/safe_browsing_private_event_router.cc",
@@ -1219,7 +1225,6 @@
       sources += [
         "api/enterprise_reporting_private/chrome_desktop_report_request_helper.cc",
         "api/enterprise_reporting_private/chrome_desktop_report_request_helper.h",
-        "api/image_writer_private/operation_nonchromeos.cc",
         "preinstalled_apps.cc",
         "preinstalled_apps.h",
       ]
@@ -1228,6 +1233,9 @@
         "//components/enterprise/common/proto:connectors_proto",
         "//components/keep_alive_registry",
       ]
+      if (enable_extractors) {
+        sources += [ "api/image_writer_private/operation_nonchromeos.cc" ]
+      }
       if (use_gio) {
         configs += [ "//build/linux:gio_config" ]
       }
@@ -1466,7 +1474,7 @@
       ]
     }
 
-    if (is_linux) {
+    if (enable_extractors && is_linux) {
       sources +=
           [ "api/image_writer_private/removable_storage_provider_linux.cc" ]
     }
--- a/chrome/browser/extensions/keyed_services/chrome_browser_context_keyed_service_factories.cc
+++ b/chrome/browser/extensions/keyed_services/chrome_browser_context_keyed_service_factories.cc
@@ -40,9 +40,14 @@
 #include "chrome/browser/extensions/permissions/permissions_updater.h"
 #include "chrome/browser/extensions/plugin_manager.h"
 #include "chrome/browser/extensions/warning_badge_service_factory.h"
+#include "chrome/services/file_util/buildflags.h"
 #include "content/public/common/buildflags.h"
 #endif  // BUILDFLAG(ENABLE_EXTENSIONS)
 
+#if BUILDFLAG(ENABLE_EXTRACTORS)
+#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
+#endif
+
 #if BUILDFLAG(IS_CHROMEOS)
 #include "chrome/browser/chromeos/extensions/component_extension_content_settings/component_extension_content_settings_allowlist_factory.h"
 #include "chrome/browser/extensions/forced_extensions/assessment_assistant_tracker.h"
--- a/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
+++ b/chrome/browser/extensions/api/api_browser_context_keyed_service_factories.cc
@@ -35,7 +35,6 @@
 #include "chrome/browser/extensions/api/braille_display_private/braille_display_private_api.h"
 #include "chrome/browser/extensions/api/enterprise_reporting_private/enterprise_reporting_private_event_router.h"
 #include "chrome/browser/extensions/api/identity/identity_api.h"
-#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
 #include "chrome/browser/extensions/api/language_settings_private/language_settings_private_delegate_factory.h"
 #include "chrome/browser/extensions/api/networking_private/networking_private_ui_delegate_factory_impl.h"
 #include "chrome/browser/extensions/api/passwords_private/passwords_private_delegate_factory.h"
@@ -47,6 +46,7 @@
 #include "chrome/browser/extensions/api/tab_capture/tab_capture_registry.h"
 #include "chrome/browser/extensions/api/tab_groups/tab_groups_event_router_factory.h"
 #include "chrome/browser/extensions/api/web_navigation/web_navigation_api.h"
+#include "chrome/services/file_util/buildflags.h"
 #include "components/safe_browsing/buildflags.h"
 #include "extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.h"
 #include "extensions/browser/api/networking_private/networking_private_delegate_factory.h"
@@ -66,6 +66,10 @@
 #include "chrome/browser/extensions/api/terminal/terminal_private_api.h"
 #endif
 
+#if BUILDFLAG(ENABLE_EXTRACTORS)
+#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
+#endif
+
 #if BUILDFLAG(ENABLE_SERVICE_DISCOVERY)
 #include "chrome/browser/extensions/api/mdns/mdns_api.h"
 #endif
@@ -115,7 +119,9 @@
 #if BUILDFLAG(IS_CHROMEOS)
   extensions::InputImeAPI::GetFactoryInstance();
 #endif
+#if BUILDFLAG(ENABLE_EXTRACTORS)
   extensions::image_writer::OperationManager::GetFactoryInstance();
+#endif
   extensions::LanguageSettingsPrivateDelegateFactory::GetInstance();
 #if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
   auto networking_private_ui_delegate_factory =
--- a/chrome/common/extensions/api/api_sources.gni
+++ b/chrome/common/extensions/api/api_sources.gni
@@ -4,6 +4,7 @@
 
 import("//build/config/features.gni")
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//pdf/features.gni")
 
 assert(enable_extensions_core)
@@ -77,7 +78,6 @@
     "downloads.idl",
     "downloads_internal.idl",
     "enterprise_reporting_private.idl",
-    "image_writer_private.idl",
     "language_settings_private.idl",
     "passwords_private.idl",
     "resources_private.idl",
@@ -95,6 +95,10 @@
     "webrtc_desktop_capture_private.idl",
   ]
 
+  if (enable_extractors) {
+    schema_sources_ += [ "image_writer_private.idl" ]
+  }
+
   if (is_chromeos) {
     schema_sources_ += [
       "accessibility_private.json",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -17,6 +17,7 @@
 import("//chrome/browser/page_load_metrics/integration_tests/sources.gni")
 import("//chrome/common/features.gni")
 import("//chrome/common/request_header_integrity/buildflags.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//chrome/services/speech/buildflags/buildflags.gni")
 import("//chrome/test/include_js_tests.gni")
 import("//components/allocation_recorder/testing/testing.gni")
@@ -4557,9 +4558,6 @@
         "../browser/extensions/api/identity/identity_apitest.cc",
         "../browser/extensions/api/identity/web_auth_flow_browsertest.cc",
         "../browser/extensions/api/idltest/idltest_apitest.cc",
-        "../browser/extensions/api/image_writer_private/extractor_browsertest.cc",
-        "../browser/extensions/api/image_writer_private/test_utils.cc",
-        "../browser/extensions/api/image_writer_private/test_utils.h",
         "../browser/extensions/api/management/management_api_browsertest.cc",
         "../browser/extensions/api/management/management_api_non_persistent_apitest.cc",
         "../browser/extensions/api/management/management_apitest.cc",
@@ -4714,6 +4712,14 @@
         "../renderer/controlled_frame/web_url_pattern_natives_browsertest.cc",
       ]
 
+      if (enable_extractors) {
+        sources += [
+          "../browser/extensions/api/image_writer_private/extractor_browsertest.cc",
+          "../browser/extensions/api/image_writer_private/test_utils.cc",
+          "../browser/extensions/api/image_writer_private/test_utils.h",
+        ]
+      }
+
       public_deps += [
         "//chrome/browser/ui/extensions:extension_enable_flow_delegate",
         "//chrome/test/supervised_user:live_tests",
@@ -4816,7 +4822,6 @@
 
       sources += [
         "../browser/extensions/api/automation/automation_apitest.cc",
-        "../browser/extensions/api/image_writer_private/image_writer_private_apitest.cc",
         "../browser/ui/extensions/extension_install_blocked_dialog_browsertest.cc",
         "../browser/ui/extensions/extension_multiple_uninstall_dialog_browsertest.cc",
         "../browser/ui/extensions/extension_uninstall_dialog_impl_browsertest.cc",
@@ -4827,6 +4832,10 @@
         "../browser/ui/extensions/settings_overridden_dialog_browsertest.cc",
       ]
 
+      if (enable_extractors) {
+        sources += [ "../browser/extensions/api/image_writer_private/image_writer_private_apitest.cc" ]
+      }
+
       if (is_chromeos) {
         sources += [
           "../browser/apps/platform_apps/api/arc_apps_private/arc_apps_private_apitest.cc",
