description: Disable ANGLE perftests
author: Michel Le Bihan <michel@lebihan.pl>

--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -5118,10 +5118,6 @@
       data_deps += [ "//third_party/dawn/src/dawn/tests:dawn_perf_tests" ]
     }
 
-    if (!is_chromeos_lacros) {
-      data_deps += [ "//third_party/angle/src/tests:angle_perftests" ]
-    }
-
     if (!is_android && !is_chromeos_lacros) {
       data_deps += [
         "//chrome/test:load_library_perf_tests",
--- a/third_party/angle/gni/angle.gni
+++ b/third_party/angle/gni/angle.gni
@@ -116,7 +116,7 @@
   }
 
   # Indicate if the rapidJSON library is available to build with in third_party/.
-  angle_has_rapidjson = angle_has_build
+  angle_has_rapidjson = false
 }
 
 declare_args() {
@@ -282,7 +282,7 @@
       is_apple &&
       (angle_enable_glsl || angle_enable_metal || use_fuzzing_engine)
 
-  angle_has_histograms = angle_has_build
+  angle_has_histograms = false
 
   # True if we want to build the ANGLE test suites. On by default in most configs.
   angle_build_tests = angle_has_build
--- a/third_party/angle/src/tests/BUILD.gn
+++ b/third_party/angle/src/tests/BUILD.gn
@@ -10,8 +10,7 @@
 declare_args() {
   build_angle_gles1_conform_tests = checkout_angle_internal
   build_angle_trace_perf_tests = false
-  build_angle_perftests =
-      is_win || is_linux || is_chromeos || is_android || is_apple || is_fuchsia
+  build_angle_perftests = false
 }
 
 declare_args() {
@@ -189,7 +188,6 @@
 
   deps = [
     ":angle_test_expectations",
-    "$angle_root:angle_json_serializer",
     "$angle_root:libANGLE",
     "$angle_root:libEGL_static",
     "$angle_root:libGLESv2_static",
@@ -419,7 +417,7 @@
   test_utils = ":angle_common_test_utils_static"
 }
 
-if (is_win || is_linux || is_chromeos || is_android || is_apple) {
+if (false) {
   import("angle_perftests.gni")
 
   # This test suite is for perf tests that use ANGLE's internals in some direct way.
@@ -1741,10 +1739,10 @@
   if (build_angle_trace_tests) {
     deps += [ ":angle_trace_tests" ]
   }
-  if (is_clang) {
+  if (false) {
     deps += [ "capture_replay_tests" ]
   }
-  if (!is_fuchsia) {
+  if (false) {
     deps += [
       ":angle_white_box_perftests",
       ":angle_white_box_tests",
