description: additional swiftshader stuff exposed by official builds
author: Andres Salomon <dilinger@debian.org>

--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -958,6 +958,7 @@
       public_deps = [ "//ui/gl:angle_library_copy" ]
     }
 
+if (false) {
     # Add the SwiftShader .dylibs in the MODULE_DIR of Chromium.app
     bundle_data("swiftshader_binaries") {
       sources = [
@@ -972,6 +973,7 @@
         "//ui/gl:swiftshader_vk_library_copy",
       ]
     }
+}
   }
 
   group("angle_library") {
@@ -980,12 +982,6 @@
     }
   }
 
-  group("swiftshader_library") {
-    if (bundle_egl_libraries) {
-      deps = [ ":swiftshader_binaries" ]
-    }
-  }
-
   if (bundle_widevine_cdm) {
     bundle_data("widevine_cdm_library_binaries") {
       sources = [ "$root_out_dir/$widevine_cdm_path/libwidevinecdm.dylib" ]
@@ -1154,7 +1150,6 @@
       ":chrome_framework_plugins",
       ":chrome_framework_resources",
       ":keystone_registration_framework",
-      ":swiftshader_library",
       ":widevine_cdm_library",
     ]
 
@@ -1241,9 +1236,6 @@
       "$root_out_dir/chrome_crashpad_handler",
       "$root_out_dir/libEGL.dylib",
       "$root_out_dir/libGLESv2.dylib",
-      "$root_out_dir/libswiftshader_libEGL.dylib",
-      "$root_out_dir/libswiftshader_libGLESv2.dylib",
-      "$root_out_dir/libvk_swiftshader.dylib",
       _framework_binary_path,
     ]
 
@@ -1276,9 +1268,6 @@
         "//third_party/angle:libEGL",
         "//third_party/angle:libGLESv2",
         "//third_party/breakpad:dump_syms",
-        "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL",
-        "//third_party/swiftshader/src/OpenGL/libGLESv2:swiftshader_libGLESv2",
-        "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan",
       ]
 
       foreach(helper_params, chrome_mac_helpers) {
@@ -1299,9 +1288,6 @@
         "$root_out_dir/chrome_crashpad_handler.dSYM",
         "$root_out_dir/libEGL.dylib.dSYM",
         "$root_out_dir/libGLESv2.dylib.dSYM",
-        "$root_out_dir/libswiftshader_libEGL.dylib.dSYM",
-        "$root_out_dir/libswiftshader_libGLESv2.dylib.dSYM",
-        "$root_out_dir/libvk_swiftshader.dylib.dSYM",
       ]
 
       deps = [
@@ -1310,9 +1296,6 @@
         "//components/crash/core/app:chrome_crashpad_handler",
         "//third_party/angle:libEGL",
         "//third_party/angle:libGLESv2",
-        "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL",
-        "//third_party/swiftshader/src/OpenGL/libGLESv2:swiftshader_libGLESv2",
-        "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan",
       ]
 
       foreach(helper_params, chrome_mac_helpers) {
@@ -1712,12 +1695,7 @@
         ":angle_gles_symbols",
         ":chrome_crashpad_symbols",
         ":chrome_symbols",
-        ":swiftshader_egl_symbols",
-        ":swiftshader_gles_symbols",
       ]
-      if (is_linux) {
-        deps += [ ":swiftshader_vk_symbols" ]
-      }
       if (!is_chromeos) {
         deps += [ ":angle_libvulkan_symbols" ]
       }
@@ -1746,6 +1724,7 @@
 
       deps = [ "//components/crash/core/app:chrome_crashpad_handler" ]
     }
+if (false) {
     extract_symbols("swiftshader_egl_symbols") {
       binary = "$root_out_dir/swiftshader/libEGL.so"
 
@@ -1756,8 +1735,6 @@
         symbol_file = "$root_out_dir/swiftshader_libegl.breakpad.$current_cpu"
       }
 
-      deps =
-          [ "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL" ]
     }
     extract_symbols("swiftshader_gles_symbols") {
       binary = "$root_out_dir/swiftshader/libGLESv2.so"
@@ -1785,6 +1762,7 @@
 
       deps = [ "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan" ]
     }
+}
     extract_symbols("angle_egl_symbols") {
       binary = "$root_out_dir/libEGL.so"
 
