description: avoid building the swiftshader library
author: Michael Gilbert <mgilbert@debian.org>

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -486,7 +486,7 @@
       ]
     }
 
-    if ((is_win || is_mac || is_linux || is_chromeos || is_fuchsia) &&
+    if (false &&
         (target_cpu == "x86" || target_cpu == "x64")) {
       deps += [
         "//third_party/swiftshader/src/Vulkan:icd_file",
--- a/third_party/angle/gni/angle.gni
+++ b/third_party/angle/gni/angle.gni
@@ -548,7 +548,6 @@
                       "$angle_root:angle_common",
                       "$angle_root:includes",
                       "$angle_root/src/tests:angle_test_expectations",
-                      "$angle_root/third_party/rapidjson:rapidjson",
                       "$angle_root/util:angle_test_utils",
                       "//testing/gmock",
                       "//testing/gtest",
--- a/gpu/vulkan/features.gni
+++ b/gpu/vulkan/features.gni
@@ -13,5 +13,5 @@
   # Enable swiftshader vulkan. Disabling it can save build time, however
   # --use-vulkan=swiftshader and some tests which use swiftshader vulkan will
   # not work.
-  enable_swiftshader_vulkan = true
+  enable_swiftshader_vulkan = false
 }
