--- a/BUILD.gn
+++ b/BUILD.gn
@@ -673,7 +673,7 @@
       ]
     }
 
-    if (is_mac || is_win || is_android || is_linux || is_chromeos) {
+    if (is_mac || is_win || is_android || is_chromeos) {
       deps += [
         "//third_party/crashpad/crashpad:crashpad_tests",
         "//third_party/crashpad/crashpad/handler:crashpad_handler",
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -309,8 +309,6 @@
           public_deps += [
             ":xdg_mime",  # Needs to be public for installer to consume files.
           ]
-
-          data_deps += [ "//components/crash/core/app:chrome_crashpad_handler" ]
         }
 
         ldflags = []
@@ -1755,7 +1753,6 @@
       deps = [
         ":angle_egl_symbols",
         ":angle_gles_symbols",
-        ":chrome_crashpad_symbols",
         ":chrome_symbols",
       ]
       if (is_linux) {
@@ -1779,18 +1776,6 @@
 
       deps = [ ":chrome" ]
     }
-    extract_symbols("chrome_crashpad_symbols") {
-      binary = "$root_out_dir/chrome_crashpad_handler"
-
-      if (current_cpu == "x86") {
-        # GYP used "ia32" so keep that naming for back-compat.
-        symbol_file = "$root_out_dir/crashpad.breakpad.ia32"
-      } else {
-        symbol_file = "$root_out_dir/crashpad.breakpad.$current_cpu"
-      }
-
-      deps = [ "//components/crash/core/app:chrome_crashpad_handler" ]
-    }
 if (false) {
     extract_symbols("swiftshader_vk_symbols") {
       binary = "$root_out_dir/libvk_swiftshader.so"
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -4700,7 +4700,7 @@
       ]
     }
 
-    if (!is_fuchsia) {
+    if (!is_fuchsia && !is_linux) {
       deps += [ "//third_party/crashpad/crashpad/client:common" ]
     }
 
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -543,7 +543,7 @@
     deps += [ "//sandbox" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_chromeos) {
     data_deps += [ "//components/crash/core/app:chrome_crashpad_handler" ]
   }
 
