--- a/third_party/crashpad/crashpad/client/crash_report_database.h
+++ b/third_party/crashpad/crashpad/client/crash_report_database.h
@@ -24,12 +24,12 @@
 #include <vector>
 
 #include "base/files/file_path.h"
-#include "util/file/file_io.h"
-#include "util/file/file_reader.h"
-#include "util/file/file_writer.h"
-#include "util/file/scoped_remove_file.h"
-#include "util/misc/metrics.h"
-#include "util/misc/uuid.h"
+#include "third_party/crashpad/crashpad/util/file/file_io.h"
+#include "third_party/crashpad/crashpad/util/file/file_reader.h"
+#include "third_party/crashpad/crashpad/util/file/file_writer.h"
+#include "third_party/crashpad/crashpad/util/file/scoped_remove_file.h"
+#include "third_party/crashpad/crashpad/util/misc/metrics.h"
+#include "third_party/crashpad/crashpad/util/misc/uuid.h"
 
 namespace crashpad {
 
--- a/third_party/crashpad/crashpad/util/file/file_reader.h
+++ b/third_party/crashpad/crashpad/util/file/file_reader.h
@@ -18,8 +18,8 @@
 #include <sys/types.h>
 
 #include "base/files/file_path.h"
-#include "util/file/file_io.h"
-#include "util/file/file_seeker.h"
+#include "third_party/crashpad/crashpad/util/file/file_io.h"
+#include "third_party/crashpad/crashpad/util/file/file_seeker.h"
 
 namespace crashpad {
 
--- a/third_party/crashpad/crashpad/util/file/file_seeker.h
+++ b/third_party/crashpad/crashpad/util/file/file_seeker.h
@@ -15,7 +15,7 @@
 #ifndef CRASHPAD_UTIL_FILE_FILE_SEEKER_H_
 #define CRASHPAD_UTIL_FILE_FILE_SEEKER_H_
 
-#include "util/file/file_io.h"
+#include "third_party/crashpad/crashpad/util/file/file_io.h"
 
 namespace crashpad {
 
--- a/third_party/crashpad/crashpad/util/file/file_writer.h
+++ b/third_party/crashpad/crashpad/util/file/file_writer.h
@@ -21,8 +21,8 @@
 
 #include "base/files/file_path.h"
 #include "build/build_config.h"
-#include "util/file/file_io.h"
-#include "util/file/file_seeker.h"
+#include "third_party/crashpad/crashpad/util/file/file_io.h"
+#include "third_party/crashpad/crashpad/util/file/file_seeker.h"
 
 namespace crashpad {
 
--- a/third_party/crashpad/crashpad/util/misc/metrics.h
+++ b/third_party/crashpad/crashpad/util/misc/metrics.h
@@ -18,7 +18,7 @@
 #include <inttypes.h>
 
 #include "build/build_config.h"
-#include "util/file/file_io.h"
+#include "third_party/crashpad/crashpad/util/file/file_io.h"
 
 #if BUILDFLAG(IS_IOS)
 #include "util/ios/ios_intermediate_dump_format.h"
