--- a/BUILD.gn
+++ b/BUILD.gn
@@ -684,14 +684,6 @@
     ]
   }
 
-  if (is_mac || is_win || is_android || is_linux || is_chromeos) {
-    deps += [
-      "//third_party/crashpad/crashpad:crashpad_tests",
-      "//third_party/crashpad/crashpad/handler:crashpad_handler",
-      "//third_party/crashpad/crashpad/tools:crashpad_database_util",
-    ]
-  }
-
   if (enable_message_center) {
     deps += [ "//ui/message_center:message_center_unittests" ]
   }
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -4540,10 +4540,6 @@
       ]
     }
 
-    if (!is_fuchsia) {
-      deps += [ "//third_party/crashpad/crashpad/client:common" ]
-    }
-
     # On Windows, the hashes are embedded in //chrome:chrome_initial rather
     # than here in :chrome_dll.
     if (!is_win) {
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -305,8 +305,6 @@
           public_deps += [
             ":xdg_mime",  # Needs to be public for installer to consume files.
           ]
-
-          data_deps += [ "//components/crash/core/app:chrome_crashpad_handler" ]
         }
 
         ldflags = []
@@ -1729,7 +1727,6 @@
       deps = [
         ":angle_egl_symbols",
         ":angle_gles_symbols",
-        ":chrome_crashpad_symbols",
         ":chrome_symbols",
       ]
       if (is_linux) {
@@ -1753,18 +1750,6 @@
 
       deps = [ ":chrome" ]
     }
-    extract_symbols("chrome_crashpad_symbols") {
-      binary = "$root_out_dir/chrome_crashpad_handler"
-
-      if (current_cpu == "x86") {
-        # GYP used "ia32" so keep that naming for back-compat.
-        symbol_file = "$root_out_dir/crashpad.breakpad.ia32"
-      } else {
-        symbol_file = "$root_out_dir/crashpad.breakpad.$current_cpu"
-      }
-
-      deps = [ "//components/crash/core/app:chrome_crashpad_handler" ]
-    }
 if (false) {
     extract_symbols("swiftshader_vk_symbols") {
       binary = "$root_out_dir/libvk_swiftshader.so"
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -573,10 +573,6 @@
     deps += [ "//sandbox" ]
   }
 
-  if (is_linux || is_chromeos) {
-    data_deps += [ "//components/crash/core/app:chrome_crashpad_handler" ]
-  }
-
   # Normally set to false (see build/args/headless.gn), but we can optionally
   # use external v8 startup data too.
   if ((!is_win || is_component_build) && v8_use_external_startup_data) {
