--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1501,7 +1501,7 @@
     ]
   }
 
-  if (!is_fuchsia) {
+  if (!is_fuchsia && !is_linux) {
     public_deps += [ "//components/crash/core/app" ]
   }
 }
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -6681,7 +6681,6 @@
     ]
     deps += [
       "//components/crash/content/browser",
-      "//components/crash/core/app",
       "//components/version_info:generate_version_info",
     ]
   }
--- a/chrome/browser/crash_upload_list/BUILD.gn
+++ b/chrome/browser/crash_upload_list/BUILD.gn
@@ -44,7 +44,7 @@
     deps += [ "//chrome/android:chrome_jni_headers" ]
   }
 
-  if (!is_fuchsia) {
+  if (!is_fuchsia && !is_linux) {
     deps += [ "//components/crash/core/app:app" ]
   }
 
--- a/chrome/browser/google/BUILD.gn
+++ b/chrome/browser/google/BUILD.gn
@@ -70,7 +70,7 @@
   if (is_posix || is_fuchsia) {
     sources += [ "google_update_settings_posix.cc" ]
   }
-  if (is_posix) {
+  if (is_posix && !is_linux) {
     deps += [ "//components/crash/core/app" ]
   }
 }
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -748,7 +748,7 @@
     ]
   }
 
-  if (!is_fuchsia) {
+  if (!is_fuchsia && !is_linux) {
     deps += [ "//components/crash/core/app" ]
   }
 
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -352,7 +352,7 @@
     ]
   }
 
-  if (!is_fuchsia) {
+  if (!is_fuchsia && !is_linux) {
     deps += [ "//components/crash/core/app" ]
   }
 
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -933,7 +933,7 @@
     deps += [ "//components/os_crypt/sync" ]
   }
 
-  if (is_win || (is_posix && !is_mac)) {
+  if (is_win || (is_posix && !is_linux && !is_mac)) {
     deps += [ "//components/crash/core/app" ]
   }
 
