--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -8,6 +8,7 @@
 import("//build/config/ozone.gni")
 import("//build/config/ui.gni")
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//components/nacl/features.gni")
 import("//extensions/buildflags/buildflags.gni")
 import("//mojo/public/tools/bindings/mojom.gni")
@@ -198,32 +199,6 @@
     "api/identity/web_auth_flow_info_bar_delegate.h",
     "api/idltest/idltest_api.cc",
     "api/idltest/idltest_api.h",
-    "api/image_writer_private/destroy_partitions_operation.cc",
-    "api/image_writer_private/destroy_partitions_operation.h",
-    "api/image_writer_private/error_constants.cc",
-    "api/image_writer_private/error_constants.h",
-    "api/image_writer_private/extraction_properties.cc",
-    "api/image_writer_private/extraction_properties.h",
-    "api/image_writer_private/image_writer_private_api.cc",
-    "api/image_writer_private/image_writer_private_api.h",
-    "api/image_writer_private/image_writer_utility_client.cc",
-    "api/image_writer_private/image_writer_utility_client.h",
-    "api/image_writer_private/operation.cc",
-    "api/image_writer_private/operation.h",
-    "api/image_writer_private/operation_manager.cc",
-    "api/image_writer_private/operation_manager.h",
-    "api/image_writer_private/removable_storage_provider.cc",
-    "api/image_writer_private/removable_storage_provider.h",
-    "api/image_writer_private/tar_extractor.cc",
-    "api/image_writer_private/tar_extractor.h",
-    "api/image_writer_private/write_from_file_operation.cc",
-    "api/image_writer_private/write_from_file_operation.h",
-    "api/image_writer_private/write_from_url_operation.cc",
-    "api/image_writer_private/write_from_url_operation.h",
-    "api/image_writer_private/xz_extractor.cc",
-    "api/image_writer_private/xz_extractor.h",
-    "api/image_writer_private/zip_extractor.cc",
-    "api/image_writer_private/zip_extractor.h",
     "api/instance_id/instance_id_api.cc",
     "api/instance_id/instance_id_api.h",
     "api/language_settings_private/language_settings_private_api.cc",
@@ -734,6 +709,37 @@
     "window_controller_list_observer.h",
   ]
 
+  if (enable_extractors) {
+    sources += [
+      "api/image_writer_private/destroy_partitions_operation.cc",
+      "api/image_writer_private/destroy_partitions_operation.h",
+      "api/image_writer_private/error_constants.cc",
+      "api/image_writer_private/error_constants.h",
+      "api/image_writer_private/extraction_properties.cc",
+      "api/image_writer_private/extraction_properties.h",
+      "api/image_writer_private/image_writer_private_api.cc",
+      "api/image_writer_private/image_writer_private_api.h",
+      "api/image_writer_private/image_writer_utility_client.cc",
+      "api/image_writer_private/image_writer_utility_client.h",
+      "api/image_writer_private/operation.cc",
+      "api/image_writer_private/operation.h",
+      "api/image_writer_private/operation_manager.cc",
+      "api/image_writer_private/operation_manager.h",
+      "api/image_writer_private/removable_storage_provider.cc",
+      "api/image_writer_private/removable_storage_provider.h",
+      "api/image_writer_private/tar_extractor.cc",
+      "api/image_writer_private/tar_extractor.h",
+      "api/image_writer_private/write_from_file_operation.cc",
+      "api/image_writer_private/write_from_file_operation.h",
+      "api/image_writer_private/write_from_url_operation.cc",
+      "api/image_writer_private/write_from_url_operation.h",
+      "api/image_writer_private/xz_extractor.cc",
+      "api/image_writer_private/xz_extractor.h",
+      "api/image_writer_private/zip_extractor.cc",
+      "api/image_writer_private/zip_extractor.h",
+    ]
+  }
+
   # Chrome OS does not support Native Messaging policies.
   if (!is_chromeos) {
     sources += [
@@ -835,6 +841,7 @@
     "//chrome/browser/web_applications/extensions",
     "//chrome/common/extensions/api:extensions_features",
     "//chrome/common/safe_browsing:proto",
+    "//chrome/services/file_util:buildflags",
     "//chrome/services/file_util/public/mojom:mojom",
     "//chrome/services/removable_storage_writer/public/mojom",
     "//components/affiliations/core/browser:affiliations",
@@ -1100,7 +1107,6 @@
     sources += [
       "api/enterprise_reporting_private/chrome_desktop_report_request_helper.cc",
       "api/enterprise_reporting_private/chrome_desktop_report_request_helper.h",
-      "api/image_writer_private/operation_nonchromeos.cc",
       "api/messaging/native_message_process_host.cc",
       "api/messaging/native_message_process_host.h",
       "api/messaging/native_messaging_launch_from_native.cc",
@@ -1116,6 +1122,9 @@
       "//components/enterprise/common/proto:connectors_proto",
       "//components/keep_alive_registry",
     ]
+    if (enable_extractors) {
+      sources += [ "api/image_writer_private/operation_nonchromeos.cc" ]
+    }
     if (is_posix) {
       sources += [ "api/messaging/native_process_launcher_posix.cc" ]
     }
@@ -1314,7 +1323,7 @@
     ]
   }
 
-  if (is_linux || is_chromeos_lacros) {
+  if (enable_extractors && (is_linux || is_chromeos_lacros)) {
     sources +=
         [ "api/image_writer_private/removable_storage_provider_linux.cc" ]
   }
--- a/chrome/browser/extensions/chrome_browser_context_keyed_service_factories.cc
+++ b/chrome/browser/extensions/chrome_browser_context_keyed_service_factories.cc
@@ -7,7 +7,6 @@
 #include "build/build_config.h"
 #include "chrome/browser/extensions/activity_log/activity_log.h"
 #include "chrome/browser/extensions/api/bookmarks/bookmarks_api_watcher.h"
-#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
 #include "chrome/browser/extensions/api/messaging/incognito_connectability.h"
 #include "chrome/browser/extensions/api/notifications/extension_notification_display_helper_factory.h"
 #include "chrome/browser/extensions/api/web_authentication_proxy/web_authentication_proxy_service.h"
@@ -26,8 +25,13 @@
 #include "chrome/browser/extensions/permissions/permissions_updater.h"
 #include "chrome/browser/extensions/plugin_manager.h"
 #include "chrome/browser/extensions/warning_badge_service_factory.h"
+#include "chrome/services/file_util/buildflags.h"
 #include "ppapi/buildflags/buildflags.h"
 
+#if BUILDFLAG(ENABLE_EXTRACTORS)
+#include "chrome/browser/extensions/api/image_writer_private/operation_manager.h"
+#endif
+
 namespace chrome_extensions {
 
 void EnsureChromeBrowserContextKeyedServiceFactoriesBuilt() {
@@ -43,7 +47,9 @@
   extensions::ExtensionNotificationDisplayHelperFactory::GetInstance();
   extensions::ExtensionSystemFactory::GetInstance();
   extensions::ExtensionWebUIOverrideRegistrar::GetFactoryInstance();
+#if BUILDFLAG(ENABLE_EXTRACTORS)
   extensions::image_writer::OperationManager::GetFactoryInstance();
+#endif
   extensions::IncognitoConnectability::EnsureFactoryBuilt();
   extensions::InstallTrackerFactory::GetInstance();
   extensions::InstallVerifierFactory::GetInstance();
--- a/chrome/common/extensions/api/api_sources.gni
+++ b/chrome/common/extensions/api/api_sources.gni
@@ -5,6 +5,7 @@
 import("//build/config/chromeos/ui_mode.gni")
 import("//build/config/features.gni")
 import("//chrome/common/features.gni")
+import("//chrome/services/file_util/public/features.gni")
 import("//pdf/features.gni")
 
 assert(enable_extensions)
@@ -35,7 +36,6 @@
   "gcm.json",
   "history.json",
   "identity.idl",
-  "image_writer_private.idl",
   "instance_id.json",
   "language_settings_private.idl",
   "notifications.idl",
@@ -68,6 +68,10 @@
   "windows.json",
 ]
 
+if (enable_extractors) {
+  schema_sources_ += [ "image_writer_private.idl" ]
+}
+
 if (!is_android) {
   schema_sources_ += [ "processes.idl" ]
 }
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -18,6 +18,7 @@
 import("//chrome/browser/sync/test/integration/sync_integration_tests.gni")
 import("//chrome/common/features.gni")
 
+import("//chrome/services/file_util/public/features.gni")
 import("//chrome/services/speech/buildflags/buildflags.gni")
 import("//chrome/test/include_js_tests.gni")
 import("//chrome/version.gni")
@@ -3676,9 +3677,6 @@
         "../browser/extensions/api/identity/web_auth_flow_browsertest.cc",
         "../browser/extensions/api/idle/idle_apitest.cc",
         "../browser/extensions/api/idltest/idltest_apitest.cc",
-        "../browser/extensions/api/image_writer_private/extractor_browsertest.cc",
-        "../browser/extensions/api/image_writer_private/test_utils.cc",
-        "../browser/extensions/api/image_writer_private/test_utils.h",
         "../browser/extensions/api/instance_id/instance_id_apitest.cc",
         "../browser/extensions/api/management/management_api_browsertest.cc",
         "../browser/extensions/api/management/management_api_non_persistent_apitest.cc",
@@ -3881,6 +3879,14 @@
         "../browser/user_bypass/user_bypass_web_contents_observer_browsertest.cc",
       ]
 
+      if (enable_extractors) {
+        sources += [
+          "../browser/extensions/api/image_writer_private/extractor_browsertest.cc",
+          "../browser/extensions/api/image_writer_private/test_utils.cc",
+          "../browser/extensions/api/image_writer_private/test_utils.h",
+        ]
+      }
+
       if (is_win) {
         data_deps += [
           "//chrome/browser/extensions/api:native_messaging_test_echo_&_host",
@@ -3957,9 +3963,12 @@
       if (!is_chromeos_lacros) {
         sources += [
           "../browser/extensions/api/automation/automation_apitest.cc",
-          "../browser/extensions/api/image_writer_private/image_writer_private_apitest.cc",
           "../browser/extensions/api/proxy/proxy_apitest.cc",
         ]
+
+        if (enable_extractors) {
+          sources += [ "../browser/extensions/api/image_writer_private/image_writer_private_apitest.cc" ]
+        }
       }
 
       if (is_chromeos) {
@@ -5354,12 +5363,17 @@
     if (is_mac || is_win || is_linux || is_chromeos_lacros) {
       sources += [
         # Tests for non mobile and non CrOS (includes Linux, Win, Mac).
-        "../browser/extensions/api/image_writer_private/image_writer_utility_client_browsertest.cc",
         "../browser/profiles/profile_statistics_browsertest.cc",
         "../browser/ui/signin/signin_reauth_view_controller_browsertest.cc",
         "../browser/ui/webui/signin/turn_sync_on_helper_browsertest.cc",
       ]
 
+      if (enable_extractors) {
+        sources += [
+          "../browser/extensions/api/image_writer_private/image_writer_utility_client_browsertest.cc",
+        ]
+      }
+
       if (!is_chromeos_lacros) {
         sources += [
           "../browser/browser_switcher/browser_switcher_browsertest.cc",
@@ -9039,15 +9053,6 @@
       "../browser/extensions/api/identity/identity_api_unittest.cc",
       "../browser/extensions/api/identity/identity_mint_queue_unittest.cc",
       "../browser/extensions/api/identity/identity_token_cache_unittest.cc",
-      "../browser/extensions/api/image_writer_private/destroy_partitions_operation_unittest.cc",
-      "../browser/extensions/api/image_writer_private/extractor_unittest.cc",
-      "../browser/extensions/api/image_writer_private/image_writer_private_api_unittest.cc",
-      "../browser/extensions/api/image_writer_private/operation_manager_unittest.cc",
-      "../browser/extensions/api/image_writer_private/operation_unittest.cc",
-      "../browser/extensions/api/image_writer_private/test_utils.cc",
-      "../browser/extensions/api/image_writer_private/test_utils.h",
-      "../browser/extensions/api/image_writer_private/write_from_file_operation_unittest.cc",
-      "../browser/extensions/api/image_writer_private/write_from_url_operation_unittest.cc",
       "../browser/extensions/api/language_settings_private/language_settings_private_api_unittest.cc",
       "../browser/extensions/api/language_settings_private/language_settings_private_delegate_unittest.cc",
       "../browser/extensions/api/management/management_api_unittest.cc",
@@ -11508,6 +11513,20 @@
         "../browser/ui/views/touch_events_interactive_uitest_win.cc",
       ]
 
+      if (enable_extractors) {
+        sources += [
+          "../browser/extensions/api/image_writer_private/destroy_partitions_operation_unittest.cc",
+          "../browser/extensions/api/image_writer_private/extractor_unittest.cc",
+          "../browser/extensions/api/image_writer_private/image_writer_private_api_unittest.cc",
+          "../browser/extensions/api/image_writer_private/operation_manager_unittest.cc",
+          "../browser/extensions/api/image_writer_private/operation_unittest.cc",
+          "../browser/extensions/api/image_writer_private/test_utils.cc",
+          "../browser/extensions/api/image_writer_private/test_utils.h",
+          "../browser/extensions/api/image_writer_private/write_from_file_operation_unittest.cc",
+          "../browser/extensions/api/image_writer_private/write_from_url_operation_unittest.cc",
+        ]
+      }
+
       if (use_aura) {
         sources += [ "../browser/ui/views/autofill/autofill_accessibility_win_browsertest.cc" ]
       }
