--- a/components/lens/lens_features.cc
+++ b/components/lens/lens_features.cc
@@ -13,7 +13,7 @@
 
 BASE_FEATURE(kLensStandalone,
              "LensStandalone",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kLensImageCompression,
              "LensImageCompression",
@@ -25,7 +25,7 @@
 
 BASE_FEATURE(kEnableLatencyLogging,
              "LensImageLatencyLogging",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kEnableRegionSearchKeyboardShortcut,
              "LensEnableRegionSearchKeyboardShortcut",
@@ -37,7 +37,7 @@
 
 BASE_FEATURE(kEnableImageSearchSidePanelFor3PDse,
              "EnableImageSearchSidePanelFor3PDse",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kLensRegionSearchStaticPage,
              "LensRegionSearchStaticPage",
@@ -45,7 +45,7 @@
 
 BASE_FEATURE(kEnableContextMenuInLensSidePanel,
              "EnableContextMenuInLensSidePanel",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kLensOverlay, "LensOverlay", base::FEATURE_DISABLED_BY_DEFAULT);
 const base::FeatureParam<int> kLensOverlayMinRamMb{&kLensOverlay, "min_ram_mb",
@@ -78,7 +78,7 @@
         true};
 
 constexpr base::FeatureParam<bool> kShouldIssuePreconnectForLens{
-    &kLensStandalone, "lens-issue-preconnect", true};
+    &kLensStandalone, "lens-issue-preconnect", false};
 
 constexpr base::FeatureParam<std::string> kPreconnectKeyForLens{
     &kLensStandalone, "lens-preconnect-key", "https://google.com"};
