--- a/components/lens/lens_features.cc
+++ b/components/lens/lens_features.cc
@@ -16,12 +16,12 @@
 
 namespace lens::features {
 
-BASE_FEATURE(kLensStandalone, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kLensStandalone, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kShowContextualTasksMenuIcon, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kLensOverlay,
-#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS) || BUILDFLAG(IS_LINUX)
              base::FEATURE_DISABLED_BY_DEFAULT
 #else
              base::FEATURE_ENABLED_BY_DEFAULT
@@ -507,7 +507,7 @@
     &kLensStandalone, "lens-html-redirect-fix", false};
 
 constexpr base::FeatureParam<bool> kShouldIssuePreconnectForLens{
-    &kLensStandalone, "lens-issue-preconnect", true};
+    &kLensStandalone, "lens-issue-preconnect", false};
 
 constexpr base::FeatureParam<std::string> kPreconnectKeyForLens{
     &kLensStandalone, "lens-preconnect-key", "https://google.com"};
