--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -46,7 +46,7 @@
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 import("//third_party/protobuf/proto_library.gni")
-import("//third_party/protobuf/proto_library.gni")
+import("//third_party/webrtc/webrtc.gni")
 import("//third_party/widevine/cdm/widevine.gni")
 import("//ui/base/ui_features.gni")
 import("//ui/gl/features.gni")
@@ -3032,7 +3032,7 @@
       "//tools/media_engagement_preload:make_dafsa_runtime",
     ]
 
-    if (!is_component_build) {
+    if (!is_component_build && rtc_build_tools) {
       # WebRTC's frame_analyzer doesn't build correctly on component builds,
       # because some compilers will get confused by the fact that WebRTC
       # symbols are exported while liking a WebRTC executable (which should
