--- a/BUILD.gn
+++ b/BUILD.gn
@@ -34,6 +34,7 @@
 import("//remoting/remoting_enable.gni")
 import("//services/on_device_model/on_device_model.gni")
 import("//third_party/closure_compiler/compile_js.gni")
+import("//third_party/webrtc/webrtc.gni")
 import("//tools/ipc_fuzzer/ipc_fuzzer.gni")
 import("//ui/base/ui_features.gni")
 import("//ui/gl/features.gni")
@@ -1771,7 +1772,7 @@
       }
     }
 
-    if (!is_component_build) {
+    if (!is_component_build && rtc_build_tools) {
       # See https://bugs.chromium.org/p/chromium/issues/detail?id=942546#c11.
       deps += [
         "//third_party/webrtc/rtc_tools:rtp_generator",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -46,6 +46,7 @@
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 import("//third_party/protobuf/proto_library.gni")
+import("//third_party/webrtc/webrtc.gni")
 import("//third_party/widevine/cdm/widevine.gni")
 import("//ui/base/ui_features.gni")
 import("//ui/gl/features.gni")
@@ -2496,7 +2497,7 @@
       "//tools/media_engagement_preload:make_dafsa_runtime",
     ]
 
-    if (!is_component_build) {
+    if (!is_component_build && rtc_build_tools) {
       # WebRTC's frame_analyzer doesn't build correctly on component builds,
       # because some compilers will get confused by the fact that WebRTC
       # symbols are exported while liking a WebRTC executable (which should
