--- a/chrome/browser/ui/webui/settings/accessibility_main_handler.h
+++ b/chrome/browser/ui/webui/settings/accessibility_main_handler.h
@@ -10,6 +10,7 @@
 #include "build/chromeos_buildflags.h"
 #include "chrome/browser/screen_ai/screen_ai_install_state.h"
 #include "chrome/browser/ui/webui/settings/settings_page_ui_handler.h"
+#include "services/screen_ai/buildflags/buildflags.h"
 
 #if BUILDFLAG(IS_CHROMEOS_ASH)
 #include "chrome/browser/ash/accessibility/accessibility_manager.h"
@@ -38,7 +39,9 @@
   void HandleGetScreenReaderState(const base::Value::List& args);
   void HandleCheckAccessibilityImageLabels(const base::Value::List& args);
 
+#if BUILDFLAG(ENABLE_SCREEN_AI_SERVICE)
   void HandleGetScreenAIInstallState(const base::Value::List& args);
+#endif  // BUILDFLAG(ENABLE_SCREEN_AI_SERVICE)
 
   void SendScreenReaderStateChanged();
 
