--- a/components/feed/feed_feature_list.cc
+++ b/components/feed/feed_feature_list.cc
@@ -23,7 +23,7 @@
 // changed, please update the cached one's default value in CachedFeatureFlags.
 BASE_FEATURE(kInterestFeedV2,
              "InterestFeedV2",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kInterestFeedV2Autoplay,
              "InterestFeedV2Autoplay",
