--- a/net/base/features.cc
+++ b/net/base/features.cc
@@ -106,10 +106,10 @@
         &kNetworkQualityEstimator,
         "EffectiveConnectionTypeRecomputationInterval", base::Seconds(10)};
 
-BASE_FEATURE(SplitCacheByIncludeCredentials, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(SplitCacheByIncludeCredentials, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(SplitCacheByNetworkIsolationKey,
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Note: Use of this feature is gated on the HTTP cache itself being
 // partitioned, which is controlled by the kSplitCacheByNetworkIsolationKey
@@ -118,7 +118,7 @@
              base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(SplitCodeCacheByNetworkIsolationKey,
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(PartitionConnectionsByNetworkIsolationKey,
              base::FEATURE_DISABLED_BY_DEFAULT);
