--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -1332,7 +1332,7 @@
 
 // Controls whether the WebUSB API is enabled:
 // https://wicg.github.io/webusb
-BASE_FEATURE(kWebUsb, "WebUSB", base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kWebUsb, "WebUSB", base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Apply `PrefetchPriority::kHighest` for Webview Prefetch API.
 BASE_FEATURE(kWebViewPrefetchHighestPrefetchPriority,
