--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -1014,7 +1014,7 @@
 
 // Controls whether site isolation should use origins instead of scheme and
 // eTLD+1.
-BASE_FEATURE(StrictOriginIsolation, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(StrictOriginIsolation, base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Controls whether subframe process reuse should be restricted according to
 // resource usage policies. Namely, a process that is already consuming too
