--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -2044,7 +2044,7 @@
 // <link url, top-level site, frame origin>
 BASE_FEATURE(kPartitionVisitedLinkDatabase,
              "PartitionVisitedLinkDatabase",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enables the use of the PaintCache for Path2D objects that are rasterized
 // out of process.  Has no effect when kCanvasOopRasterization is disabled.
