--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -746,7 +746,7 @@
 
 BASE_FEATURE(kDisableThirdPartyStoragePartitioningDeprecationTrial2,
              "DisableThirdPartyStoragePartitioningDeprecationTrial2",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Kill switch for not WebGL DrawingBuffer using SharedImage without
 // GpuMemoryBuffer (overlay or low latency canvas).
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -1349,7 +1349,7 @@
       origin_trial_type: "deprecation",
       origin_trial_allows_insecure: true,
       origin_trial_allows_third_party: true,
-      status: "experimental",
+      origin_trial_os: ["linux"],
       base_feature: "none",
       browser_process_read_write_access: true,
     },
