--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1942,7 +1942,7 @@
     ]
   }
 
-  if (is_posix && !is_mac) {
+  if (is_posix && !is_linux && !is_mac) {
     sources += [ "//chrome/app/chrome_crash_reporter_client.cc" ]
     deps += [
       "//chrome/app:chrome_crash_reporter_client",
@@ -4272,7 +4272,7 @@
     public_deps += [ "//chrome/browser/downgrade:impl" ]
   }
 
-  if (is_posix && !is_mac) {
+  if (is_posix && !is_linux && !is_mac) {
     deps += [
       "//chrome/app:chrome_crash_reporter_client",
       "//components/crash/content/browser",
--- a/chrome/browser/metrics/BUILD.gn
+++ b/chrome/browser/metrics/BUILD.gn
@@ -221,7 +221,7 @@
     ]
   }
 
-  if (is_win || is_android || is_linux) {
+  if (is_win || is_android) {
     public += [ "chrome_metrics_service_crash_reporter.h" ]
   }
 
@@ -492,7 +492,7 @@
     deps += [ "//chrome/browser/updates:update_metrics" ]
   }
 
-  if (is_win || is_android || is_linux) {
+  if (is_win || is_android) {
     sources += [ "chrome_metrics_service_crash_reporter.cc" ]
   }
 
--- a/chrome/browser/metrics/chrome_metrics_service_client.cc
+++ b/chrome/browser/metrics/chrome_metrics_service_client.cc
@@ -208,7 +208,7 @@
 #include "components/metrics/motherboard_metrics_provider.h"
 #endif
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID)
 #include "chrome/browser/metrics/chrome_metrics_service_crash_reporter.h"
 #endif
 
@@ -235,7 +235,7 @@
 // Needs to be kept in sync with the writer in
 // third_party/crashpad/crashpad/handler/handler_main.cc.
 const char kCrashpadHistogramAllocatorName[] = "CrashpadMetrics";
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID)
 ChromeMetricsServiceCrashReporter& GetCrashReporter() {
   static base::NoDestructor<ChromeMetricsServiceCrashReporter> crash_reporter;
   return *crash_reporter;
@@ -664,7 +664,7 @@
 void ChromeMetricsServiceClient::OnEnvironmentUpdate(std::string* environment) {
   // TODO(https://bugs.chromium.org/p/crashpad/issues/detail?id=135): call this
   // on Mac when the Crashpad API supports it.
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID)
   // Register the environment with the crash reporter. Note that there is a
   // window from startup to this point during which crash reports will not have
   // an environment set.
--- a/chrome/common/crash_keys.cc
+++ b/chrome/common/crash_keys.cc
@@ -18,6 +18,7 @@
 #include "base/strings/stringprintf.h"
 #include "build/build_config.h"
 #include "chrome/common/chrome_switches.h"
+#include "components/crash/core/common/crash_buildflags.h"
 #include "components/crash/core/common/crash_key.h"
 #include "components/crash/core/common/crash_keys.h"
 #include "components/webui/flags/flags_ui_switches.h"
@@ -56,7 +57,13 @@
   ~CrashKeyWithName() = delete;
 
   std::string_view Name() const { return name_; }
-  std::string_view Value() const { return crash_key_.value(); }
+  std::string_view Value() const {
+#if BUILDFLAG(USE_CRASH_KEY_STUBS)
+    return "";
+#else
+    return crash_key_.value();
+#endif
+  }
   void Clear() { crash_key_.Clear(); }
   void Set(std::string_view value) { crash_key_.Set(value); }
