--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1926,7 +1926,7 @@
     ]
   }
 
-  if (is_posix && !is_mac) {
+  if (is_posix && !is_linux && !is_mac) {
     sources += [ "//chrome/app/chrome_crash_reporter_client.cc" ]
     deps += [
       "//chrome/app:chrome_crash_reporter_client",
@@ -4265,7 +4265,7 @@
     public_deps += [ "//chrome/browser/downgrade:impl" ]
   }
 
-  if (is_posix && !is_mac) {
+  if (is_posix && !is_linux && !is_mac) {
     deps += [
       "//chrome/app:chrome_crash_reporter_client",
       "//components/crash/content/browser",
--- a/chrome/browser/metrics/chrome_metrics_service_client.cc
+++ b/chrome/browser/metrics/chrome_metrics_service_client.cc
@@ -204,7 +204,7 @@
 #include "components/metrics/motherboard_metrics_provider.h"
 #endif
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID)
 #include "chrome/browser/metrics/chrome_metrics_service_crash_reporter.h"
 #endif
 
@@ -231,7 +231,7 @@
 // Needs to be kept in sync with the writer in
 // third_party/crashpad/crashpad/handler/handler_main.cc.
 const char kCrashpadHistogramAllocatorName[] = "CrashpadMetrics";
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID)
 ChromeMetricsServiceCrashReporter& GetCrashReporter() {
   static base::NoDestructor<ChromeMetricsServiceCrashReporter> crash_reporter;
   return *crash_reporter;
@@ -638,7 +638,7 @@
 void ChromeMetricsServiceClient::OnEnvironmentUpdate(std::string* environment) {
   // TODO(https://bugs.chromium.org/p/crashpad/issues/detail?id=135): call this
   // on Mac when the Crashpad API supports it.
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_ANDROID)
   // Register the environment with the crash reporter. Note that there is a
   // window from startup to this point during which crash reports will not have
   // an environment set.
