--- a/debian/submodules/ungoogled-chromium/utils/prune_binaries.py
+++ b/debian/submodules/ungoogled-chromium/utils/prune_binaries.py
@@ -135,7 +135,6 @@
             get_logger().debug('files that could not be pruned:\n%s',
                                '\n'.join(f for f in unremovable_files))
         get_logger().error('%d files could not be pruned:\n%s', len(unremovable_files), file_list)
-        sys.exit(1)
 
 
 def main():
