# Additional changes for disabling SafeBrowsing

--- a/content/browser/file_system_access/mock_native_file_system_permission_context.cc
+++ b/content/browser/file_system_access/mock_native_file_system_permission_context.cc
@@ -21,11 +21,4 @@
                                    callback);
 }
 
-void MockNativeFileSystemPermissionContext::PerformAfterWriteChecks(
-    std::unique_ptr<NativeFileSystemWriteItem> item,
-    GlobalFrameRoutingId frame_id,
-    base::OnceCallback<void(AfterWriteCheckResult)> callback) {
-  PerformAfterWriteChecks_(item.get(), frame_id, callback);
-}
-
 }  // namespace content
--- a/content/browser/file_system_access/mock_native_file_system_permission_context.h
+++ b/content/browser/file_system_access/mock_native_file_system_permission_context.h
@@ -44,15 +44,6 @@
            GlobalFrameRoutingId frame_id,
            base::OnceCallback<void(SensitiveDirectoryResult)>& callback));
 
-  void PerformAfterWriteChecks(
-      std::unique_ptr<NativeFileSystemWriteItem> item,
-      GlobalFrameRoutingId frame_id,
-      base::OnceCallback<void(AfterWriteCheckResult)> callback) override;
-  MOCK_METHOD3(PerformAfterWriteChecks_,
-               void(NativeFileSystemWriteItem* item,
-                    GlobalFrameRoutingId frame_id,
-                    base::OnceCallback<void(AfterWriteCheckResult)>& callback));
-
   MOCK_METHOD1(CanObtainReadPermission, bool(const url::Origin& origin));
   MOCK_METHOD1(CanObtainWritePermission, bool(const url::Origin& origin));
 };
