--- a/gpu/command_buffer/service/shared_image_interface_in_process.cc
+++ b/gpu/command_buffer/service/shared_image_interface_in_process.cc
@@ -166,7 +166,7 @@
             params->gpu_preferences, params->gpu_workarounds,
             params->gpu_feature_info, params->context_state,
             params->shared_image_manager,
-            params->context_state->memory_tracker(),
+            (params->context_state)?params->context_state->memory_tracker():NULL,
             params->is_for_display_compositor);
         return shared_image_factory;
       },
