--- a/chrome/browser/resources/settings/appearance_page/appearance_page.ts
+++ b/chrome/browser/resources/settings/appearance_page/appearance_page.ts
@@ -406,9 +406,8 @@
   }
 
   /** @return Whether to show the "USE GTK" button. */
-  private showUseGtk_(themeId: string): boolean {
-    return (!!themeId || this.systemTheme_ !== SystemTheme.GTK) &&
-        !this.appearanceBrowserProxy_.isChildAccount();
+  private showUseGtk_(): boolean {
+    return false;
   }
 
   /** @return Whether to show the "USE QT" button. */
