--- a/chrome/browser/resources/settings/appearance_page/appearance_page.ts
+++ b/chrome/browser/resources/settings/appearance_page/appearance_page.ts
@@ -489,9 +489,7 @@
 
   /** @return Whether to show the "USE QT" button. */
   protected showUseQt_(): boolean {
-    const themeId = this.themeIdPref_?.value;
-    return (!!themeId || this.systemTheme_ !== SystemTheme.QT) &&
-        !this.appearanceBrowserProxy_.isChildAccount();
+    return false;
   }
 
   /**
