--- a/chrome/browser/resources/settings/appearance_page/appearance_page.ts
+++ b/chrome/browser/resources/settings/appearance_page/appearance_page.ts
@@ -411,9 +411,8 @@
   }
 
   /** @return Whether to show the "USE QT" button. */
-  private showUseQt_(themeId: string): boolean {
-    return (!!themeId || this.systemTheme_ !== SystemTheme.QT) &&
-        !this.appearanceBrowserProxy_.isChildAccount();
+  private showUseQt_(): boolean {
+    return false;
   }
 
   /**
