--- a/net/filter/gzip_header.cc
+++ b/net/filter/gzip_header.cc
@@ -9,7 +9,7 @@
 #include <algorithm>
 
 #include "base/check_op.h"
-#include "third_party/zlib/zlib.h"
+#include <zlib.h>
 
 namespace net {
 
--- a/ui/gfx/codec/png_codec.cc
+++ b/ui/gfx/codec/png_codec.cc
@@ -17,7 +17,7 @@
 #include "third_party/skia/include/core/SkColorPriv.h"
 #include "third_party/skia/include/core/SkUnPreMultiply.h"
 #include "third_party/skia/include/encode/SkPngEncoder.h"
-#include "third_party/zlib/zlib.h"
+#include <zlib.h>
 #include "ui/gfx/codec/vector_wstream.h"
 #include "ui/gfx/geometry/size.h"
 
