--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -8536,7 +8536,6 @@
       "//chrome/browser/webauthn:test_support",
       "//chrome/browser/webauthn/proto",
       "//chrome/common/read_anything:mojo_bindings",
-      "//chrome/services/speech:unit_tests",
       "//components/app_constants",
       "//components/application_locale_storage",
       "//components/color",
@@ -8607,6 +8606,10 @@
       "//ui/webui/resources/js/browser_command:mojo_bindings",
     ]
 
+    if (enable_speech_service) {
+      deps += [ "//chrome/services/speech:unit_tests" ]
+    }
+
     if (enable_video_effects) {
       deps += [ "//services/video_effects/test:test_support" ]
     }
