--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -6829,7 +6829,6 @@
       "//chrome/browser/webauthn/proto",
       "//chrome/common/read_anything:mojo_bindings",
       "//chrome/services/readaloud:unit_tests",
-      "//chrome/services/speech:unit_tests",
       "//components/app_constants",
       "//components/application_locale_storage",
       "//components/color",
@@ -6908,6 +6907,11 @@
     if (include_js2gtest_tests && is_chromeos) {
       data += js2gtest_js_libraries
     }
+
+    if (enable_speech_service) {
+      deps += [ "//chrome/services/speech:unit_tests" ]
+    }
+
   }
 
   if (enable_webui_ntp) {
