--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -8599,7 +8599,6 @@
       "//chrome/browser/webauthn:test_support",
       "//chrome/browser/webauthn/proto",
       "//chrome/common/read_anything:mojo_bindings",
-      "//chrome/services/speech:unit_tests",
       "//components/app_constants",
       "//components/application_locale_storage",
       "//components/color",
@@ -8672,6 +8671,10 @@
       "//ui/webui/tracked_element:unit_tests",
     ]
 
+    if (enable_speech_service) {
+      deps += [ "//chrome/services/speech:unit_tests" ]
+    }
+
     if (enable_video_effects) {
       deps += [ "//services/video_effects/test:test_support" ]
     }
