--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -7613,7 +7613,6 @@
       "//chrome/browser/webauthn:test_support",
       "//chrome/browser/webauthn/proto",
       "//chrome/common/read_anything:mojo_bindings",
-      "//chrome/services/speech:unit_tests",
       "//components/app_constants",
       "//components/application_locale_storage",
       "//components/color",
@@ -7692,6 +7691,11 @@
     if (include_js2gtest_tests && is_chromeos) {
       data += js2gtest_js_libraries
     }
+
+    if (enable_speech_service) {
+      deps += [ "//chrome/services/speech:unit_tests" ]
+    }
+
   }
 
   if (enable_webui_ntp) {
