--- a/components/enterprise/BUILD.gn
+++ b/components/enterprise/BUILD.gn
@@ -110,6 +110,7 @@
     "//components/certificate_matching",
     "//components/device_signals/core/browser",
     "//components/device_signals/core/common:features",
+    "//components/enterprise/buildflags:buildflags",
     "//components/enterprise/common/proto:saas_usage_report_event",
     "//components/enterprise/connectors/core:constants",
     "//components/enterprise/device_attestation",
