--- a/build/config/clang/clang.gni
+++ b/build/config/clang/clang.gni
@@ -5,6 +5,7 @@
 import("//build/config/cast.gni")
 import("//build/config/cronet/config.gni")
 import("//build/config/gclient_args.gni")
+import("//build/config/sysroot.gni")
 import("//build/toolchain/cc_wrapper.gni")
 import("//build/toolchain/kythe.gni")
 import("//build/toolchain/rbe.gni")
@@ -43,6 +44,7 @@
 ]
 clang_modules_platform_supported =
     # s390x should work, but IBM doesn't want it enabled for that cpu arch.
+    use_sysroot &&
     (is_linux && current_cpu != "s390x") ||
     (is_android &&
      filter_include(android_supported_cpus, [ current_cpu ]) != []) ||
