--- a/chrome/test/chromedriver/BUILD.gn
+++ b/chrome/test/chromedriver/BUILD.gn
@@ -357,7 +357,8 @@
     "//base",
     "//build:branding_buildflags",
     "//build:chromeos_buildflags",
-    "//chrome/common",
+    "//chrome/common:constants",
+    "//chrome/common:version_header",
     "//chrome/test/chromedriver/constants:version_header",
     "//components/crx_file",
     "//components/embedder_support",
--- a/tools/v8_context_snapshot/BUILD.gn
+++ b/tools/v8_context_snapshot/BUILD.gn
@@ -83,7 +83,6 @@
       sources = [ "v8_context_snapshot_generator.cc" ]
 
       deps = [
-        "//chrome/common",
         "//gin:gin",
         "//mojo/core/embedder",
         "//services/service_manager/public/cpp",
