--- a/BUILD.gn
+++ b/BUILD.gn
@@ -505,7 +505,8 @@
     }
 
     if ((is_win || is_mac || is_linux || is_chromeos || is_fuchsia) &&
-        (target_cpu == "x86" || target_cpu == "x64")) {
+        (target_cpu == "x86" || target_cpu == "x64") &&
+         enable_swiftshader_vulkan) {
       deps += [
         "//third_party/swiftshader/src/Vulkan:icd_file",
         "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan",
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -21,6 +21,7 @@
 import("//chrome/process_version_rc_template.gni")
 import("//components/optimization_guide/features.gni")
 import("//extensions/buildflags/buildflags.gni")
+import("//gpu/vulkan/features.gni")
 import("//media/media_options.gni")
 import("//services/on_device_model/on_device_model.gni")
 import("//services/shape_detection/features.gni")
@@ -1677,7 +1678,7 @@
       deps = [
         ":chrome_symbols",
       ]
-      if (is_linux) {
+      if (is_linux && enable_swiftshader_vulkan) {
         deps += [ ":swiftshader_vk_symbols" ]
       }
       if (!is_chromeos && angle_shared_libvulkan) {
@@ -1723,17 +1724,19 @@
         deps = [ "//components/crash/core/app:chrome_crashpad_handler" ]
       }
     }
-    extract_symbols("swiftshader_vk_symbols") {
-      binary = "$root_out_dir/libvk_swiftshader.so"
+    if (enable_swiftshader_vulkan) {
+      extract_symbols("swiftshader_vk_symbols") {
+        binary = "$root_out_dir/libvk_swiftshader.so"
 
-      if (current_cpu == "x86") {
-        # GYP used "ia32" so keep that naming for back-compat.
-        symbol_file = "$root_out_dir/libvk_swiftshader.breakpad.ia32"
-      } else {
-        symbol_file = "$root_out_dir/libvk_swiftshader.breakpad.$current_cpu"
-      }
+        if (current_cpu == "x86") {
+          # GYP used "ia32" so keep that naming for back-compat.
+          symbol_file = "$root_out_dir/libvk_swiftshader.breakpad.ia32"
+        } else {
+          symbol_file = "$root_out_dir/libvk_swiftshader.breakpad.$current_cpu"
+        }
 
-      deps = [ "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan" ]
+        deps = [ "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan" ]
+      }
     }
     extract_symbols("angle_egl_symbols") {
       binary = "$root_out_dir/libEGL.so"
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -82,7 +82,7 @@
   packaging_files_shlibs += [ "$root_out_dir/libvulkan.so.1" ]
 }
 
-if (enable_swiftshader) {
+if (enable_swiftshader && enable_swiftshader_vulkan) {
   packaging_files_shlibs += [ "$root_out_dir/libvk_swiftshader.so" ]
 }
 
@@ -106,7 +106,7 @@
                     "$root_out_dir/PrivacySandboxAttestationsPreloaded/manifest.json",
                     "$root_out_dir/PrivacySandboxAttestationsPreloaded/privacy-sandbox-attestations.dat",
                   ]
-if (enable_swiftshader) {
+if (enable_swiftshader && enable_swiftshader_vulkan) {
   packaging_files += [ "$root_out_dir/vk_swiftshader_icd.json" ]
 }
 
--- a/gpu/vulkan/features.gni
+++ b/gpu/vulkan/features.gni
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//build/config/ui.gni")
+import("//ui/gl/features.gni")
 
 # Features used by targets inside and outside of |gpu/vulkan|.
 # For details see declare_args() in build/config/BUILDCONFIG.gn.
@@ -13,5 +14,5 @@
   # Enable swiftshader vulkan. Disabling it can save build time, however
   # --use-vulkan=swiftshader and some tests which use swiftshader vulkan will
   # not work.
-  enable_swiftshader_vulkan = true
+  enable_swiftshader_vulkan = enable_swiftshader
 }
--- a/third_party/angle/gni/angle.gni
+++ b/third_party/angle/gni/angle.gni
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//build_overrides/angle.gni")
+import("//ui/gl/features.gni")
 if (angle_has_build) {
   import("//build/config/gclient_args.gni")
 }
@@ -330,7 +331,8 @@
 declare_args() {
   # Currently SwiftShader's Vulkan front-end doesn't build on Android.
   # Disabled by default on MSVC because of compiler warnings/errors.
-  angle_enable_swiftshader = angle_enable_vulkan && !is_android && is_clang
+  angle_enable_swiftshader =
+    angle_enable_vulkan && enable_swiftshader && !is_android && is_clang
 
   # Translator backend support:
   angle_enable_hlsl = angle_enable_d3d9 || angle_enable_d3d11
--- a/third_party/dawn/scripts/dawn_features.gni
+++ b/third_party/dawn/scripts/dawn_features.gni
@@ -27,6 +27,7 @@
 
 import("//build/config/clang/clang.gni")
 import("//build_overrides/build.gni")
+import("//ui/gl/features.gni")
 import("dawn_overrides_with_defaults.gni")
 
 declare_args() {
@@ -68,7 +69,7 @@
   # Enables SwiftShader as the fallback adapter. Requires dawn_swiftshader_dir
   # to be set to take effect.
   # TODO(dawn:1536): Enable SwiftShader for Android.
-  dawn_use_swiftshader = !is_android && !is_ios
+  dawn_use_swiftshader = !is_android && !is_ios && enable_swiftshader
 
   # Whether we allow building DXC.
   # DXC requires SM6.0+ which is blocklisted on x86.
