--- a/third_party/webrtc/webrtc.gni
+++ b/third_party/webrtc/webrtc.gni
@@ -549,10 +549,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -666,10 +666,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -758,10 +758,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -900,10 +900,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -958,10 +958,10 @@
     # executables, but not libraries to use absl flags.
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl_full" ]
