--- a/third_party/webrtc/webrtc.gni
+++ b/third_party/webrtc/webrtc.gni
@@ -561,10 +561,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -678,10 +678,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -770,10 +770,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
@@ -912,10 +912,10 @@
     # Abseil dependencies need to be converted to //third_party/abseil-cpp:absl when build_with_chromium=true
     if (build_with_chromium && defined(deps)) {
       absl_dependencies =
-          filter_labels_include(deps, [ "//third_party/abseil-cpp/*" ])
+          filter_include(deps, [ "//third_party/abseil-cpp/*" ])
       if (absl_dependencies != []) {
         filtered_deps =
-            filter_labels_exclude(deps, [ "//third_party/abseil-cpp/*" ])
+            filter_exclude(deps, [ "//third_party/abseil-cpp/*" ])
         deps = []
         deps = filtered_deps
         deps += [ "//third_party/abseil-cpp:absl" ]
