--- a/chrome/common/chrome_paths.cc
+++ b/chrome/common/chrome_paths.cc
@@ -382,6 +382,17 @@
 
 #if BUILDFLAG(ENABLE_WIDEVINE)
     case chrome::DIR_BUNDLED_WIDEVINE_CDM:
+      // A modified version of the Fedora patch
+      // Prefer /usr/lib/WidevineCdm package path with $HOME/.local/lib as a fallback
+      if (base::PathExists(base::FilePath(FILE_PATH_LITERAL("/usr/lib/WidevineCdm/_platform_specific/linux_x64/libwidevinecdm.so")))) {
+        cur = base::FilePath(FILE_PATH_LITERAL("/usr/lib/WidevineCdm"));
+        break;
+      }
+      base::PathService::Get(base::DIR_HOME, &cur);
+      cur = cur.Append(FILE_PATH_LITERAL(".local/lib/libwidevinecdm.so"));
+      if (base::PathExists(cur)) {
+        break;
+      }
       if (!GetComponentDirectory(&cur)) {
         return false;
       }
--- a/third_party/widevine/cdm/BUILD.gn
+++ b/third_party/widevine/cdm/BUILD.gn
@@ -22,7 +22,7 @@
 
   flags = [
     "ENABLE_WIDEVINE=$enable_widevine",
-    "BUNDLE_WIDEVINE_CDM=$bundle_widevine_cdm",
+    "BUNDLE_WIDEVINE_CDM=true",
     "ENABLE_WIDEVINE_CDM_COMPONENT=$enable_widevine_cdm_component",
     "ENABLE_MEDIA_FOUNDATION_WIDEVINE_CDM=$enable_media_foundation_widevine_cdm",
   ]
