--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2174,6 +2174,17 @@
       cflags_cc += [ "-Wunique-object-duplication" ]
     }
 
+    if (is_linux) {
+      # Excessive warnings
+      cflags += [
+        "-Wno-deprecated-builtins",
+        "-Wno-deprecated-copy",
+        "-Wno-include-angled-in-module-purview",
+        "-Wno-unnecessary-virtual-specifier",
+        "-Wno-vla-cxx-extension",
+      ]
+    }
+
     if (is_wasm) {
       # Abseil's `symbolize_emscripten.inc` file currently has macros that are
       # not compatible with C++98, so turn off the warnings.
