--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -815,6 +815,10 @@
         # TODO(crbug.com/40219076): Investigate if we can get the > 6% perf win
         # of import_instr_limit 30 with a binary size hit smaller than ~2 MiB.
         import_instr_limit = 5
+      } else if (is_linux) {
+        # Raise the import-hot-multiplier from the default of 10 to 14 to give
+        # a hot limit of 420, analagous to an import limit of 42 (42x10 = 420).
+        ldflags += [ "-Wl,-mllvm,-import-hot-multiplier=14" ]
       }
 
       ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
