--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -762,6 +762,11 @@
         # TODO(crbug.com/1308318): Investigate if we can get the > 6% perf win
         # of import_instr_limit 30 with a binary size hit smaller than ~2 MiB.
         import_instr_limit = 5
+      } else if (is_linux) {
+        # Use a slightly lower import_instr_limit value of 24 for linux and
+        # raise the import-hot-multiplier from the default of 10 to 15.
+        import_instr_limit = 24
+        ldflags += [ "-Wl,-mllvm,-import-hot-multiplier=15" ]
       }
 
       ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
