--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -794,6 +794,10 @@
         # TODO(crbug.com/1308318): Investigate if we can get the > 6% perf win
         # of import_instr_limit 30 with a binary size hit smaller than ~2 MiB.
         import_instr_limit = 5
+      } else if (is_linux) {
+        # Raise the import-hot-multiplier from the default of 10 to 15 to give
+        # a hot limit of 450, analagous to an import limit of 45 (45x10 = 450).
+        ldflags += [ "-Wl,-mllvm,-import-hot-multiplier=15" ]
       }
 
       ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
