--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -152,7 +152,7 @@
 
   # Zero init has favorable performance/size tradeoffs for Chrome OS
   # but was not evaluated for other platforms.
-  init_stack_vars_zero = is_chromeos
+  init_stack_vars_zero = (is_chromeos || is_linux)
 
   # This argument is to control whether enabling text section splitting in the
   # final binary. When enabled, the separated text sections with prefix
