--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2636,6 +2636,7 @@
 # [1]: https://pinpoint-dot-chromeperf.appspot.com/job/132bc772be0000
 # [2]: https://crrev.com/c/5447532
 config("optimize") {
+  ldflags = common_optimize_on_ldflags
   if (is_win) {
     # clang-cl's /O2 corresponds to clang's -O3, and really want -O2 for
     # consistency with the other platforms.
@@ -2675,12 +2676,15 @@
   } else {
     cflags = [ "-O2" ] + common_optimize_on_cflags
 
+    if (use_lld) {
+      ldflags += [ "-Wl,--lto-CGO2" ]
+    }
+
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
     # to use.
     rustflags = [ "-Copt-level=3" ]
   }
-  ldflags = common_optimize_on_ldflags
 }
 
 # Turn off optimizations.
@@ -2740,6 +2744,9 @@
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O2" ] + common_optimize_on_cflags
+      if (use_lld) {
+        ldflags += [ "-Wl,--lto-CGO2" ]
+      }
     }
     rustflags = [ "-Copt-level=3" ]
   }
@@ -2776,6 +2783,9 @@
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
       cflags = [ "-O3" ] + common_optimize_on_cflags
+      if (use_lld) {
+        ldflags += [ "-Wl,--lto-CGO3" ]
+      }
     }
     rustflags = [ "-Copt-level=3" ]
   }
