--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2334,11 +2334,6 @@
       "-ffunction-sections",
       "-fdebug-types-section",
     ]
-    if ((!is_nacl || is_nacl_saigo) && is_clang) {
-      # We don't care about unique section names, this makes object files a bit
-      # smaller.
-      common_optimize_on_cflags += [ "-fno-unique-section-names" ]
-    }
 
     common_optimize_on_ldflags += [
       # Specifically tell the linker to perform optimizations.
