--- a/net/BUILD.gn
+++ b/net/BUILD.gn
@@ -1713,6 +1713,7 @@
       "filter/brotli_source_stream.cc",
       "filter/brotli_source_stream.h",
     ]
+    include_dirs = [ "//third_party/brotli/include" ]
   } else {
     sources += [ "filter/brotli_source_stream_disabled.cc" ]
   }
--- a/ui/base/BUILD.gn
+++ b/ui/base/BUILD.gn
@@ -225,6 +225,7 @@
   ]
 
   libs = []
+  include_dirs = [ "//third_party/brotli/include" ]
 
   if (is_android) {
     sources += [
