--- a/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
+++ b/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
@@ -22,6 +22,7 @@
 #include <unistd.h>
 
 #include "base/allocator/partition_alloc_features.h"
+#include "base/allocator/partition_allocator/src/partition_alloc/buildflags.h"
 #include "base/feature_list.h"
 #include "base/features.h"
 #include "base/notreached.h"
@@ -356,11 +357,16 @@
       .Cases({FUTEX_LOCK_PI, FUTEX_UNLOCK_PI, FUTEX_TRYLOCK_PI,
               FUTEX_WAIT_REQUEUE_PI, FUTEX_CMP_REQUEUE_PI, FUTEX_LOCK_PI2},
              (base::KernelSupportsPriorityInheritanceFutex() &&
+#if PA_BUILDFLAG(ENABLE_PARTITION_LOCK_PRIORITY_INHERITANCE)
                       (base::FeatureList::IsEnabled(
                            base::features::kUsePriorityInheritanceMutex) ||
                        base::FeatureList::IsEnabled(
                            base::features::
                                kPartitionAllocUsePriorityInheritanceLocks))
+#else
+                      (base::FeatureList::IsEnabled(
+                           base::features::kUsePriorityInheritanceMutex))
+#endif  // PA_BUILDFLAG(ENABLE_PARTITION_LOCK_PRIORITY_INHERITANCE)
                   ? Allow()
                   : error))
 #endif  // BUILDFLAG(ENABLE_MUTEX_PRIORITY_INHERITANCE)
