--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -165,7 +165,7 @@
 #
 # TODO(https://crbug.com/1482525): Re-enable ThinLTO for Rust on LaCrOS
 # TODO(b/300937673): Re-enable ThinLTO for Rust on ash-chrome
-toolchain_supports_rust_thin_lto = !is_chromeos
+toolchain_supports_rust_thin_lto = !is_chromeos && !is_linux
 
 # We want to store rust_sysroot as a source-relative variable for ninja
 # portability. In practice if an external toolchain was specified, it might
