--- a/base/i18n/message_formatter.cc
+++ b/base/i18n/message_formatter.cc
@@ -8,10 +8,10 @@
 #include "base/logging.h"
 #include "base/numerics/safe_conversions.h"
 #include "base/time/time.h"
-#include "third_party/icu/source/common/unicode/unistr.h"
-#include "third_party/icu/source/common/unicode/utypes.h"
-#include "third_party/icu/source/i18n/unicode/fmtable.h"
-#include "third_party/icu/source/i18n/unicode/msgfmt.h"
+#include <unicode/unistr.h>
+#include <unicode/utypes.h>
+#include <unicode/fmtable.h>
+#include <unicode/msgfmt.h>
 
 using icu::UnicodeString;
 
--- a/base/i18n/number_formatting.cc
+++ b/base/i18n/number_formatting.cc
@@ -16,8 +16,8 @@
 #include "base/strings/string_util.h"
 #include "base/strings/stringprintf.h"
 #include "base/strings/utf_string_conversions.h"
-#include "third_party/icu/source/common/unicode/ustring.h"
-#include "third_party/icu/source/i18n/unicode/numfmt.h"
+#include <unicode/ustring.h>
+#include <unicode/numfmt.h>
 
 namespace base {
 
--- a/base/i18n/time_formatting.cc
+++ b/base/i18n/time_formatting.cc
@@ -15,13 +15,13 @@
 #include "base/strings/utf_string_conversions.h"
 #include "base/time/time.h"
 #include "build/chromeos_buildflags.h"
-#include "third_party/icu/source/common/unicode/utypes.h"
-#include "third_party/icu/source/i18n/unicode/datefmt.h"
-#include "third_party/icu/source/i18n/unicode/dtitvfmt.h"
-#include "third_party/icu/source/i18n/unicode/dtptngen.h"
-#include "third_party/icu/source/i18n/unicode/fmtable.h"
-#include "third_party/icu/source/i18n/unicode/measfmt.h"
-#include "third_party/icu/source/i18n/unicode/smpdtfmt.h"
+#include <unicode/utypes.h>
+#include <unicode/datefmt.h>
+#include <unicode/dtitvfmt.h>
+#include <unicode/dtptngen.h>
+#include <unicode/fmtable.h>
+#include <unicode/measfmt.h>
+#include <unicode/smpdtfmt.h>
 
 namespace base {
 namespace {
--- a/chrome/browser/ui/webui/downloads/downloads_list_tracker.cc
+++ b/chrome/browser/ui/webui/downloads/downloads_list_tracker.cc
@@ -33,7 +33,7 @@
 #include "mojo/public/cpp/bindings/pending_remote.h"
 #include "mojo/public/cpp/bindings/remote.h"
 #include "net/base/filename_util.h"
-#include "third_party/icu/source/i18n/unicode/datefmt.h"
+#include <unicode/datefmt.h>
 #include "ui/base/l10n/time_format.h"
 
 using content::BrowserContext;
--- a/components/payments/core/currency_formatter.cc
+++ b/components/payments/core/currency_formatter.cc
@@ -10,9 +10,9 @@
 #include "base/numerics/safe_conversions.h"
 #include "base/strings/string_util.h"
 #include "base/strings/utf_string_conversions.h"
-#include "third_party/icu/source/common/unicode/stringpiece.h"
-#include "third_party/icu/source/common/unicode/uchar.h"
-#include "third_party/icu/source/common/unicode/utypes.h"
+#include <unicode/stringpiece.h>
+#include <unicode/uchar.h>
+#include <unicode/utypes.h>
 
 namespace payments {
 
--- a/components/payments/core/currency_formatter.h
+++ b/components/payments/core/currency_formatter.h
@@ -9,9 +9,9 @@
 #include <string>
 
 #include "base/macros.h"
-#include "third_party/icu/source/common/unicode/locid.h"
-#include "third_party/icu/source/common/unicode/unistr.h"
-#include "third_party/icu/source/i18n/unicode/numfmt.h"
+#include <unicode/locid.h>
+#include <unicode/unistr.h>
+#include <unicode/numfmt.h>
 
 namespace payments {
 
--- a/components/spellcheck/renderer/spellcheck_worditerator.cc
+++ b/components/spellcheck/renderer/spellcheck_worditerator.cc
@@ -17,10 +17,10 @@
 #include "base/strings/stringprintf.h"
 #include "base/strings/utf_string_conversions.h"
 #include "components/spellcheck/renderer/spellcheck.h"
-#include "third_party/icu/source/common/unicode/normlzr.h"
-#include "third_party/icu/source/common/unicode/schriter.h"
-#include "third_party/icu/source/common/unicode/uscript.h"
-#include "third_party/icu/source/i18n/unicode/ulocdata.h"
+#include <unicode/normlzr.h>
+#include <unicode/schriter.h>
+#include <unicode/uscript.h>
+#include <unicode/ulocdata.h>
 
 // SpellcheckCharAttribute implementation:
 
--- a/components/sync/base/time.cc
+++ b/components/sync/base/time.cc
@@ -8,8 +8,8 @@
 
 #include "base/i18n/unicodestring.h"
 #include "base/strings/utf_string_conversions.h"
-#include "third_party/icu/source/common/unicode/utypes.h"
-#include "third_party/icu/source/i18n/unicode/smpdtfmt.h"
+#include <unicode/utypes.h>
+#include <unicode/smpdtfmt.h>
 
 namespace syncer {
 
--- a/components/url_formatter/spoof_checks/idn_spoof_checker.cc
+++ b/components/url_formatter/spoof_checks/idn_spoof_checker.cc
@@ -16,11 +16,11 @@
 #include "base/threading/thread_local_storage.h"
 #include "build/build_config.h"
 #include "net/base/lookup_string_in_fixed_set.h"
-#include "third_party/icu/source/common/unicode/schriter.h"
-#include "third_party/icu/source/common/unicode/unistr.h"
-#include "third_party/icu/source/i18n/unicode/regex.h"
-#include "third_party/icu/source/i18n/unicode/translit.h"
-#include "third_party/icu/source/i18n/unicode/uspoof.h"
+#include <unicode/schriter.h>
+#include <unicode/unistr.h>
+#include <unicode/regex.h>
+#include <unicode/translit.h>
+#include <unicode/uspoof.h>
 
 namespace url_formatter {
 
--- a/third_party/blink/renderer/platform/wtf/date_math.cc
+++ b/third_party/blink/renderer/platform/wtf/date_math.cc
@@ -89,6 +89,7 @@
 
 #include <unicode/basictz.h>
 #include <unicode/timezone.h>
+#include <unicode/ucal.h>
 
 #if defined(OS_WIN)
 #include <windows.h>
--- a/third_party/tensorflow-text/src/tensorflow_text/core/kernels/wordpiece_tokenizer.cc
+++ b/third_party/tensorflow-text/src/tensorflow_text/core/kernels/wordpiece_tokenizer.cc
@@ -17,9 +17,9 @@
 #include "absl/strings/str_cat.h"
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
-#include "third_party/icu/source/common/unicode/schriter.h"
-#include "third_party/icu/source/common/unicode/unistr.h"
-#include "third_party/icu/source/common/unicode/utf8.h"
+#include <unicode/schriter.h>
+#include <unicode/unistr.h>
+#include <unicode/utf8.h>
 
 namespace tensorflow {
 namespace text {
--- a/ui/base/l10n/formatter.h
+++ b/ui/base/l10n/formatter.h
@@ -13,9 +13,9 @@
 #include "base/component_export.h"
 #include "base/lazy_instance.h"
 #include "base/macros.h"
-#include "third_party/icu/source/common/unicode/unistr.h"
-#include "third_party/icu/source/i18n/unicode/msgfmt.h"
-#include "third_party/icu/source/i18n/unicode/plurrule.h"
+#include <unicode/unistr.h>
+#include <unicode/msgfmt.h>
+#include <unicode/plurrule.h>
 #include "ui/base/l10n/time_format.h"
 
 namespace ui {
--- a/ui/base/l10n/l10n_util.cc
+++ b/ui/base/l10n/l10n_util.cc
@@ -31,8 +31,8 @@
 #include "base/strings/utf_string_conversions.h"
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
-#include "third_party/icu/source/common/unicode/rbbi.h"
-#include "third_party/icu/source/common/unicode/uloc.h"
+#include <unicode/rbbi.h>
+#include <unicode/uloc.h>
 #include "ui/base/l10n/l10n_util_collator.h"
 #include "ui/base/resource/resource_bundle.h"
 #include "ui/base/ui_base_paths.h"
--- a/ui/base/l10n/time_format.cc
+++ b/ui/base/l10n/time_format.cc
@@ -13,7 +13,7 @@
 #include "base/numerics/safe_conversions.h"
 #include "base/strings/string_util.h"
 #include "base/time/time.h"
-#include "third_party/icu/source/common/unicode/unistr.h"
+#include <unicode/unistr.h>
 #include "ui/base/l10n/formatter.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/strings/grit/ui_strings.h"
--- a/ui/gfx/render_text.cc
+++ b/ui/gfx/render_text.cc
@@ -22,9 +22,9 @@
 #include "build/build_config.h"
 #include "cc/paint/paint_canvas.h"
 #include "cc/paint/paint_shader.h"
-#include "third_party/icu/source/common/unicode/rbbi.h"
-#include "third_party/icu/source/common/unicode/uchar.h"
-#include "third_party/icu/source/common/unicode/utf16.h"
+#include <unicode/rbbi.h>
+#include <unicode/uchar.h>
+#include <unicode/utf16.h>
 #include "third_party/skia/include/core/SkDrawLooper.h"
 #include "third_party/skia/include/core/SkFontStyle.h"
 #include "third_party/skia/include/core/SkTextBlob.h"
--- a/ui/gfx/text_elider.cc
+++ b/ui/gfx/text_elider.cc
@@ -28,9 +28,9 @@
 #include "base/strings/sys_string_conversions.h"
 #include "base/strings/utf_string_conversions.h"
 #include "build/build_config.h"
-#include "third_party/icu/source/common/unicode/rbbi.h"
-#include "third_party/icu/source/common/unicode/uloc.h"
-#include "third_party/icu/source/common/unicode/umachine.h"
+#include <unicode/rbbi.h>
+#include <unicode/uloc.h>
+#include <unicode/umachine.h>
 #include "ui/gfx/font_list.h"
 #include "ui/gfx/geometry/rect_conversions.h"
 #include "ui/gfx/render_text.h"
