--- a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config_components.h
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config_components.h
@@ -354,7 +354,7 @@
 #define CONFIG_ATRAC3P_DECODER 0
 #define CONFIG_ATRAC3PAL_DECODER 0
 #define CONFIG_ATRAC9_DECODER 0
-#define CONFIG_ALAC_DECODER 0
+#define CONFIG_ALAC_DECODER 1
 #define CONFIG_BINKAUDIO_DCT_DECODER 0
 #define CONFIG_BINKAUDIO_RDFT_DECODER 0
 #define CONFIG_BMV_AUDIO_DECODER 0
--- a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/libavcodec/codec_list.c
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/libavcodec/codec_list.c
@@ -4,6 +4,7 @@
     &ff_flac_decoder,
     &ff_mp3_decoder,
     &ff_vorbis_decoder,
+    &ff_alac_decoder,
     &ff_pcm_alaw_decoder,
     &ff_pcm_f32le_decoder,
     &ff_pcm_mulaw_decoder,
--- a/third_party/ffmpeg/ffmpeg_generated.gni
+++ b/third_party/ffmpeg/ffmpeg_generated.gni
@@ -272,6 +272,20 @@
 
 if ((current_cpu == "arm64" && ffmpeg_branding == "Chrome") ||
     (current_cpu == "x64" && ffmpeg_branding == "Chrome") ||
+    (is_android && current_cpu == "x86" && ffmpeg_branding == "Chrome") ||
+    (is_apple && ffmpeg_branding == "Chrome") ||
+    (is_win && ffmpeg_branding == "Chrome") ||
+    (use_linux_config && ffmpeg_branding == "Chrome")) {
+  ffmpeg_c_sources += [
+    "libavcodec/alac.c",
+    "libavcodec/alac_data.c",
+    "libavcodec/alacdsp.c",
+    "libavcodec/alacenc.c",
+  ]
+}
+
+if ((current_cpu == "arm64" && ffmpeg_branding == "Chrome") ||
+    (current_cpu == "x64" && ffmpeg_branding == "Chrome") ||
     (is_android && current_cpu == "arm" && arm_use_neon &&
      ffmpeg_branding == "Chrome") ||
     (is_android && current_cpu == "x86" && ffmpeg_branding == "Chrome") ||
@@ -465,6 +479,14 @@
   ]
 }
 
+if ((current_cpu == "x64" && ffmpeg_branding == "Chrome") ||
+    (is_android && current_cpu == "x86" && ffmpeg_branding == "Chrome") ||
+    (is_win && current_cpu == "x86" && ffmpeg_branding == "Chrome") ||
+    (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "Chrome")) {
+  ffmpeg_c_sources += [ "libavcodec/x86/alacdsp_init.c" ]
+  ffmpeg_asm_sources += [ "libavcodec/x86/alacdsp.asm" ]
+}
+
 if (use_linux_config && current_cpu == "arm" && arm_use_neon &&
     ffmpeg_branding == "Chrome") {
   ffmpeg_gas_sources += [
