--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2531,11 +2531,14 @@
       "-mllvm",
       "-polly",
       "-mllvm",
+      "-polly-vectorizer=stripmine",
+      "-mllvm",
       "-polly-run-dce",
     ]
     if (use_lld && use_thin_lto) {
       extra_ldflags = [
         "-Wl,-mllvm,-polly",
+        "-Wl,-mllvm,-polly-vectorizer=stripmine",
         "-Wl,-mllvm,-polly-run-dce",
       ]
     }
