--- a/third_party/webrtc/modules/video_coding/codecs/av1/dav1d_decoder.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/av1/dav1d_decoder.cc
@@ -26,10 +26,11 @@
 #include "common_video/include/video_frame_buffer.h"
 #include "modules/video_coding/include/video_error_codes.h"
 #include "rtc_base/logging.h"
-#include "third_party/dav1d/libdav1d/include/dav1d/data.h"
-#include "third_party/dav1d/libdav1d/include/dav1d/dav1d.h"
-#include "third_party/dav1d/libdav1d/include/dav1d/headers.h"
-#include "third_party/dav1d/libdav1d/include/dav1d/picture.h"
+
+#include <dav1d/data.h>
+#include <dav1d/dav1d.h>
+#include <dav1d/headers.h>
+#include <dav1d/picture.h>
 
 namespace webrtc {
 namespace {
--- a/media/filters/dav1d_video_decoder.cc
+++ b/media/filters/dav1d_video_decoder.cc
@@ -25,7 +25,7 @@
 #include "third_party/skia/include/core/SkData.h"
 
 extern "C" {
-#include "third_party/dav1d/libdav1d/include/dav1d/dav1d.h"
+#include <dav1d/dav1d.h>
 }
 
 namespace media {
