--- a/device/bluetooth/BUILD.gn
+++ b/device/bluetooth/BUILD.gn
@@ -601,12 +601,6 @@
           "chromeos_platform_features.h",
         ]
       }
-      if (use_dbus) {
-        public_deps += [
-          "//components/dbus",
-          "//dbus",
-        ]
-      }
     } else {  # !use_bluez
       defines += [ "LINUX_WITHOUT_DBUS" ]
       sources += [
@@ -615,6 +609,13 @@
       ]
     }
 
+    if (use_dbus) {
+      public_deps += [
+        "//components/dbus",
+        "//dbus",
+      ]
+    }
+
     if (!use_bluez) {
       defines += [ "NO_PLATFORM_BLUETOOTH" ]
       sources += [ "bluetooth_adapter_stub.cc" ]
