--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1843,16 +1843,11 @@
 
   # Copies some scripts and resources that are used for desktop integration.
   copy("xdg_mime") {
-    sources = [
-      "//chrome/tools/build/linux/chrome-wrapper",
-      "//third_party/xdg-utils/scripts/xdg-mime",
-      "//third_party/xdg-utils/scripts/xdg-settings",
-    ]
     if (is_linux) {
-      sources +=
+      sources =
           [ "//chrome/app/theme/$branding_path_component/product_logo_48.png" ]
     } else {
-      sources += [
+      sources = [
         "//chrome/app/theme/$branding_path_component/linux/product_logo_48.png",
       ]
     }
