--- a/components/autofill/core/common/autofill_features.cc
+++ b/components/autofill/core/common/autofill_features.cc
@@ -947,7 +947,7 @@
 // i.e., https://other.autofill.server:port/tbproxy/af/
 BASE_FEATURE(kAutofillServerCommunication,
              "AutofillServerCommunication",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enables showing DOM Node ID of elements.
 BASE_FEATURE(kShowDomNodeIDs,
--- a/components/autofill/core/common/autofill_prefs.cc
+++ b/components/autofill/core/common/autofill_prefs.cc
@@ -41,7 +41,7 @@
       kAutofillCreditCardEnabled, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
-      kAutofillPaymentCvcStorage, true,
+      kAutofillPaymentCvcStorage, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
       kAutofillPaymentCardBenefits, true,
@@ -81,7 +81,7 @@
 #endif
 
   // Deprecated prefs registered for migration.
-  registry->RegisterBooleanPref(kAutofillEnabledDeprecated, true);
+  registry->RegisterBooleanPref(kAutofillEnabledDeprecated, false);
   registry->RegisterStringPref(kAutofillAblationSeedPref, "");
   registry->RegisterBooleanPref(kAutofillRanQuasiDuplicateExtraDeduplication,
                                 false);
