--- a/components/component_updater/component_updater_service.cc
+++ b/components/component_updater/component_updater_service.cc
@@ -583,8 +583,7 @@
   // If the preference is not set the component updates are enabled by default
   // unless in Chrome for Testing where we never want components to be updated
   // automatically.
-  static constexpr bool kComponentUpdatesEnabledByDefault =
-      !BUILDFLAG(CHROME_FOR_TESTING);
+  static constexpr bool kComponentUpdatesEnabledByDefault = false;
 
   registry->RegisterBooleanPref(prefs::kComponentUpdatesEnabled,
                                 kComponentUpdatesEnabledByDefault);
