--- a/components/component_updater/component_updater_service.cc
+++ b/components/component_updater/component_updater_service.cc
@@ -549,8 +549,7 @@
   // If the preference is not set the component updates are enabled by default
   // unless in Chrome for Testing where we never want components to be updated
   // automatically.
-  constexpr bool kComponentUpdatesEnabledByDefault =
-      !BUILDFLAG(CHROME_FOR_TESTING);
+  constexpr bool kComponentUpdatesEnabledByDefault = false;
 
   registry->RegisterBooleanPref(prefs::kComponentUpdatesEnabled,
                                 kComponentUpdatesEnabledByDefault);
