--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -900,7 +900,7 @@
 // https://www.chromestatus.com/feature/5745285984681984
 BASE_FEATURE(kSignedHTTPExchange,
              "SignedHTTPExchange",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // If enabled, GetUserMedia API will only work when the concerned tab is in
 // focus
