--- a/chrome/browser/ui/startup/BUILD.gn
+++ b/chrome/browser/ui/startup/BUILD.gn
@@ -42,7 +42,6 @@
   if (!is_android) {
     sources += [
       "automation_infobar_delegate.h",
-      "google_api_keys_infobar_delegate.h",
       "infobar_utils.h",
       "launch_mode_recorder.h",
       "obsolete_system_infobar_delegate.h",
@@ -155,7 +154,6 @@
   if (!is_android) {
     sources += [
       "automation_infobar_delegate.cc",
-      "google_api_keys_infobar_delegate.cc",
       "infobar_utils.cc",
       "launch_mode_recorder.cc",
       "obsolete_system_infobar_delegate.cc",
