--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -1411,8 +1411,6 @@
       "singleton_tabs.h",
       "startup/automation_infobar_delegate.cc",
       "startup/automation_infobar_delegate.h",
-      "startup/google_api_keys_infobar_delegate.cc",
-      "startup/google_api_keys_infobar_delegate.h",
       "startup/infobar_utils.cc",
       "startup/infobar_utils.h",
       "startup/launch_mode_recorder.cc",
