--- a/third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.cc
+++ b/third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.cc
@@ -107,13 +107,15 @@
 };
 
 WebRTCIPHandlingPolicy GetWebRTCIPHandlingPolicy(const String& preference) {
+  if (preference == kWebRTCIPHandlingDefault)
+    return kDefault;
   if (preference == kWebRTCIPHandlingDefaultPublicAndPrivateInterfaces)
     return kDefaultPublicAndPrivateInterfaces;
   if (preference == kWebRTCIPHandlingDefaultPublicInterfaceOnly)
     return kDefaultPublicInterfaceOnly;
   if (preference == kWebRTCIPHandlingDisableNonProxiedUdp)
     return kDisableNonProxiedUdp;
-  return kDefault;
+  return kDisableNonProxiedUdp;
 }
 
 bool IsValidPortRange(uint16_t min_port, uint16_t max_port) {
