--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -216,7 +216,7 @@
 // swap BrowsingContextGroups(BrowsingInstances).
 BASE_FEATURE(kClearCrossSiteCrossBrowsingContextGroupWindowName,
              "ClearCrossSiteCrossBrowsingContextGroupWindowName",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kCompositeBGColorAnimation,
              "CompositeBGColorAnimation",
--- a/third_party/blink/renderer/core/loader/document_loader.cc
+++ b/third_party/blink/renderer/core/loader/document_loader.cc
@@ -2883,11 +2883,10 @@
       !frame_->DomWindow()->GetSecurityOrigin()->IsSameOriginWith(
           previous_window->GetSecurityOrigin());
   if (should_clear_window_name) {
-    // TODO(andypaicu): experimentalSetNullName will just record the fact
-    // that the name would be nulled and if the name is accessed after we will
-    // fire a UseCounter. If we decide to move forward with this change, we'd
-    // actually clean the name here.
-    // frame_->tree().setName(g_null_atom);
+    // TODO(andypaicu): experimentalSetNullName was intended to record the
+    // fact that the name would be nulled and fire a UseCounter if the name
+    // was accessed. We now actually null the name before running this.
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().ExperimentalSetNulledName();
   }
 
@@ -2895,9 +2894,10 @@
       previous_window && frame_->IsOutermostMainFrame() &&
       is_cross_site_cross_browsing_context_group_;
   if (should_clear_cross_site_cross_browsing_context_group_window_name) {
-    // TODO(shuuran): CrossSiteCrossBrowsingContextGroupSetNulledName will just
-    // record the fact that the name would be nulled and if the name is accessed
-    // after we will fire a UseCounter.
+    // TODO(shuuran): CrossSiteCrossBrowsingContextGroupSetNulledName was meant
+    // to record the fact that the name would be nulled and fire a UseCounter
+    // if the name was accessed. We now null the name before running this.
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().CrossSiteCrossBrowsingContextGroupSetNulledName();
   }
 
