--- a/chrome/browser/ui/omnibox/chrome_omnibox_navigation_observer.cc
+++ b/chrome/browser/ui/omnibox/chrome_omnibox_navigation_observer.cc
@@ -250,13 +250,6 @@
       profile_(profile),
       show_infobar_(std::move(show_infobar)) {
   NavigationUserData::CreateForNavigationHandle(navigation, this);
-  if (alternative_nav_match_.destination_url.is_valid()) {
-    loader_ = std::make_unique<AlternativeNavigationURLLoader>(
-        alternative_nav_match.destination_url, this,
-        base::BindOnce(
-            &ChromeOmniboxNavigationObserver::OnAlternativeLoaderDone, this),
-        GetURLLoaderFactory(loader_factory, profile));
-  }
 }
 
 ChromeOmniboxNavigationObserver::~ChromeOmniboxNavigationObserver() {
