--- a/media/base/supported_types.cc
+++ b/media/base/supported_types.cc
@@ -274,6 +274,9 @@
 }
 
 bool IsAACSupported(const AudioType& type) {
+#if BUILDFLAG(IS_LINUX) && BUILDFLAG(USE_PROPRIETARY_CODECS)
+  return true;
+#else
   if (type.profile != AudioCodecProfile::kXHE_AAC) {
     return true;
   }
@@ -283,6 +286,7 @@
 #else
   return false;
 #endif
+#endif  //BUILDFLAG(IS_LINUX) && BUILDFLAG(USE_PROPRIETARY_CODECS)
 }
 
 bool IsDolbyVisionProfileSupported(const VideoType& type) {
--- a/media/filters/ffmpeg_audio_decoder.cc
+++ b/media/filters/ffmpeg_audio_decoder.cc
@@ -96,15 +96,6 @@
     return;
   }
 
-  // TODO(dalecurtis): Remove this if ffmpeg ever gets xHE-AAC support.
-  if (config.profile() == AudioCodecProfile::kXHE_AAC) {
-    std::move(bound_init_cb)
-        .Run(DecoderStatus(DecoderStatus::Codes::kUnsupportedProfile)
-                 .WithData("decoder", "FFmpegAudioDecoder")
-                 .WithData("profile", config.profile()));
-    return;
-  }
-
   if (!ConfigureDecoder(config)) {
     av_sample_format_ = 0;
     std::move(bound_init_cb).Run(DecoderStatus::Codes::kUnsupportedConfig);
